# Assemble genomes of model organisms by Anchr

[TOC levels=1-3]: # ""

- [Assemble genomes of model organisms by Anchr](#assemble-genomes-of-model-organisms-by-anchr)
- [More tools on downloading and preprocessing data](#more-tools-on-downloading-and-preprocessing-data)
  - [Extra external executables](#extra-external-executables)
  - [Other leading assemblers](#other-leading-assemblers)
- [*Escherichia* virus Lambda](#escherichia-virus-lambda)
  - [lambda: reference](#lambda-reference)
  - [lambda: download](#lambda-download)
  - [lambda: template](#lambda-template)
  - [lambda: run](#lambda-run)
- [*Escherichia coli* str. K-12 substr. MG1655](#escherichia-coli-str-k-12-substr-mg1655)
  - [mg1655: reference](#mg1655-reference)
  - [mg1655: download](#mg1655-download)
  - [mg1655: template](#mg1655-template)
  - [mg1655: run](#mg1655-run)
- [*Escherichia coli* str. K-12 substr. DH5alpha](#escherichia-coli-str-k-12-substr-dh5alpha)
  - [dh5alpha: reference](#dh5alpha-reference)
  - [dh5alpha: download](#dh5alpha-download)
  - [dh5alpha: template](#dh5alpha-template)
  - [dh5alpha: run](#dh5alpha-run)


# More tools on downloading and preprocessing data

## Extra external executables

```shell script
brew install aria2 curl         # downloading tools
brew install miller

brew tap brewsci/bio
brew tap brewsci/science

brew install --HEAD quast       # assembly quality assessment. https://github.com/ablab/quast/issues/140
quast --test                    # may recompile the bundled nucmer

brew install --ignore-dependencies picard-tools

#kat comp -t 4 -n -o R 2_illumina/R1.fq.gz 2_illumina/R2.fq.gz -m 51
#kat plot spectra-mx -i -o R-spectra R-main.mx

```

## Other leading assemblers

```shell script
brew install spades
brew install megahit
brew install wang-q/tap/platanus

```

# *Escherichia* virus Lambda

Runs under macOS and Linux.

## lambda: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/lambda/1_genome
cd ~/data/anchr/lambda/1_genome

cp ~/data/anchr/ref/lambda/genome.fa .

touch paralogs.fa

```

## lambda: download

```shell script
mkdir -p ~/data/anchr/lambda/ena
cd ~/data/anchr/lambda/ena

cat << EOF > source.csv
SRX2365802,Lambda,HiSeq 2500
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -x 9 -s 3 -c -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

# sampling reads as test materials
seqtk sample -s 23 SRR5042715_1.fastq.gz 100000 | pigz > R1.fq.gz
seqtk sample -s 23 SRR5042715_2.fastq.gz 100000 | pigz > R2.fq.gz

```

| name   | srx        | platform | layout | ilength | srr        | spot     | base  |
|:-------|:-----------|:---------|:-------|:--------|:-----------|:---------|:------|
| Lambda | SRX2365802 | ILLUMINA | PAIRED |         | SRR5042715 | 16540237 | 3.33G |

* Illumina

```shell script
cd ~/data/anchr/lambda

mkdir -p 2_illumina
cd 2_illumina

seqtk sample -s 23 ../ena/SRR5042715_1.fastq.gz 100000 | pigz > R1.fq.gz
seqtk sample -s 23 ../ena/SRR5042715_2.fastq.gz 100000 | pigz > R2.fq.gz

```

## lambda: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=lambda

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 48502 \
    --parallel 4 \
    --xmx 6g \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --cutoff 5 --cutk 31" \
    --qual "20 25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "bcalm tadpole" \
    --statp 2 \
    --readl 108

```

## lambda: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=lambda

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 

bash 0_master.sh
#bash 0_cleanup.sh

```

Table: statInsertSize

| Group             |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|------:|-------------------------------:|
| R.genome.bbtools  | 430.6 |    413 | 367.3 |                         96.29% |
| R.tadpole.bbtools | 418.4 |    407 | 101.4 |                         82.81% |
| R.genome.picard   | 424.6 |    413 | 103.1 |                             FR |
| R.tadpole.picard  | 418.5 |    407 | 101.4 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 334       | 46251           | 0.0000       | 51.29   |
| R.31 | 297       | 46451           | 0.0000       | 50.64   |
| R.41 | 260       | 46286           | 0.0000       | 50.53   |
| R.51 | 217       | 46045           | 0.0000       | 50.65   |
| R.61 | 180       | 46218           | 0.0000       | 50.77   |
| R.71 | 143       | 46402           | 0.0000       | 50.77   |
| R.81 | 104       | 46385           | 0.0000       | 50.80   |


Table: statReads

| Name       |   N50 |    Sum |      # |
|:-----------|------:|-------:|-------:|
| Genome     | 48502 |  48502 |      1 |
| Paralogs   |     0 |      0 |      0 |
| Illumina.R |   108 |  21.6M | 200000 |
| trim.R     |   105 | 18.82M | 181790 |
| Q20L60     |   105 |  18.6M | 179679 |
| Q25L60     |   105 | 17.94M | 175041 |
| Q30L60     |   105 | 15.99M | 161943 |


Table: statTrimReads

| Name     | N50 |    Sum |      # |
|:---------|----:|-------:|-------:|
| clumpify | 108 | 21.56M | 199606 |
| highpass | 108 |  21.5M | 199078 |
| trim     | 105 | 18.82M | 181790 |
| filter   | 105 | 18.82M | 181790 |
| R1       | 105 |  9.48M |  90895 |
| R2       | 105 |  9.34M |  90895 |
| Rs       |   0 |      0 |      0 |


```text
#R.trim
#Matched	3807	1.91232%
#Name	Reads	ReadsPct
Reverse_adapter	1879	0.94385%
TruSeq_Universal_Adapter	1793	0.90065%
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	235058
#error_kmers	185792
#genomic_kmers	49266
#main_peak	269
#genome_size_in_peaks	2577144
#genome_size	2634418
#haploid_genome_size	48785
#fold_coverage	5
#haploid_fold_coverage	269
#ploidy	54
#het_rate	0.00080
#percent_repeat_in_peaks	0.000
#percent_repeat	98.128
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |    Sum |      # |
|:--------------|----:|-------:|-------:|
| clumped       | 105 | 18.82M | 181786 |
| ecco          | 105 | 18.82M | 181786 |
| eccc          | 105 | 18.82M | 181786 |
| ecct          | 105 | 18.77M | 181266 |
| extended      | 145 | 25.97M | 181266 |
| merged.raw    | 389 | 15.01M |  40051 |
| unmerged.raw  | 145 | 14.42M | 101164 |
| unmerged.trim | 145 | 14.42M | 101156 |
| M1            | 389 | 15.01M |  40029 |
| U1            | 145 |  7.27M |  50578 |
| U2            | 145 |  7.14M |  50578 |
| Us            |   0 |      0 |      0 |
| M.cor         | 267 | 29.46M | 181214 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 163.4 |    175 |  31.9 |          0.44% |
| M.ihist.merge.txt  | 374.9 |    383 |  47.4 |         44.19% |


Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |   EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|-------:|---------:|----------:|
| Q0L0.R   | 388.0 |  370.5 |    4.50% | "73" | 48.5K | 49.26K |     1.02 | 0:00'10'' |
| Q20L60.R | 383.5 |  368.4 |    3.94% | "73" | 48.5K | 48.97K |     1.01 | 0:00'11'' |
| Q25L60.R | 370.0 |  358.0 |    3.24% | "73" | 48.5K | 48.62K |     1.00 | 0:00'11'' |
| Q30L60.R | 329.9 |  321.7 |    2.50% | "73" | 48.5K | 48.52K |     1.00 | 0:00'20'' |


Table: statUnitigsAnchors.md

| Name          | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  99.45% |     14712 | 47.92K |  4 |        78 |   653 | 12 |   40.0 |  2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:10 |
| Q0L0X40P001   |   40.0 |  99.88% |     35738 | 48.18K |  3 |       152 |   642 | 10 |   41.0 |  5.0 |   8.7 |  82.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:10 |
| Q0L0X40P002   |   40.0 |  99.69% |     43840 | 49.95K |  4 |        41 |   510 | 16 |   40.0 |  5.0 |   8.3 |  80.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q0L0X80P000   |   80.0 |  98.66% |     10570 | 47.79K |  6 |        42 |   369 | 12 |   80.0 |  6.5 |  20.2 | 149.2 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q0L0X80P001   |   80.0 |  97.81% |     35479 | 47.15K |  4 |        62 |   292 | 10 |   80.0 |  9.0 |  17.7 | 160.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:09 |
| Q0L0X80P002   |   80.0 |  98.50% |     30645 | 46.13K |  3 |      1114 |  1.6K | 11 |   80.0 |  7.0 |  19.7 | 151.5 | "31,41,51,61,71,81" |   0:00:04 |   0:00:09 |
| Q20L60X40P000 |   40.0 |  99.57% |     11439 | 48.99K |  6 |      1100 | 1.64K | 16 |   39.0 |  3.0 |  10.0 |  72.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q20L60X40P001 |   40.0 |  99.72% |     35880 | 48.38K |  4 |        43 |   398 | 13 |   40.0 |  5.0 |   8.3 |  80.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q20L60X40P002 |   40.0 |  98.08% |     30591 |  46.2K |  2 |      1090 | 1.41K |  5 |   40.0 |  4.0 |   9.3 |  78.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q20L60X80P000 |   80.0 |  98.02% |      8172 | 46.45K |  6 |      1020 |  1.5K | 14 |   80.0 |  8.0 |  18.7 | 156.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:09 |
| Q20L60X80P001 |   80.0 |  98.70% |     30616 | 47.34K |  4 |       101 |   511 | 10 |   81.0 |  7.0 |  20.0 | 153.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:09 |
| Q20L60X80P002 |   80.0 |  98.90% |     14366 | 47.73K |  4 |        37 |   264 |  9 |   80.0 |  7.0 |  19.7 | 151.5 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q25L60X40P000 |   40.0 |  99.90% |     16218 | 46.94K |  4 |      1129 | 1.34K |  7 |   40.0 |  1.0 |  12.3 |  64.5 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q25L60X40P001 |   40.0 |  99.81% |     43839 | 49.07K |  3 |        50 |   333 |  9 |   41.0 |  5.0 |   8.7 |  82.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:10 |
| Q25L60X40P002 |   40.0 |  98.90% |     35192 | 47.67K |  3 |        55 |   273 |  8 |   40.0 |  5.0 |   8.3 |  80.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q25L60X80P000 |   80.0 |  99.94% |     43792 | 48.13K |  3 |       207 |   820 | 11 |   83.0 |  8.0 |  19.7 | 160.5 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q25L60X80P001 |   80.0 |  98.86% |     35238 | 47.75K |  4 |        53 |   252 | 10 |   80.0 |  8.0 |  18.7 | 156.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:09 |
| Q25L60X80P002 |   80.0 |  98.45% |     16517 | 46.09K |  4 |      1236 | 1.82K | 15 |   79.0 |  5.5 |  20.8 | 143.2 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q30L60X40P000 |   40.0 |  99.96% |     35693 | 48.09K |  4 |        92 |   874 | 13 |   41.0 |  4.0 |   9.7 |  79.5 | "31,41,51,61,71,81" |   0:00:03 |   0:00:10 |
| Q30L60X40P001 |   40.0 |  99.97% |     48100 |  48.1K |  1 |       186 |   668 | 10 |   41.0 |  4.0 |   9.7 |  79.5 | "31,41,51,61,71,81" |   0:00:03 |   0:00:10 |
| Q30L60X40P002 |   40.0 |  99.89% |     10604 | 39.87K |  5 |        84 |   498 |  8 |   41.0 |  1.0 |  12.7 |  66.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| Q30L60X80P000 |   80.0 |  98.70% |     35706 |  46.3K |  2 |      1236 | 1.51K |  7 |   83.0 | 10.0 |  17.7 | 166.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q30L60X80P001 |   80.0 |  99.93% |     35642 | 47.98K |  2 |       177 |   621 |  6 |   81.0 |  3.0 |  24.0 | 135.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:09 |
| Q30L60X80P002 |   80.0 |  99.95% |     14363 | 48.46K |  3 |       178 |   553 | 10 |   76.0 |  9.0 |  16.3 | 152.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:09 |


Table: statMRUnitigsAnchors.md

| Name      | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  99.41% |     44027 | 48.57K |  3 |        62 | 104 |  3 |   40.0 |  7.0 |   6.3 |  80.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:10 |
| MRX40P001 |   40.0 |  99.58% |     47322 | 48.56K |  2 |        36 |  39 |  2 |   39.0 |  6.5 |   6.5 |  78.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| MRX40P002 |   40.0 |  99.28% |     35793 | 48.34K |  3 |       130 | 337 |  5 |   40.0 |  4.0 |   9.3 |  78.0 | "31,41,51,61,71,81" |   0:00:03 |   0:00:09 |
| MRX80P000 |   80.0 |  99.41% |     44025 | 48.55K |  3 |        62 | 123 |  4 |   79.0 | 12.0 |  14.3 | 158.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| MRX80P002 |   80.0 |  99.17% |     35432 | 47.87K |  3 |       189 | 589 |  6 |   80.0 |  7.0 |  19.7 | 151.5 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |


Table: statMergeAnchors.md

| Name                       | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median | MAD | lower | upper | RunTimeAN |
|:---------------------------|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|----:|------:|------:|----------:|
| 7_merge_anchors            |   0.00% |     48514 | 48.51K |  1 |         0 |     0 |  0 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_mr_unitigs_anchors |   0.00% |     48514 | 48.51K |  1 |         0 |     0 |  0 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_unitigs_anchors    |   0.00% |     48336 | 48.34K |  1 |      1236 | 2.32K |  2 |    0.0 | 0.0 |   0.0 |   0.0 |           |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median | MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|----:|------:|------:|----------:|
| 8_spades     |  99.83% |     48024 | 48.02K |  1 |       267 | 498 |  2 |  370.0 | 1.0 | 122.3 | 559.5 |   0:00:10 |
| 8_mr_spades  |  99.57% |     26356 | 48.08K |  2 |       429 | 543 |  3 |  603.0 | 1.0 | 200.0 | 909.0 |   0:00:11 |
| 8_megahit    |  99.83% |     48024 | 48.02K |  1 |       267 | 495 |  2 |  370.0 | 1.0 | 122.3 | 559.5 |   0:00:10 |
| 8_mr_megahit |  99.74% |     48072 | 48.07K |  1 |       242 | 442 |  2 |  605.0 | 0.0 | 201.7 | 907.5 |   0:00:09 |


Table: statFinal

| Name                     |   N50 |   Sum |  # |
|:-------------------------|------:|------:|---:|
| Genome                   | 48502 | 48502 |  1 |
| Paralogs                 |     0 |     0 |  0 |
| 7_merge_anchors.anchors  | 48514 | 48514 |  1 |
| 7_merge_anchors.others   |     0 |     0 |  0 |
| spades.contig            | 48522 | 48600 |  2 |
| spades.scaffold          | 48522 | 48600 |  2 |
| spades.non-contained     | 48522 | 48522 |  1 |
| spades_MR.contig         | 48627 | 48753 |  2 |
| spades_MR.scaffold       | 48627 | 48753 |  2 |
| spades_MR.non-contained  | 48627 | 48627 |  1 |
| megahit.contig           | 48519 | 48519 |  1 |
| megahit.non-contained    | 48519 | 48519 |  1 |
| megahit_MR.contig        | 48514 | 48514 |  1 |
| megahit_MR.non-contained | 48514 | 48514 |  1 |
| platanus.non-contained   |     0 |     0 |  0 |


# *Escherichia coli* str. K-12 substr. MG1655

## mg1655: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/mg1655/1_genome
cd ~/data/anchr/mg1655/1_genome

cp ~/data/anchr/ref/mg1655/genome.fa .
cp ~/data/anchr/ref/mg1655/paralogs.fa .

```

## mg1655: download

* Illumina

```shell script
cd ~/data/anchr/mg1655

mkdir -p 2_illumina
cd 2_illumina

aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz
aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz

ln -s MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz R1.fq.gz
ln -s MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz R2.fq.gz

```

## mg1655: template

* Rsync to hpcc

```shell script
rsync -avP \
    ~/data/anchr/mg1655/ \
    wangq@202.119.37.251:data/anchr/mg1655

# rsync -avP wangq@202.119.37.251:data/anchr/mg1655/ ~/data/anchr/mg1655

```

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4641652 \
    --parallel 24 \
    --xmx 80g \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --tile --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --redo \
    \
    --extend

```

## mg1655: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 

# BASE_NAME=mg1655 bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
#bkill -J "${BASE_NAME}-*"

#bash 0_master.sh
#bash 0_cleanup.sh
#rm -fr 4_down_sampling 6_down_sampling

```

Table: statInsertSize

| Group             |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|------:|-------------------------------:|
| R.genome.bbtools  | 321.9 |    298 | 967.3 |                         96.08% |
| R.tadpole.bbtools | 294.9 |    296 |  22.0 |                         81.62% |
| R.genome.picard   | 298.2 |    298 |  18.0 |                             FR |
| R.tadpole.picard  | 295.1 |    296 |  21.6 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 292       | 4390759         | 0.1630       | 54.20   |
| R.31 | 265       | 4388871         | 0.1382       | 54.04   |
| R.41 | 239       | 4375332         | 0.1108       | 53.90   |
| R.51 | 216       | 4407280         | 0.1088       | 53.78   |
| R.61 | 191       | 4393867         | 0.1081       | 53.68   |
| R.71 | 168       | 4390152         | 0.1042       | 53.58   |
| R.81 | 144       | 4387398         | 0.0989       | 53.50   |


Table: statReads

| Name       |     N50 |     Sum |        # |
|:-----------|--------:|--------:|---------:|
| Genome     | 4641652 | 4641652 |        1 |
| Paralogs   |    1937 |  187300 |      106 |
| Illumina.R |     151 |   1.73G | 11458940 |
| trim.R     |     149 |   1.43G | 10357292 |
| Q25L60     |     148 |   1.32G |  9933554 |
| Q30L60     |     128 |   1.11G |  9306325 |


Table: statTrimReads

| Name           | N50 |     Sum |        # |
|:---------------|----:|--------:|---------:|
| clumpify       | 151 |   1.73G | 11439000 |
| filteredbytile | 151 |   1.67G | 11054084 |
| highpass       | 151 |   1.66G | 10981112 |
| trim           | 149 |   1.43G | 10357292 |
| filter         | 149 |   1.43G | 10357292 |
| R1             | 150 |  735.6M |  5178646 |
| R2             | 144 | 690.02M |  5178646 |
| Rs             |   0 |       0 |        0 |


```text
#R.trim
#Matched	17472	0.15911%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	20701875
#error_kmers	16177144
#genomic_kmers	4524731
#main_peak	246
#genome_size_in_peaks	4596479
#genome_size	4625996
#haploid_genome_size	4625996
#fold_coverage	246
#haploid_fold_coverage	246
#ploidy	1
#percent_repeat_in_peaks	1.563
#percent_repeat	1.746
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |    Sum |        # |
|:--------------|----:|-------:|---------:|
| clumped       | 149 |  1.43G | 10356116 |
| ecco          | 149 |  1.43G | 10356116 |
| eccc          | 149 |  1.43G | 10356116 |
| ecct          | 149 |  1.42G | 10307810 |
| extended      | 189 |  1.83G | 10307810 |
| merged.raw    | 339 |  1.72G |  5082018 |
| unmerged.raw  | 175 | 21.16M |   143774 |
| unmerged.trim | 175 | 21.15M |   143714 |
| M1            | 339 |  1.71G |  5054540 |
| U1            | 181 |  11.1M |    71857 |
| U2            | 168 | 10.05M |    71857 |
| Us            |   0 |      0 |        0 |
| M.cor         | 338 |  1.73G | 10252794 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 271.8 |    277 |  23.4 |         10.84% |
| M.ihist.merge.txt  | 337.7 |    338 |  19.3 |         98.61% |


Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 307.1 |  285.5 |    7.05% | "67" | 4.64M | 4.66M |     1.00 | 0:02'29'' |
| Q25L60.R | 283.9 |  273.0 |    3.84% | "87" | 4.64M | 4.57M |     0.98 | 0:02'23'' |
| Q30L60.R | 238.2 |  233.4 |    2.03% | "73" | 4.64M | 4.55M |     0.98 | 0:02'04'' |


Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |    # | N50Others |     Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|-----:|----------:|--------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  96.08% |      7027 | 4.35M |  867 |       674 | 191.65K | 1876 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:45 |   0:00:33 |
| Q0L0X40P001   |   40.0 |  96.08% |      7220 | 4.32M |  842 |       817 | 195.83K | 1793 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:45 |   0:00:34 |
| Q0L0X40P002   |   40.0 |  96.16% |      7587 | 4.37M |  830 |       582 | 181.47K | 1807 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:46 |   0:00:34 |
| Q0L0X80P000   |   80.0 |  92.26% |      4680 | 4.28M | 1183 |        80 | 145.62K | 2484 |   79.0 | 6.0 |  20.3 | 145.5 | "31,41,51,61,71,81" |   0:01:15 |   0:00:35 |
| Q0L0X80P001   |   80.0 |  91.99% |      4806 | 4.26M | 1153 |        77 | 142.02K | 2413 |   79.0 | 6.0 |  20.3 | 145.5 | "31,41,51,61,71,81" |   0:01:14 |   0:00:34 |
| Q0L0X80P002   |   80.0 |  91.90% |      4958 | 4.26M | 1170 |        85 | 148.03K | 2438 |   79.0 | 6.0 |  20.3 | 145.5 | "31,41,51,61,71,81" |   0:01:14 |   0:00:34 |
| Q25L60X40P000 |   40.0 |  97.83% |      9297 | 4.18M |  662 |       897 | 233.21K | 1624 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:46 |   0:00:37 |
| Q25L60X40P001 |   40.0 |  97.70% |      9508 | 4.13M |  649 |       738 | 202.02K | 1540 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:45 |   0:00:38 |
| Q25L60X40P002 |   40.0 |  97.79% |     10673 | 4.28M |  635 |       720 | 218.65K | 1565 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:46 |   0:00:37 |
| Q25L60X80P000 |   80.0 |  96.39% |      9120 | 4.42M |  720 |       311 |  118.9K | 1617 |   79.0 | 5.0 |  21.3 | 141.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:36 |
| Q25L60X80P001 |   80.0 |  96.62% |      9529 | 4.43M |  699 |       427 | 120.86K | 1607 |   79.0 | 5.0 |  21.3 | 141.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:36 |
| Q25L60X80P002 |   80.0 |  96.66% |      9463 | 4.44M |  685 |       315 | 113.07K | 1568 |   79.0 | 5.0 |  21.3 | 141.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:38 |
| Q30L60X40P000 |   40.0 |  98.54% |     15297 | 4.04M |  458 |       963 | 193.35K | 1503 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:00:44 |   0:00:41 |
| Q30L60X40P001 |   40.0 |  98.54% |      9440 | 3.55M |  562 |      1000 | 265.43K | 1456 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:43 |
| Q30L60X40P002 |   40.0 |  98.51% |     14820 | 4.15M |  467 |       920 | 193.23K | 1541 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:00:44 |   0:00:42 |
| Q30L60X80P000 |   80.0 |  98.53% |     13670 | 4.02M |  471 |       901 | 171.59K | 1305 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:11 |   0:00:46 |
| Q30L60X80P001 |   80.0 |  98.44% |     16384 | 4.27M |  420 |       549 | 122.74K | 1296 |   80.0 | 5.0 |  21.7 | 142.5 | "31,41,51,61,71,81" |   0:01:11 |   0:00:44 |


Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|--------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  95.96% |     21056 | 4.46M | 360 |       131 |  82.39K | 713 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:57 |   0:00:31 |
| MRX40P001 |   40.0 |  95.93% |     20992 | 4.46M | 380 |       132 |   84.4K | 719 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:57 |   0:00:31 |
| MRX40P002 |   40.0 |  96.04% |     19676 | 4.46M | 362 |       133 |  86.47K | 685 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:57 |   0:00:32 |
| MRX80P000 |   80.0 |  94.79% |     15173 | 4.44M | 451 |       123 | 101.69K | 936 |   79.0 | 5.0 |  21.3 | 141.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:32 |
| MRX80P001 |   80.0 |  95.01% |     15097 | 4.43M | 436 |       129 | 109.08K | 907 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:37 |   0:00:33 |
| MRX80P002 |   80.0 |  94.94% |     15166 | 4.44M | 452 |       126 |    106K | 940 |   79.0 | 5.0 |  21.3 | 141.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:32 |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|--------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.31% |     16372 | 4.44M | 421 |       726 |  194.1K | 1975 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:05 |   0:00:47 |
| Q0L0X40P001   |   40.0 |  98.35% |     22369 |  4.5M | 348 |       323 | 144.56K | 1882 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:01:05 |   0:00:46 |
| Q0L0X40P002   |   40.0 |  98.33% |     16890 | 4.38M | 417 |       328 | 164.98K | 1981 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:04 |   0:00:46 |
| Q0L0X80P000   |   80.0 |  98.02% |     18742 | 4.45M | 384 |       417 | 127.69K | 1461 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:15 |   0:00:46 |
| Q0L0X80P001   |   80.0 |  98.03% |     19574 | 4.49M | 376 |       382 | 122.36K | 1448 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:15 |   0:00:44 |
| Q0L0X80P002   |   80.0 |  97.97% |     19084 |  4.5M | 382 |        97 | 108.41K | 1455 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:15 |   0:00:45 |
| Q25L60X40P000 |   40.0 |  98.39% |     17679 | 4.49M | 447 |       718 |  180.5K | 2187 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:01:04 |   0:00:46 |
| Q25L60X40P001 |   40.0 |  98.34% |     18744 | 4.48M | 418 |       526 | 164.99K | 2111 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:01:05 |   0:00:43 |
| Q25L60X40P002 |   40.0 |  98.41% |     14258 |  4.4M | 518 |       854 | 273.35K | 2237 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:05 |   0:00:43 |
| Q25L60X80P000 |   80.0 |  98.54% |     18568 | 4.41M | 394 |       402 | 136.88K | 1671 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:49 |
| Q25L60X80P001 |   80.0 |  98.56% |     16689 | 4.41M | 395 |       508 | 150.44K | 1699 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:14 |   0:00:47 |
| Q25L60X80P002 |   80.0 |  98.50% |     22707 | 4.46M | 338 |       115 | 110.11K | 1572 |   80.0 | 5.0 |  21.7 | 142.5 | "31,41,51,61,71,81" |   0:01:14 |   0:00:47 |
| Q30L60X40P000 |   40.0 |  97.95% |     10808 | 4.35M | 634 |       976 | 273.67K | 2430 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:01:05 |   0:00:41 |
| Q30L60X40P001 |   40.0 |  97.88% |     10534 | 4.39M | 653 |      1013 | 291.26K | 2474 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:01:05 |   0:00:40 |
| Q30L60X40P002 |   40.0 |  97.91% |     10344 | 4.39M | 658 |       945 | 252.82K | 2435 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:01:06 |   0:00:42 |
| Q30L60X80P000 |   80.0 |  98.62% |     16396 | 4.38M | 421 |       863 | 187.05K | 1889 |   80.0 | 5.0 |  21.7 | 142.5 | "31,41,51,61,71,81" |   0:01:14 |   0:00:50 |
| Q30L60X80P001 |   80.0 |  98.61% |     19679 | 4.43M | 377 |       989 | 164.89K | 1842 |   80.0 | 6.0 |  20.7 | 147.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:49 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.84% |     27420 | 2.79M | 201 |       502 | 55.09K | 315 |   39.0 | 1.0 |  12.0 |  63.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:30 |
| MRX40P001 |   40.0 |  96.87% |     29173 | 3.48M | 229 |       306 |  53.4K | 318 |   39.0 | 1.0 |  12.0 |  63.0 | "31,41,51,61,71,81" |   0:01:08 |   0:00:32 |
| MRX40P002 |   40.0 |  96.99% |     21902 | 3.15M | 234 |       630 | 60.92K | 305 |   39.0 | 1.0 |  12.0 |  63.0 | "31,41,51,61,71,81" |   0:01:09 |   0:00:31 |
| MRX80P000 |   80.0 |  96.26% |     33101 | 4.39M | 228 |       124 | 50.81K | 439 |   79.0 | 3.0 |  23.3 | 132.0 | "31,41,51,61,71,81" |   0:01:22 |   0:00:33 |
| MRX80P001 |   80.0 |  96.32% |     35597 | 4.49M | 217 |       126 | 50.98K | 415 |   79.0 | 3.0 |  23.3 | 132.0 | "31,41,51,61,71,81" |   0:01:23 |   0:00:33 |
| MRX80P002 |   80.0 |  96.29% |     32850 |  4.4M | 213 |       135 | 54.45K | 414 |   79.0 | 3.0 |  23.3 | 132.0 | "31,41,51,61,71,81" |   0:01:21 |   0:00:32 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|--------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.17% |     16372 |  4.3M | 408 |       473 | 133.24K | 1341 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:43 |
| Q0L0X40P001   |   40.0 |  98.23% |     15459 | 4.21M | 443 |       715 | 152.35K | 1356 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:34 |   0:00:42 |
| Q0L0X40P002   |   40.0 |  98.28% |     16021 |  4.3M | 435 |       529 |  148.9K | 1440 |   39.0 | 2.0 |  11.0 |  67.5 | "31,41,51,61,71,81" |   0:00:35 |   0:00:43 |
| Q0L0X80P000   |   80.0 |  97.06% |     13989 | 4.43M | 488 |       269 | 101.13K | 1216 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:00:44 |   0:00:36 |
| Q0L0X80P001   |   80.0 |  97.15% |     14198 | 4.47M | 506 |       328 |  98.18K | 1222 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:00:45 |   0:00:38 |
| Q0L0X80P002   |   80.0 |  97.22% |     14124 | 4.48M | 495 |       163 |  94.51K | 1245 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:00:44 |   0:00:37 |
| Q25L60X40P000 |   40.0 |  98.59% |     13618 | 4.11M | 481 |       758 | 176.14K | 1534 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:45 |
| Q25L60X40P001 |   40.0 |  98.52% |     14274 | 4.19M | 467 |       711 | 198.98K | 1467 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:44 |
| Q25L60X40P002 |   40.0 |  98.51% |     15127 |  4.1M | 458 |       882 | 207.66K | 1444 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:44 |
| Q25L60X80P000 |   80.0 |  97.99% |     16100 | 4.33M | 448 |       635 | 116.32K | 1245 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:00:44 |   0:00:41 |
| Q25L60X80P001 |   80.0 |  98.09% |     15836 | 4.41M | 430 |       572 | 123.42K | 1249 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:00:45 |   0:00:41 |
| Q25L60X80P002 |   80.0 |  98.12% |     16272 | 4.42M | 436 |       481 | 107.46K | 1232 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:00:44 |   0:00:42 |
| Q30L60X40P000 |   40.0 |  98.52% |     15297 | 4.18M | 465 |       931 | 213.83K | 1676 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:00:37 |   0:00:42 |
| Q30L60X40P001 |   40.0 |  98.52% |     14192 | 4.19M | 479 |       925 | 215.33K | 1747 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:00:35 |   0:00:43 |
| Q30L60X40P002 |   40.0 |  98.50% |     15101 | 4.32M | 474 |       928 | 196.57K | 1709 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:00:35 |   0:00:41 |
| Q30L60X80P000 |   80.0 |  98.59% |     17782 |  4.3M | 390 |       813 | 133.98K | 1339 |   80.0 | 5.0 |  21.7 | 142.5 | "31,41,51,61,71,81" |   0:00:41 |   0:00:47 |
| Q30L60X80P001 |   80.0 |  98.57% |     18702 | 4.27M | 387 |       549 | 135.17K | 1381 |   80.0 | 5.0 |  21.7 | 142.5 | "31,41,51,61,71,81" |   0:00:43 |   0:00:47 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.55% |     25647 | 3.16M | 236 |       383 | 62.81K | 359 |   39.0 | 1.0 |  12.0 |  63.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:30 |
| MRX40P001 |   40.0 |  96.49% |     27844 | 3.47M | 253 |       201 | 53.59K | 345 |   39.0 | 1.0 |  12.0 |  63.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:30 |
| MRX40P002 |   40.0 |  96.57% |     20701 | 3.08M | 244 |       604 | 63.68K | 324 |   39.0 | 1.0 |  12.0 |  63.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:29 |
| MRX80P000 |   80.0 |  96.01% |     27958 | 4.38M | 275 |       125 | 61.06K | 533 |   79.0 | 3.0 |  23.3 | 132.0 | "31,41,51,61,71,81" |   0:00:51 |   0:00:33 |
| MRX80P001 |   80.0 |  96.09% |     30616 | 4.48M | 259 |       126 | 60.58K | 504 |   79.0 | 3.0 |  23.3 | 132.0 | "31,41,51,61,71,81" |   0:00:50 |   0:00:32 |
| MRX80P002 |   80.0 |  96.01% |     29037 | 4.39M | 255 |       139 | 68.44K | 496 |   79.0 | 3.0 |  23.3 | 132.0 | "31,41,51,61,71,81" |   0:00:49 |   0:00:32 |


Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|----:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  97.37% |     41165 |  4.5M | 192 |      4090 | 979.94K | 381 |  285.0 | 11.0 |  84.0 | 477.0 |   0:00:57 |
| 7_merge_mr_unitigs_bcalm      |  98.75% |     40222 | 4.47M | 201 |      1800 | 163.83K |  94 |  285.0 | 10.5 |  84.5 | 474.8 |   0:01:22 |
| 7_merge_mr_unitigs_superreads |  98.76% |     40236 |  4.5M | 208 |      1358 |  97.14K |  69 |  285.0 | 11.0 |  84.0 | 477.0 |   0:01:25 |
| 7_merge_mr_unitigs_tadpole    |  98.86% |     40221 |  4.5M | 204 |      2889 | 200.16K |  98 |  284.0 | 10.0 |  84.7 | 471.0 |   0:01:41 |
| 7_merge_unitigs_bcalm         |  98.96% |     49577 | 4.51M | 184 |      4237 | 582.55K | 237 |  285.0 | 14.0 |  81.0 | 490.5 |   0:01:35 |
| 7_merge_unitigs_superreads    |  98.95% |     43480 | 4.51M | 189 |      1537 | 381.22K | 266 |  284.0 | 14.0 |  80.7 | 489.0 |   0:01:38 |
| 7_merge_unitigs_tadpole       |  99.01% |     48895 | 4.51M | 183 |      1715 | 391.76K | 239 |  285.0 | 14.0 |  81.0 | 490.5 |   0:01:48 |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.54% |     54546 | 2.71M | 102 |      1117 | 25.78K | 136 |  285.0 | 10.0 |  85.0 | 472.5 |   0:00:40 |
| 8_mr_spades  |  98.37% |    125743 | 4.31M |  77 |      1029 | 32.81K | 152 |  371.0 | 10.0 | 113.7 | 601.5 |   0:00:42 |
| 8_megahit    |  98.16% |     40805 | 3.47M | 157 |      1133 | 33.11K | 216 |  285.0 | 10.0 |  85.0 | 472.5 |   0:00:40 |
| 8_mr_megahit |  98.76% |    117694 | 4.33M |  76 |      1064 |  31.6K | 143 |  371.0 |  9.5 | 114.2 | 599.2 |   0:00:42 |
| 8_platanus   |  97.87% |     41234 | 1.97M |  92 |      1027 |    21K | 114 |  285.0 |  8.5 |  86.5 | 465.8 |   0:00:41 |


Table: statFinal

| Name                     |     N50 |     Sum |    # |
|:-------------------------|--------:|--------:|-----:|
| Genome                   | 4641652 | 4641652 |    1 |
| Paralogs                 |    1937 |  187300 |  106 |
| 7_merge_anchors.anchors  |   41165 | 4499645 |  192 |
| 7_merge_anchors.others   |    4090 |  979940 |  381 |
| glue_anchors             |   41165 | 4499593 |  191 |
| fill_anchors             |  126710 | 4535194 |   75 |
| spades.contig            |  132608 | 4573776 |  137 |
| spades.scaffold          |  133063 | 4573960 |  133 |
| spades.non-contained     |  132608 | 4555564 |   75 |
| mr_spades.contig         |  148607 | 4587780 |  149 |
| mr_spades.scaffold       |  148607 | 4587980 |  147 |
| mr_spades.non-contained  |  148607 | 4570445 |   76 |
| megahit.contig           |   82825 | 4568113 |  149 |
| megahit.non-contained    |   82825 | 4550837 |  106 |
| mr_megahit.contig        |  132896 | 4612492 |  138 |
| mr_megahit.non-contained |  132896 | 4584959 |   68 |
| platanus.contig          |   14854 | 4712374 | 1172 |
| platanus.scaffold        |  148483 | 4577844 |  142 |
| platanus.non-contained   |  176491 | 4560060 |   64 |


# *Escherichia coli* str. K-12 substr. DH5alpha

## dh5alpha: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/dh5alpha/1_genome
cd ~/data/anchr/dh5alpha/1_genome

cp ~/data/anchr/ref/dh5alpha/genome.fa .
cp ~/data/anchr/ref/dh5alpha/paralogs.fa .

```

## dh5alpha: download

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p ena
cd ena

cat << EOF > source.csv
SRP251726,dh5alpha,HiSeq 2500 PE125
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -x 9 -s 3 -c -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

```

| name     | srx        | platform        | layout | ilength | srr         | spot    | base  |
|:---------|:-----------|:----------------|:-------|:--------|:------------|:--------|:------|
| dh5alpha | SRX7856678 | ILLUMINA        | PAIRED |         | SRR11245239 | 5881654 | 1.37G |
| dh5alpha | SRX7856679 | OXFORD_NANOPORE | SINGLE |         | SRR11245238 | 346489  | 3.35G |


* Illumina

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p 2_illumina
cd 2_illumina

ln -s ../ena/SRR11245239_1.fastq.gz R1.fq.gz
ln -s ../ena/SRR11245239_2.fastq.gz R2.fq.gz

```

## dh5alpha: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4583637 \
    --parallel 24 \
    --xmx 80g \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --redo \
    \
    --extend

```

## dh5alpha: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 

# BASE_NAME=dh5alpha bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
#bkill -J "${BASE_NAME}-*"

#bash 0_master.sh
#bash 0_cleanup.sh
#rm -fr 4_down_sampling 6_down_sampling

```


Table: statInsertSize

| Group             |  Mean | Median |  STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|-------:|-------------------------------:|
| R.genome.bbtools  | 432.8 |    346 | 1241.8 |                         99.98% |
| R.tadpole.bbtools | 389.1 |    341 |  206.1 |                         95.10% |
| R.genome.picard   | 394.8 |    346 |  208.3 |                             FR |
| R.tadpole.picard  | 389.1 |    341 |  205.8 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 260       | 4346013         | 0.0814       | 51.62   |
| R.31 | 233       | 4360998         | 0.0563       | 51.54   |
| R.41 | 209       | 4433088         | 0.0607       | 51.50   |
| R.51 | 182       | 4413726         | 0.0150       | 51.47   |
| R.61 | 157       | 4456634         | 0.0332       | 51.45   |
| R.71 | 132       | 4495949         | 0.0194       | 51.44   |
| R.81 | 107       | 4475448         | 0.0160       | 51.42   |


Table: statReads

| Name       |     N50 |     Sum |        # |
|:-----------|--------:|--------:|---------:|
| Genome     | 4583637 | 4583637 |        1 |
| Paralogs   |    1737 |  188158 |      111 |
| Illumina.R |     125 |   1.47G | 11763308 |
| trim.R     |     125 |   1.37G | 10962178 |
| Q25L60     |     125 |   1.25G | 10280852 |
| Q30L60     |     125 |   1.13G |  9405463 |


Table: statTrimReads

| Name     | N50 |     Sum |        # |
|:---------|----:|--------:|---------:|
| clumpify | 125 |   1.37G | 10970448 |
| highpass | 125 |   1.37G | 10966054 |
| trim     | 125 |   1.37G | 10962178 |
| filter   | 125 |   1.37G | 10962178 |
| R1       | 125 | 682.99M |  5481089 |
| R2       | 125 | 683.63M |  5481089 |
| Rs       |   0 |       0 |        0 |


```text
#R.trim
#Matched	5620	0.05125%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	35779038
#error_kmers	31301375
#genomic_kmers	4477663
#main_peak	223
#genome_size_in_peaks	4561456
#genome_size	4571195
#haploid_genome_size	4571195
#fold_coverage	223
#haploid_fold_coverage	223
#ploidy	1
#percent_repeat_in_peaks	1.841
#percent_repeat	1.987
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |     Sum |        # |
|:--------------|----:|--------:|---------:|
| clumped       | 125 |   1.37G | 10959360 |
| ecco          | 125 |   1.37G | 10959360 |
| eccc          | 125 |   1.37G | 10959360 |
| ecct          | 125 |   1.37G | 10952780 |
| extended      | 165 |    1.8G | 10952780 |
| merged.raw    | 343 |    1.1G |  3511394 |
| unmerged.raw  | 165 | 645.94M |  3929992 |
| unmerged.trim | 165 | 645.94M |  3929992 |
| M1            | 343 |   1.06G |  3404121 |
| U1            | 165 | 322.88M |  1964996 |
| U2            | 165 | 323.06M |  1964996 |
| Us            |   0 |       0 |        0 |
| M.cor         | 250 |   1.71G | 10738234 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 180.7 |    181 |  32.1 |         23.40% |
| M.ihist.merge.txt  | 312.5 |    310 |  87.7 |         64.12% |


Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 298.2 |  262.2 |   12.07% | "87" | 4.58M |  4.6M |     1.00 | 0:02'30'' |
| Q25L60.R | 273.7 |  253.6 |    7.35% | "87" | 4.58M | 4.53M |     0.99 | 0:02'24'' |
| Q30L60.R | 246.5 |  232.4 |    5.71% | "87" | 4.58M | 4.52M |     0.99 | 0:02'13'' |


Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.21% |     27714 | 4.42M | 300 |       764 | 61.11K |  665 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:33 |
| Q0L0X40P001   |   40.0 |  97.27% |     23854 | 4.41M | 323 |       741 | 66.58K |  706 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:33 |
| Q0L0X40P002   |   40.0 |  97.02% |     22855 |  4.3M | 305 |       886 | 66.53K |  672 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:32 |
| Q0L0X80P000   |   80.0 |  95.89% |     12523 | 4.36M | 546 |        45 | 60.97K | 1149 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:07 |   0:00:34 |
| Q0L0X80P001   |   80.0 |  95.93% |     12841 | 4.31M | 530 |        41 | 51.84K | 1119 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:08 |   0:00:34 |
| Q0L0X80P002   |   80.0 |  95.97% |     12651 | 4.41M | 539 |        43 | 57.47K | 1151 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:07 |   0:00:35 |
| Q25L60X40P000 |   40.0 |  97.63% |     35158 | 4.32M | 228 |       278 | 36.67K |  506 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:35 |
| Q25L60X40P001 |   40.0 |  97.64% |     31891 | 4.43M | 248 |       972 | 61.78K |  568 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:34 |
| Q25L60X40P002 |   40.0 |  97.66% |     34674 | 4.43M | 228 |       668 | 50.05K |  546 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:34 |
| Q25L60X80P000 |   80.0 |  96.96% |     24934 | 4.33M | 310 |        41 |  32.2K |  688 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:05 |   0:00:35 |
| Q25L60X80P001 |   80.0 |  97.10% |     25202 |  4.4M | 307 |        42 | 35.68K |  699 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:05 |   0:00:35 |
| Q25L60X80P002 |   80.0 |  97.14% |     23380 | 4.34M | 314 |        47 |  40.6K |  718 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:35 |
| Q30L60X40P000 |   40.0 |  97.99% |     39350 | 4.33M | 215 |       666 | 46.09K |  522 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:39 |
| Q30L60X40P001 |   40.0 |  97.81% |     35162 | 4.43M | 218 |       935 | 62.04K |  536 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:35 |
| Q30L60X40P002 |   40.0 |  97.80% |     32722 | 4.43M | 241 |       714 | 58.26K |  544 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:38 |
| Q30L60X80P000 |   80.0 |  97.26% |     28270 | 4.44M | 269 |        49 | 35.64K |  608 |   80.0 | 3.0 |  23.7 | 133.5 | "31,41,51,61,71,81" |   0:01:06 |   0:00:36 |
| Q30L60X80P001 |   80.0 |  97.22% |     26298 | 4.44M | 290 |        50 | 38.19K |  648 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:05 |   0:00:35 |


Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.63% |     57829 |  4.3M | 137 |       145 | 42.89K | 334 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:51 |   0:00:37 |
| MRX40P001 |   40.0 |  97.58% |     54182 | 4.43M | 146 |       342 | 45.46K | 352 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:36 |
| MRX40P002 |   40.0 |  97.60% |     57818 | 4.27M | 134 |        96 | 32.06K | 333 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:36 |
| MRX80P000 |   80.0 |  97.51% |     59600 | 4.27M | 124 |        85 |  32.8K | 335 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:39 |
| MRX80P001 |   80.0 |  97.53% |     64057 | 4.45M | 119 |        81 | 28.77K | 324 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:39 |
| MRX80P002 |   80.0 |  97.57% |     63528 | 4.44M | 123 |        89 | 33.83K | 333 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:39 |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.38% |     46117 |  4.4M | 167 |       111 | 76.45K |  987 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:44 |
| Q0L0X40P001   |   40.0 |  98.49% |     49695 | 4.44M | 176 |       553 | 92.15K | 1081 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:46 |
| Q0L0X40P002   |   40.0 |  98.43% |     42230 | 4.32M | 174 |       489 | 80.99K | 1020 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:45 |
| Q0L0X80P000   |   80.0 |  97.71% |     64104 | 4.44M | 125 |       678 | 34.12K |  326 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:23 |   0:00:38 |
| Q0L0X80P001   |   80.0 |  97.85% |     64127 | 4.45M | 125 |       650 | 30.33K |  325 |   79.0 | 4.0 |  22.3 | 136.5 | "31,41,51,61,71,81" |   0:01:23 |   0:00:40 |
| Q0L0X80P002   |   80.0 |  97.94% |     59644 | 4.43M | 127 |       202 | 25.74K |  320 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:23 |   0:00:42 |
| Q25L60X40P000 |   40.0 |  98.49% |     42369 | 4.23M | 181 |       381 | 82.38K | 1065 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:46 |
| Q25L60X40P001 |   40.0 |  98.52% |     44726 | 4.39M | 175 |       256 | 83.97K | 1084 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:45 |
| Q25L60X40P002 |   40.0 |  98.52% |     46657 | 4.44M | 175 |       161 | 80.43K | 1076 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:45 |
| Q25L60X80P000 |   80.0 |  98.12% |     67333 | 4.45M | 124 |       428 | 29.64K |  341 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:24 |   0:00:42 |
| Q25L60X80P001 |   80.0 |  98.06% |     64123 | 4.45M | 123 |       638 | 35.61K |  345 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:24 |   0:00:42 |
| Q25L60X80P002 |   80.0 |  98.04% |     63658 | 4.45M | 122 |       585 | 30.97K |  336 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:23 |   0:00:41 |
| Q30L60X40P000 |   40.0 |  98.53% |     42198 | 4.44M | 195 |        88 | 81.36K | 1159 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:47 |
| Q30L60X40P001 |   40.0 |  98.49% |     44709 | 4.43M | 177 |       491 | 90.27K | 1123 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:46 |
| Q30L60X40P002 |   40.0 |  98.51% |     43499 | 4.26M | 183 |       535 | 87.83K | 1076 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:44 |
| Q30L60X80P000 |   80.0 |  98.05% |     64117 | 4.45M | 124 |       291 | 30.78K |  352 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:23 |   0:00:40 |
| Q30L60X80P001 |   80.0 |  98.03% |     63659 | 4.45M | 125 |       909 | 36.67K |  368 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:41 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.49% |     57829 | 4.37M | 139 |       873 | 43.29K | 296 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:15 |   0:00:36 |
| MRX40P001 |   40.0 |  97.26% |     63599 | 4.44M | 123 |       261 | 37.97K | 304 |   40.0 | 3.0 |  10.3 |  73.5 | "31,41,51,61,71,81" |   0:01:14 |   0:00:34 |
| MRX40P002 |   40.0 |  97.44% |     57818 | 4.44M | 140 |       373 |  40.9K | 308 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:37 |
| MRX80P000 |   80.0 |  96.98% |     64083 | 4.41M | 117 |        92 | 25.19K | 231 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:34 |
| MRX80P001 |   80.0 |  96.85% |     73609 | 4.45M | 115 |        84 | 21.67K | 229 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:34 |
| MRX80P002 |   80.0 |  96.99% |     64055 | 4.45M | 118 |       104 | 27.15K | 240 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:01:31 |   0:00:35 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.01% |     47105 | 4.21M | 150 |      1030 | 54.74K | 436 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:39 |
| Q0L0X40P001   |   40.0 |  98.28% |     50471 | 4.43M | 163 |      1003 | 66.04K | 470 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:43 |
| Q0L0X40P002   |   40.0 |  98.17% |     43278 | 4.13M | 156 |       704 | 46.07K | 459 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:41 |
| Q0L0X80P000   |   80.0 |  97.10% |     60835 | 4.36M | 131 |       678 | 20.44K | 261 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:35 |
| Q0L0X80P001   |   80.0 |  97.12% |     60831 | 4.34M | 131 |        76 |  18.7K | 264 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:32 |
| Q0L0X80P002   |   80.0 |  97.10% |     54683 | 4.45M | 144 |       528 | 23.34K | 274 |   80.0 | 3.5 |  23.2 | 135.8 | "31,41,51,61,71,81" |   0:00:42 |   0:00:32 |
| Q25L60X40P000 |   40.0 |  98.37% |     43291 | 4.25M | 171 |       532 | 45.45K | 484 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:41 |
| Q25L60X40P001 |   40.0 |  98.28% |     50351 | 4.32M | 169 |      1026 | 57.23K | 498 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:41 |
| Q25L60X40P002 |   40.0 |  98.41% |     52432 | 4.44M | 161 |      1005 |  60.5K | 503 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:42 |
| Q25L60X80P000 |   80.0 |  97.34% |     67333 | 4.45M | 129 |       242 | 19.65K | 266 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:34 |
| Q25L60X80P001 |   80.0 |  97.35% |     63583 | 4.45M | 129 |       708 | 24.15K | 268 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:35 |
| Q25L60X80P002 |   80.0 |  97.32% |     60833 | 4.44M | 132 |       897 | 25.19K | 259 |   80.0 | 3.5 |  23.2 | 135.8 | "31,41,51,61,71,81" |   0:00:43 |   0:00:34 |
| Q30L60X40P000 |   40.0 |  98.36% |     42232 | 4.32M | 179 |       770 | 60.89K | 558 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:42 |
| Q30L60X40P001 |   40.0 |  98.27% |     43251 | 4.43M | 169 |       768 | 52.14K | 508 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:40 |
| Q30L60X40P002 |   40.0 |  98.53% |     44359 |  4.3M | 179 |      1359 | 98.41K | 563 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:44 |
| Q30L60X80P000 |   80.0 |  97.59% |     60831 | 4.45M | 128 |       225 | 21.31K | 279 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:34 |
| Q30L60X80P001 |   80.0 |  97.46% |     61151 | 4.45M | 131 |      1022 | 24.11K | 262 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:33 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.10% |     58632 | 4.31M | 134 |       498 |  33.4K | 248 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:31 |
| MRX40P001 |   40.0 |  96.80% |     53984 | 4.12M | 136 |       522 |  34.5K | 251 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:39 |   0:00:31 |
| MRX40P002 |   40.0 |  96.98% |     57818 | 4.26M | 136 |       409 | 31.29K | 252 |   40.0 | 2.0 |  11.3 |  69.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:31 |
| MRX80P000 |   80.0 |  96.85% |     63601 | 4.27M | 116 |        92 | 24.72K | 231 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:49 |   0:00:33 |
| MRX80P001 |   80.0 |  96.83% |     73609 | 4.28M | 111 |        82 |  20.9K | 225 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:48 |   0:00:34 |
| MRX80P002 |   80.0 |  96.87% |     64055 | 4.45M | 118 |        94 | 25.27K | 235 |   80.0 | 4.0 |  22.7 | 138.0 | "31,41,51,61,71,81" |   0:00:49 |   0:00:33 |


Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|----:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  97.16% |     73544 | 4.44M | 114 |     37166 | 961.16K | 163 |  263.0 | 13.0 |  74.7 | 453.0 |   0:00:50 |
| 7_merge_mr_unitigs_bcalm      |  98.93% |     73598 | 4.45M | 116 |     34047 | 165.81K |  42 |  262.0 | 13.0 |  74.3 | 451.5 |   0:01:39 |
| 7_merge_mr_unitigs_superreads |  98.93% |     73598 | 4.44M | 116 |     32348 | 143.09K |  35 |  262.0 | 13.0 |  74.3 | 451.5 |   0:01:39 |
| 7_merge_mr_unitigs_tadpole    |  98.89% |     73598 | 4.44M | 114 |     32348 | 145.93K |  37 |  262.0 | 13.0 |  74.3 | 451.5 |   0:01:34 |
| 7_merge_unitigs_bcalm         |  98.99% |     73613 | 4.45M | 116 |     29553 | 502.25K | 147 |  262.0 | 12.0 |  75.3 | 447.0 |   0:01:44 |
| 7_merge_unitigs_superreads    |  99.00% |     73618 | 4.45M | 115 |      5564 |  168.3K |  67 |  264.0 | 14.0 |  74.0 | 459.0 |   0:01:44 |
| 7_merge_unitigs_tadpole       |  99.03% |     73617 | 4.45M | 115 |     37793 | 435.14K |  82 |  263.0 | 13.0 |  74.7 | 453.0 |   0:01:46 |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.45% |    112304 | 4.29M |  84 |      1036 | 24.41K | 157 |  263.0 | 12.0 |  75.7 | 448.5 |   0:00:44 |
| 8_mr_spades  |  98.95% |    125905 | 4.16M |  70 |       758 | 24.86K | 124 |  376.0 | 18.0 | 107.3 | 645.0 |   0:00:50 |
| 8_megahit    |  98.44% |     73656 | 4.46M | 118 |      1039 | 30.37K | 229 |  263.0 | 14.0 |  73.7 | 457.5 |   0:00:44 |
| 8_mr_megahit |  99.17% |    126044 | 4.31M |  78 |       564 | 32.18K | 150 |  376.0 | 17.0 | 108.3 | 640.5 |   0:00:48 |
| 8_platanus   |  97.80% |    114596 | 4.47M |  80 |       421 | 18.53K | 145 |  263.0 | 12.0 |  75.7 | 448.5 |   0:00:44 |


Table: statFinal

| Name                     |     N50 |     Sum |   # |
|:-------------------------|--------:|--------:|----:|
| Genome                   | 4583637 | 4583637 |   1 |
| Paralogs                 |    1737 |  188158 | 111 |
| 7_merge_anchors.anchors  |   73544 | 4440345 | 114 |
| 7_merge_anchors.others   |   37166 |  961155 | 163 |
| glue_anchors             |   73544 | 4440345 | 114 |
| fill_anchors             |  114510 | 4454209 |  79 |
| spades.contig            |  125538 | 4514902 | 165 |
| spades.scaffold          |  143522 | 4515652 | 155 |
| spades.non-contained     |  132337 | 4493720 |  79 |
| mr_spades.contig         |  155808 | 4523706 |  88 |
| mr_spades.scaffold       |  203812 | 4524016 |  84 |
| mr_spades.non-contained  |  173119 | 4513909 |  60 |
| megahit.contig           |   82955 | 4508257 | 167 |
| megahit.non-contained    |   82955 | 4488231 | 116 |
| mr_megahit.contig        |  133730 | 4566779 | 185 |
| mr_megahit.non-contained |  133730 | 4520443 |  73 |
| platanus.contig          |  108571 | 4558622 | 408 |
| platanus.scaffold        |  125790 | 4514164 | 168 |
| platanus.non-contained   |  132296 | 4492200 |  65 |

