# Assemble genomes of model organisms by Anchr

[TOC levels=1-3]: # ""

- [Assemble genomes of model organisms by Anchr](#assemble-genomes-of-model-organisms-by-anchr)
- [More tools on downloading and preprocessing data](#more-tools-on-downloading-and-preprocessing-data)
  - [Extra external executables](#extra-external-executables)
  - [Other leading assemblers](#other-leading-assemblers)
- [*Escherichia* virus Lambda](#escherichia-virus-lambda)
  - [lambda: reference](#lambda-reference)
  - [lambda: download](#lambda-download)
  - [lambda: template](#lambda-template)
  - [lambda: run](#lambda-run)
- [*Escherichia coli* str. K-12 substr. MG1655](#escherichia-coli-str-k-12-substr-mg1655)
  - [mg1655: reference](#mg1655-reference)
  - [mg1655: download](#mg1655-download)
  - [mg1655: template](#mg1655-template)
  - [mg1655: run](#mg1655-run)
- [*Escherichia coli* str. K-12 substr. DH5alpha](#escherichia-coli-str-k-12-substr-dh5alpha)
  - [dh5alpha: reference](#dh5alpha-reference)
  - [dh5alpha: download](#dh5alpha-download)
  - [dh5alpha: template](#dh5alpha-template)
  - [dh5alpha: run](#dh5alpha-run)


# More tools on downloading and preprocessing data

## Extra external executables

```shell script
brew install aria2 curl         # downloading tools
brew install miller

brew tap brewsci/bio
brew tap brewsci/science

brew install --HEAD quast       # assembly quality assessment. https://github.com/ablab/quast/issues/140
quast --test                    # may recompile the bundled nucmer

brew install --ignore-dependencies picard-tools

#kat comp -t 4 -n -o R 2_illumina/R1.fq.gz 2_illumina/R2.fq.gz -m 51
#kat plot spectra-mx -i -o R-spectra R-main.mx

```

## Other leading assemblers

```shell script
brew install spades
brew install megahit
brew install wang-q/tap/platanus

```

# *Escherichia* virus Lambda

Runs under macOS and Linux.

## lambda: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/lambda/1_genome
cd ~/data/anchr/lambda/1_genome

cp ~/data/anchr/ref/lambda/genome.fa .

touch paralogs.fa

```

## lambda: download

```shell script
mkdir -p ~/data/anchr/lambda/ena
cd ~/data/anchr/lambda/ena

cat << EOF > source.csv
SRX2365802,Lambda,HiSeq 2500
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -x 9 -s 3 -c -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

# sampling reads as test materials
seqtk sample -s 23 SRR5042715_1.fastq.gz 100000 | pigz > R1.fq.gz
seqtk sample -s 23 SRR5042715_2.fastq.gz 100000 | pigz > R2.fq.gz

```

| name   | srx        | platform | layout | ilength | srr        | spot     | base  |
|:-------|:-----------|:---------|:-------|:--------|:-----------|:---------|:------|
| Lambda | SRX2365802 | ILLUMINA | PAIRED |         | SRR5042715 | 16540237 | 3.33G |

* Illumina

```shell script
cd ~/data/anchr/lambda

mkdir -p 2_illumina
cd 2_illumina

seqtk sample -s 23 ../ena/SRR5042715_1.fastq.gz 100000 | pigz > R1.fq.gz
seqtk sample -s 23 ../ena/SRR5042715_2.fastq.gz 100000 | pigz > R2.fq.gz

```

## lambda: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=lambda

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 48502 \
    --parallel 4 \
    --xmx 6g \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --cutoff 30 --cutk 31" \
    --qual "20 25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "bcalm tadpole" \
    --statp 2 \
    --readl 108 \
    --uscale 2 \
    --lscale 3

```

## lambda: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=lambda

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 

bash 0_master.sh
#bash 0_cleanup.sh

```

Table: statInsertSize

| Group             |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|------:|-------------------------------:|
| R.genome.bbtools  | 430.6 |    413 | 367.3 |                         96.29% |
| R.tadpole.bbtools | 418.3 |    407 | 101.3 |                         81.92% |
| R.genome.picard   | 424.6 |  412.5 | 103.1 |                             FR |
| R.tadpole.picard  | 418.3 |    407 | 101.2 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 334       | 46251           | 0.0000       | 51.29   |
| R.31 | 297       | 46451           | 0.0000       | 50.64   |
| R.41 | 260       | 46286           | 0.0000       | 50.53   |
| R.51 | 217       | 46045           | 0.0000       | 50.65   |
| R.61 | 180       | 46218           | 0.0000       | 50.77   |
| R.71 | 143       | 46402           | 0.0000       | 50.77   |
| R.81 | 104       | 46385           | 0.0000       | 50.80   |


Table: statReads

| Name       |   N50 |    Sum |      # |
|:-----------|------:|-------:|-------:|
| Genome     | 48502 |  48502 |      1 |
| Paralogs   |     0 |      0 |      0 |
| Illumina.R |   108 |  21.6M | 200000 |
| trim.R     |   105 | 18.82M | 181790 |
| Q20L60     |   105 |  18.6M | 179679 |
| Q25L60     |   105 | 17.94M | 175041 |
| Q30L60     |   105 | 15.99M | 161943 |


Table: statTrimReads

| Name     | N50 |    Sum |      # |
|:---------|----:|-------:|-------:|
| clumpify | 108 | 21.56M | 199606 |
| highpass | 108 |  21.5M | 199066 |
| trim     | 105 | 18.82M | 181790 |
| filter   | 105 | 18.82M | 181790 |
| R1       | 105 |  9.48M |  90895 |
| R2       | 105 |  9.34M |  90895 |
| Rs       |   0 |      0 |      0 |


```text
#R.trim
#Matched	3805	1.91143%
#Name	Reads	ReadsPct
Reverse_adapter	1878	0.94341%
TruSeq_Universal_Adapter	1792	0.90020%
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	235058
#error_kmers	185792
#genomic_kmers	49266
#main_peak	269
#genome_size_in_peaks	2577144
#genome_size	2634418
#haploid_genome_size	48785
#fold_coverage	5
#haploid_fold_coverage	269
#ploidy	54
#het_rate	0.00080
#percent_repeat_in_peaks	0.000
#percent_repeat	98.128
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |    Sum |      # |
|:--------------|----:|-------:|-------:|
| clumped       | 105 | 18.82M | 181786 |
| ecco          | 105 | 18.82M | 181786 |
| eccc          | 105 | 18.82M | 181786 |
| ecct          | 105 | 18.77M | 181266 |
| extended      | 145 | 25.97M | 181266 |
| merged.raw    | 389 | 15.01M |  40051 |
| unmerged.raw  | 145 | 14.42M | 101164 |
| unmerged.trim | 145 | 14.42M | 101156 |
| M1            | 389 | 15.01M |  40029 |
| U1            | 145 |  7.27M |  50578 |
| U2            | 145 |  7.14M |  50578 |
| Us            |   0 |      0 |      0 |
| M.cor         | 267 | 29.46M | 181214 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 163.4 |    175 |  31.9 |          0.44% |
| M.ihist.merge.txt  | 374.9 |    383 |  47.4 |         44.19% |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.22% |     46140 | 47.14K |  2 |        77 |   640 | 12 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q0L0X40P001   |   40.0 |  97.33% |     35666 | 48.08K |  2 |        85 |   373 |  9 |   39.0 | 4.0 |  10.3 |  94.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q0L0X40P002   |   40.0 |  97.22% |     35581 | 48.03K |  2 |        71 |   447 |  8 |   38.0 | 2.0 |  11.3 |  84.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q0L0X80P000   |   80.0 |  94.67% |     35676 | 46.23K |  2 |      1144 | 1.36K |  7 |   78.0 | 4.0 |  23.3 | 172.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:12 |
| Q0L0X80P001   |   80.0 |  96.11% |     16516 | 47.94K |  3 |        92 |   496 | 12 |   76.0 | 4.5 |  22.3 | 170.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:12 |
| Q0L0X80P002   |   80.0 |  97.77% |     35621 | 47.09K |  2 |      1109 | 1.68K | 11 |   79.0 | 6.0 |  22.3 | 182.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q20L60X40P000 |   40.0 |  97.46% |     35651 | 48.09K |  2 |        83 |   295 |  6 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q20L60X40P001 |   40.0 |  97.55% |     35666 | 48.41K |  3 |        54 |   289 | 10 |   39.0 | 5.0 |   9.7 |  98.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q20L60X40P002 |   40.0 |  95.32% |     35611 | 48.01K |  3 |        83 |   278 |  5 |   38.0 | 2.0 |  11.3 |  84.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q20L60X80P000 |   80.0 |  98.23% |     35649 | 48.15K |  3 |        69 |   510 | 11 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q20L60X80P001 |   80.0 |  95.63% |     30610 | 47.08K |  4 |       210 |   405 |  9 |   78.0 | 8.0 |  20.7 | 188.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:12 |
| Q20L60X80P002 |   80.0 |  94.21% |     16727 | 47.45K |  5 |        68 |   264 | 11 |   76.0 | 8.5 |  19.7 | 186.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:12 |
| Q25L60X40P000 |   40.0 |  96.53% |     48039 | 48.04K |  1 |       256 |   328 |  2 |   38.0 | 0.0 |  12.7 |  76.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q25L60X40P001 |   40.0 |  98.02% |     30602 | 48.15K |  4 |        69 |   575 | 19 |   39.0 | 3.0 |  11.0 |  90.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q25L60X40P002 |   40.0 |  96.48% |     35672 | 48.17K |  2 |       170 |   324 |  6 |   39.0 | 5.0 |   9.7 |  98.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q25L60X80P000 |   80.0 |  97.52% |     35676 | 48.17K |  3 |       101 |   343 | 10 |   78.0 | 7.0 |  21.3 | 184.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q25L60X80P001 |   80.0 |  98.01% |     35650 | 45.37K |  3 |       876 | 2.61K | 10 |   80.0 | 0.0 |  26.7 | 160.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q25L60X80P002 |   80.0 |  96.53% |     48004 |    48K |  1 |       250 |   387 |  2 |   78.0 | 1.0 |  25.3 | 160.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q30L60X40P000 |   40.0 |  97.33% |     26354 | 47.96K |  4 |         0 |     0 |  0 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q30L60X40P001 |   40.0 |  97.14% |     35395 | 47.91K |  3 |       102 |   425 |  4 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q30L60X40P002 |   40.0 |  97.14% |     30570 |  47.7K |  3 |       286 |   519 |  4 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q30L60X80P000 |   80.0 |  98.63% |     26331 |  47.9K |  4 |       142 |   265 |  4 |   78.0 | 0.5 |  25.7 | 158.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q30L60X80P001 |   80.0 |  98.83% |     35642 | 48.07K |  3 |       101 |   589 |  9 |   80.0 | 2.0 |  25.3 | 168.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q30L60X80P002 |   80.0 |  98.12% |     35606 | 47.98K |  3 |       244 |   666 |  7 |   80.0 | 5.0 |  23.3 | 180.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  99.81% |     48187 | 48.19K |  1 |       185 | 327 |  2 |   40.0 | 1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| MRX40P001 |   40.0 |  99.59% |     35435 | 48.02K |  2 |       158 | 368 |  3 |   40.0 | 1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| MRX40P002 |   40.0 |  99.61% |     47973 | 47.97K |  1 |       248 | 415 |  2 |   40.0 | 1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| MRX80P000 |   80.0 |  99.66% |     35817 | 48.25K |  2 |       236 | 342 |  3 |   81.0 | 6.0 |  23.0 | 186.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:10 |
| MRX80P001 |   80.0 |  99.22% |     14224 | 48.03K |  4 |       205 | 724 |  7 |   80.0 | 6.0 |  22.7 | 184.0 | "31,41,51,61,71,81" |   0:00:08 |   0:00:10 |
| MRX80P002 |   80.0 |  99.61% |     35808 | 48.29K |  2 |       184 | 304 |  4 |   79.0 | 5.5 |  22.7 | 180.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:10 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  96.45% |     46156 | 47.16K |  2 |       125 |   574 |  8 |   38.0 | 1.0 |  12.0 |  80.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q0L0X40P001   |   40.0 |  95.59% |     35675 | 48.09K |  2 |        91 |   336 |  7 |   38.0 | 3.5 |  10.3 |  90.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q0L0X40P002   |   40.0 |  95.03% |     45317 | 48.01K |  2 |       267 |   373 |  3 |   38.0 | 0.0 |  12.7 |  76.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q0L0X80P000   |   80.0 |  97.97% |     35670 | 46.22K |  2 |      1236 | 1.97K | 10 |   78.0 | 0.0 |  26.0 | 156.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q0L0X80P001   |   80.0 |  95.65% |     35668 | 48.12K |  2 |       201 |   556 |  5 |   76.0 | 7.0 |  20.7 | 180.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q0L0X80P002   |   80.0 |  95.92% |     47095 |  47.1K |  1 |      1114 | 1.37K |  5 |   75.0 | 7.0 |  20.3 | 178.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q20L60X40P000 |   40.0 |  97.56% |     35651 |  48.1K |  2 |        88 |   455 |  6 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q20L60X40P002 |   40.0 |  95.48% |     35611 | 48.01K |  3 |       225 |   399 |  4 |   38.0 | 2.0 |  11.3 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q20L60X80P000 |   80.0 |  98.27% |     35632 | 48.13K |  3 |       107 |   745 | 10 |   79.0 | 0.0 |  26.3 | 158.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q20L60X80P001 |   80.0 |  97.14% |     35404 | 47.82K |  2 |       126 |   549 |  6 |   78.0 | 2.0 |  24.7 | 164.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q20L60X80P002 |   80.0 |  95.70% |     16728 | 48.26K |  4 |       174 |   561 | 10 |   76.0 | 8.0 |  20.0 | 184.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X40P000 |   40.0 |  96.54% |     48050 | 48.05K |  1 |       256 |   334 |  2 |   38.0 | 0.0 |  12.7 |  76.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X40P001 |   40.0 |  96.40% |     30608 | 48.18K |  4 |       114 |   213 |  9 |   38.0 | 4.0 |  10.0 |  92.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q25L60X40P002 |   40.0 |  96.39% |     37524 | 48.08K |  2 |       174 |   307 |  3 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q25L60X80P000 |   80.0 |  97.97% |     35674 | 48.17K |  3 |       210 |   602 |  8 |   78.0 | 5.5 |  22.3 | 178.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X80P001 |   80.0 |  98.59% |     35678 | 49.05K |  3 |      1114 | 1.65K | 10 |   79.0 | 5.5 |  22.7 | 180.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X80P002 |   80.0 |  96.63% |     48004 |    48K |  1 |       260 |   510 |  2 |   77.0 | 0.0 |  25.7 | 154.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q30L60X40P000 |   40.0 |  97.47% |     26354 | 47.96K |  4 |         0 |     0 |  0 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q30L60X40P001 |   40.0 |  97.16% |     35395 | 47.91K |  3 |       188 |   519 |  4 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| Q30L60X40P002 |   40.0 |  97.16% |     30570 |  47.7K |  3 |       157 |   583 |  4 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q30L60X80P000 |   80.0 |  97.41% |     26331 |  47.9K |  4 |         0 |     0 |  0 |   78.0 | 1.0 |  25.3 | 160.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q30L60X80P001 |   80.0 |  97.32% |     37389 | 48.05K |  2 |       230 |   345 |  3 |   78.0 | 1.0 |  25.3 | 160.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:11 |
| Q30L60X80P002 |   80.0 |  98.00% |     48079 | 48.08K |  1 |       262 |   440 |  4 |   75.0 | 6.0 |  21.0 | 174.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  99.81% |     48187 | 48.19K |  1 |       185 | 327 |  2 |   40.0 |  1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| MRX40P001 |   40.0 |  99.60% |     35435 | 48.02K |  2 |       158 | 386 |  3 |   40.0 |  1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| MRX40P002 |   40.0 |  99.72% |     47989 | 47.99K |  1 |       277 | 525 |  2 |   40.0 |  1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| MRX80P000 |   80.0 |  99.66% |     35817 | 48.25K |  2 |       236 | 342 |  3 |   81.0 |  6.0 |  23.0 | 186.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| MRX80P001 |   80.0 |  99.53% |     47276 | 48.32K |  2 |       182 | 270 |  4 |   81.0 | 15.0 |  17.0 | 222.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| MRX80P002 |   80.0 |  99.61% |     35809 |  48.3K |  2 |       184 | 298 |  3 |   79.0 |  5.5 |  22.7 | 180.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:11 |


Table: statMergeAnchors.md

| Name                       | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median | MAD | lower | upper | RunTimeAN |
|:---------------------------|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|----:|------:|------:|----------:|
| 7_merge_anchors            |   0.00% |     48372 | 48.37K |  1 |      1236 | 1.24K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_mr_unitigs_bcalm   |   0.00% |     48364 | 48.36K |  1 |      1236 | 1.24K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_mr_unitigs_tadpole |   0.00% |     48372 | 48.37K |  1 |         0 |     0 |  0 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_unitigs_bcalm      |   0.00% |     48180 | 48.18K |  1 |      1144 | 1.14K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_unitigs_tadpole    |   0.00% |     48196 |  48.2K |  1 |      1236 | 1.24K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median | MAD | lower |  upper | RunTimeAN |
|:-------------|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|----:|------:|-------:|----------:|
| 8_mr_spades  |  99.57% |     26405 | 48.13K |  2 |       429 | 494 |  3 |  603.0 | 1.0 | 200.3 | 1210.0 |   0:00:11 |
| 8_megahit    |  95.15% |     48024 | 48.02K |  1 |       262 | 490 |  2 |  370.0 | 1.0 | 122.7 |  744.0 |   0:00:12 |
| 8_mr_megahit |  99.74% |     48072 | 48.07K |  1 |       242 | 442 |  2 |  605.0 | 0.0 | 201.7 | 1210.0 |   0:00:11 |


Table: statFinal

| Name                     |   N50 |   Sum |  # |
|:-------------------------|------:|------:|---:|
| Genome                   | 48502 | 48502 |  1 |
| Paralogs                 |     0 |     0 |  0 |
| 7_merge_anchors.anchors  | 48372 | 48372 |  1 |
| 7_merge_anchors.others   |  1236 |  1236 |  1 |
| spades.non-contained     |     0 |     0 |  0 |
| mr_spades.contig         | 48627 | 48753 |  2 |
| mr_spades.scaffold       | 48627 | 48753 |  2 |
| mr_spades.non-contained  | 48627 | 48627 |  1 |
| megahit.contig           | 48514 | 48514 |  1 |
| megahit.non-contained    | 48514 | 48514 |  1 |
| mr_megahit.contig        | 48514 | 48514 |  1 |
| mr_megahit.non-contained | 48514 | 48514 |  1 |
| platanus.non-contained   |     0 |     0 |  0 |


# *Escherichia coli* str. K-12 substr. MG1655

## mg1655: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/mg1655/1_genome
cd ~/data/anchr/mg1655/1_genome

cp ~/data/anchr/ref/mg1655/genome.fa .
cp ~/data/anchr/ref/mg1655/paralogs.fa .
cp ~/data/anchr/ref/mg1655/repetitives.fa .

```

## mg1655: download

* Illumina

```shell script
cd ~/data/anchr/mg1655

mkdir -p 2_illumina
cd 2_illumina

aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz
aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz

ln -s MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz R1.fq.gz
ln -s MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz R2.fq.gz

```

## mg1655: template

* Rsync to hpcc

```shell script
rsync -avP \
    ~/data/anchr/mg1655/ \
    wangq@202.119.37.251:data/anchr/mg1655

# rsync -avP wangq@202.119.37.251:data/anchr/mg1655/ ~/data/anchr/mg1655

```

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4641652 \
    --parallel 24 \
    --xmx 80g \
    --queue mpi \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --tile --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --readl 151 \
    --uscale 2 \
    --lscale 3 \
    --redo \
    \
    --extend

```

## mg1655: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 
# rm -fr 4_down_sampling 6_down_sampling

# BASE_NAME=mg1655 bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
# bkill -J "${BASE_NAME}-*"

# bash 0_master.sh
# bash 0_cleanup.sh

```

Table: statInsertSize

| Group             |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|------:|-------------------------------:|
| R.genome.bbtools  | 321.9 |    298 | 967.3 |                         96.08% |
| R.tadpole.bbtools | 294.7 |    296 |  22.0 |                         81.33% |
| R.genome.picard   | 298.2 |    298 |  18.0 |                             FR |
| R.tadpole.picard  | 294.9 |    296 |  21.6 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 292       | 4390759         | 0.1630       | 54.20   |
| R.31 | 265       | 4388871         | 0.1382       | 54.04   |
| R.41 | 239       | 4375332         | 0.1108       | 53.90   |
| R.51 | 216       | 4407280         | 0.1088       | 53.78   |
| R.61 | 191       | 4393867         | 0.1081       | 53.68   |
| R.71 | 168       | 4390152         | 0.1042       | 53.58   |
| R.81 | 144       | 4387393         | 0.0989       | 53.50   |


Table: statReads

| Name       |     N50 |     Sum |        # |
|:-----------|--------:|--------:|---------:|
| Genome     | 4641652 | 4641652 |        1 |
| Paralogs   |    1937 |  187300 |      106 |
| Illumina.R |     151 |   1.73G | 11458940 |
| trim.R     |     149 |   1.43G | 10361742 |
| Q25L60     |     148 |   1.32G |  9937758 |
| Q30L60     |     128 |   1.11G |  9310071 |


Table: statTrimReads

| Name           | N50 |     Sum |        # |
|:---------------|----:|--------:|---------:|
| clumpify       | 151 |   1.73G | 11439000 |
| filteredbytile | 151 |   1.67G | 11058426 |
| highpass       | 151 |   1.66G | 10985728 |
| trim           | 149 |   1.43G | 10361742 |
| filter         | 149 |   1.43G | 10361742 |
| R1             | 150 | 735.87M |  5180871 |
| R2             | 144 |  690.3M |  5180871 |
| Rs             |   0 |       0 |        0 |


```text
#R.trim
#Matched	17482	0.15913%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	20703730
#error_kmers	16178962
#genomic_kmers	4524768
#main_peak	245
#genome_size_in_peaks	4593892
#genome_size	4627576
#haploid_genome_size	4627576
#fold_coverage	245
#haploid_fold_coverage	245
#ploidy	1
#percent_repeat_in_peaks	1.507
#percent_repeat	1.761
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |    Sum |        # |
|:--------------|----:|-------:|---------:|
| clumped       | 149 |  1.43G | 10360570 |
| ecco          | 149 |  1.43G | 10360570 |
| eccc          | 149 |  1.43G | 10360570 |
| ecct          | 149 |  1.42G | 10312020 |
| extended      | 189 |  1.83G | 10312020 |
| merged.raw    | 339 |  1.72G |  5084228 |
| unmerged.raw  | 175 | 21.13M |   143564 |
| unmerged.trim | 175 | 21.12M |   143506 |
| M1            | 339 |  1.71G |  5056768 |
| U1            | 181 | 11.08M |    71753 |
| U2            | 169 | 10.04M |    71753 |
| Us            |   0 |      0 |        0 |
| M.cor         | 338 |  1.73G | 10257042 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 271.8 |    277 |  23.4 |         10.84% |
| M.ihist.merge.txt  | 337.7 |    338 |  19.3 |         98.61% |


Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 307.3 |  285.6 |    7.05% | "67" | 4.64M | 4.66M |     1.00 | 0:02'38'' |
| Q25L60.R | 284.0 |  273.1 |    3.84% | "61" | 4.64M | 4.57M |     0.98 | 0:02'29'' |
| Q30L60.R | 238.3 |  233.5 |    2.03% | "51" | 4.64M | 4.55M |     0.98 | 0:02'10'' |


Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |    # | N50Others |    Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|-----:|----------:|-------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  96.09% |      8851 | 4.47M |  729 |       271 | 69.93K | 1763 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:01:02 |   0:00:49 |
| Q0L0X40P001   |   40.0 |  96.12% |      8365 | 4.47M |  792 |       519 |  84.2K | 1858 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:49 |
| Q0L0X40P002   |   40.0 |  95.84% |      9226 | 4.46M |  722 |       317 | 75.73K | 1756 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:40 |
| Q0L0X80P000   |   80.0 |  92.25% |      4781 | 4.34M | 1189 |       794 | 78.86K | 2533 |   79.0 | 6.0 |  22.3 | 182.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:41 |
| Q0L0X80P001   |   80.0 |  92.24% |      4991 | 4.35M | 1161 |       174 | 59.63K | 2471 |   79.0 | 6.0 |  22.3 | 182.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:42 |
| Q0L0X80P002   |   80.0 |  92.00% |      4755 | 4.33M | 1172 |       550 |  70.5K | 2476 |   79.0 | 6.0 |  22.3 | 182.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:38 |
| Q25L60X40P000 |   40.0 |  97.75% |     13614 | 4.48M |  528 |       308 | 67.93K | 1481 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:43 |
| Q25L60X40P001 |   40.0 |  97.83% |     13848 | 4.49M |  515 |       525 | 81.14K | 1551 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:45 |
| Q25L60X40P002 |   40.0 |  97.68% |     14367 | 4.47M |  505 |       560 | 78.68K | 1455 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:43 |
| Q25L60X80P000 |   80.0 |  96.50% |      9097 | 4.48M |  709 |       274 | 53.95K | 1653 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:40 |
| Q25L60X80P001 |   80.0 |  96.64% |      9695 | 4.48M |  707 |       518 | 64.53K | 1669 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:41 |
| Q25L60X80P002 |   80.0 |  96.66% |     10103 | 4.48M |  681 |       411 | 59.68K | 1637 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:40 |
| Q30L60X40P000 |   40.0 |  98.54% |     15594 | 4.18M |  429 |       711 | 95.87K | 1492 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:47 |
| Q30L60X40P001 |   40.0 |  98.54% |     15816 | 4.29M |  429 |       617 | 81.42K | 1510 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:48 |
| Q30L60X40P002 |   40.0 |  98.52% |     15734 | 4.23M |  445 |       780 | 93.49K | 1480 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:45 |
| Q30L60X80P000 |   80.0 |  98.52% |     18236 | 4.34M |  389 |       688 | 68.37K | 1248 |   80.0 | 5.0 |  23.3 | 180.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:52 |
| Q30L60X80P001 |   80.0 |  98.49% |     18534 | 4.36M |  394 |       762 | 70.05K | 1257 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:08 |   0:00:50 |


Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.01% |     28453 | 4.49M | 285 |       128 | 59.53K | 614 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:01:22 |   0:00:47 |
| MRX40P001 |   40.0 |  96.12% |     25559 | 4.48M | 290 |       134 | 62.82K | 616 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:54 |   0:00:47 |
| MRX40P002 |   40.0 |  95.99% |     26988 | 4.48M | 298 |       137 | 69.11K | 634 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:54 |   0:00:35 |
| MRX80P000 |   80.0 |  94.93% |     16127 | 4.45M | 432 |       133 |  89.1K | 900 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:36 |
| MRX80P001 |   80.0 |  94.68% |     16133 | 4.44M | 447 |       134 | 98.62K | 936 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:32 |   0:00:37 |
| MRX80P002 |   80.0 |  94.97% |     15775 | 4.45M | 435 |       135 | 91.94K | 912 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:37 |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|--------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.94% |     15626 | 4.52M | 443 |       120 |   67.9K | 1814 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:59 |
| Q0L0X40P001   |   40.0 |  97.90% |     15256 | 4.48M | 469 |        86 |  62.29K | 1907 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:15 |   0:00:58 |
| Q0L0X40P002   |   40.0 |  97.99% |     16454 |  4.5M | 437 |       638 |  93.31K | 1817 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:48 |
| Q0L0X80P000   |   80.0 |  95.94% |      8445 | 4.49M | 762 |       140 |  55.11K | 1842 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:38 |
| Q0L0X80P001   |   80.0 |  96.08% |      9630 | 4.48M | 706 |        92 |  42.25K | 1779 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:42 |
| Q0L0X80P002   |   80.0 |  96.04% |      8999 | 4.48M | 717 |       171 |  52.85K | 1783 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:40 |
| Q25L60X40P000 |   40.0 |  98.42% |     16211 | 4.52M | 423 |       129 |  67.25K | 1970 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:01:17 |   0:00:51 |
| Q25L60X40P001 |   40.0 |  98.35% |     17362 | 4.51M | 414 |       336 |   74.5K | 1844 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:48 |
| Q25L60X40P002 |   40.0 |  98.43% |     19482 | 4.51M | 390 |       607 |  77.66K | 1792 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:50 |
| Q25L60X80P000 |   80.0 |  97.67% |     13904 | 4.51M | 510 |        88 |  43.02K | 1603 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:46 |
| Q25L60X80P001 |   80.0 |  97.66% |     13865 | 4.51M | 509 |       215 |  53.91K | 1599 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:45 |
| Q25L60X80P002 |   80.0 |  97.84% |     14381 | 4.51M | 509 |       345 |  54.42K | 1595 |   79.0 | 5.0 |  23.0 | 178.0 | "31,41,51,61,71,81" |   0:01:34 |   0:00:46 |
| Q30L60X40P000 |   40.0 |  98.40% |     14806 | 4.47M | 489 |       802 | 101.32K | 1984 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:01:16 |   0:00:47 |
| Q30L60X40P001 |   40.0 |  98.41% |     14858 | 4.42M | 472 |       689 |  88.85K | 2053 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:01:15 |   0:00:49 |
| Q30L60X40P002 |   40.0 |  98.40% |     15252 | 4.47M | 487 |       637 |  84.23K | 2039 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:48 |
| Q30L60X80P000 |   80.0 |  98.55% |     18505 |  4.4M | 384 |       586 |  62.06K | 1570 |   80.0 | 5.0 |  23.3 | 180.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:52 |
| Q30L60X80P001 |   80.0 |  98.53% |     18588 | 4.45M | 394 |       736 |  70.66K | 1594 |   80.0 | 5.0 |  23.3 | 180.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:52 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.21% |     41113 |  4.5M | 192 |       123 | 38.23K | 396 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:01:36 |   0:00:41 |
| MRX40P001 |   40.0 |  96.45% |     42177 |  4.5M | 193 |       132 | 40.37K | 390 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:01:16 |   0:00:41 |
| MRX40P002 |   40.0 |  96.34% |     43061 |  4.5M | 198 |       134 | 43.42K | 408 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:01:17 |   0:00:34 |
| MRX80P000 |   80.0 |  95.53% |     26056 | 4.48M | 302 |       131 | 60.59K | 616 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:36 |
| MRX80P001 |   80.0 |  95.38% |     24052 | 4.47M | 318 |       138 |  69.8K | 642 |   79.0 | 3.0 |  24.3 | 170.0 | "31,41,51,61,71,81" |   0:01:32 |   0:00:37 |
| MRX80P002 |   80.0 |  95.60% |     21507 | 4.48M | 317 |       131 | 62.36K | 644 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:37 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.16% |     24563 | 4.52M | 331 |       126 | 55.75K | 1254 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:48 |
| Q0L0X40P001   |   40.0 |  98.28% |     22741 | 4.52M | 344 |       121 | 53.69K | 1318 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:46 |
| Q0L0X40P002   |   40.0 |  98.24% |     21595 | 4.52M | 333 |       232 | 58.65K | 1315 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:49 |
| Q0L0X80P000   |   80.0 |  97.21% |     14852 | 4.51M | 472 |       414 | 49.17K | 1219 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:40 |
| Q0L0X80P001   |   80.0 |  97.28% |     14642 | 4.51M | 464 |       112 | 38.72K | 1227 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:43 |
| Q0L0X80P002   |   80.0 |  97.17% |     15372 | 4.51M | 461 |       301 | 44.93K | 1206 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:43 |
| Q25L60X40P000 |   40.0 |  98.43% |     19463 |  4.5M | 354 |       248 | 60.17K | 1411 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:46 |
| Q25L60X40P001 |   40.0 |  98.53% |     21175 | 4.51M | 355 |       270 | 62.96K | 1450 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:47 |
| Q25L60X40P002 |   40.0 |  98.49% |     22148 | 4.43M | 330 |       539 | 65.21K | 1329 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:50 |
| Q25L60X80P000 |   80.0 |  98.11% |     19461 | 4.51M | 373 |       145 | 46.35K | 1237 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:50 |
| Q25L60X80P001 |   80.0 |  98.07% |     19127 | 4.51M | 396 |       605 | 60.06K | 1216 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:47 |
| Q25L60X80P002 |   80.0 |  98.24% |     19022 | 4.47M | 384 |       518 | 58.97K | 1230 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:47 |   0:00:50 |
| Q30L60X40P000 |   40.0 |  98.52% |     15159 | 4.43M | 450 |       751 | 97.89K | 1701 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:48 |
| Q30L60X40P001 |   40.0 |  98.51% |     15923 | 4.34M | 438 |       702 | 88.21K | 1735 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:47 |
| Q30L60X40P002 |   40.0 |  98.46% |     16279 | 4.39M | 444 |       751 | 83.27K | 1631 |   40.0 | 3.0 |  11.3 |  92.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:46 |
| Q30L60X80P000 |   80.0 |  98.59% |     19641 | 4.31M | 365 |       591 | 62.37K | 1359 |   80.0 | 5.0 |  23.3 | 180.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:52 |
| Q30L60X80P001 |   80.0 |  98.51% |     19766 | 4.34M | 367 |       716 | 70.22K | 1333 |   80.0 | 5.0 |  23.3 | 180.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:49 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.54% |     53713 | 4.51M | 156 |       127 | 31.48K | 312 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:01:02 |   0:00:39 |
| MRX40P001 |   40.0 |  96.56% |     50628 | 4.51M | 164 |       123 | 32.13K | 322 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:38 |
| MRX40P002 |   40.0 |  96.50% |     50573 |  4.5M | 163 |       143 | 40.65K | 328 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:36 |
| MRX80P000 |   80.0 |  96.00% |     31635 | 4.49M | 232 |       130 | 46.03K | 474 |   79.0 | 3.0 |  24.3 | 170.0 | "31,41,51,61,71,81" |   0:00:51 |   0:00:37 |
| MRX80P001 |   80.0 |  95.92% |     30691 | 4.48M | 244 |       137 | 55.84K | 496 |   79.0 | 3.0 |  24.3 | 170.0 | "31,41,51,61,71,81" |   0:00:51 |   0:00:35 |
| MRX80P002 |   80.0 |  95.98% |     30780 | 4.49M | 245 |       131 | 48.87K | 499 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:51 |   0:00:35 |


Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|----:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  97.76% |     46108 | 4.54M | 182 |     18216 |  631.2K | 162 |  285.0 | 11.0 |  87.7 | 614.0 |   0:01:01 |
| 7_merge_mr_unitigs_bcalm      |  98.82% |     42195 | 4.52M | 192 |     25031 | 159.25K |  28 |  285.0 | 10.0 |  88.3 | 610.0 |   0:01:56 |
| 7_merge_mr_unitigs_superreads |  98.84% |     41047 | 4.52M | 202 |     18216 |  90.73K |  28 |  284.0 | 10.5 |  87.7 | 610.0 |   0:01:55 |
| 7_merge_mr_unitigs_tadpole    |  98.89% |     44780 | 4.37M | 185 |     13183 | 114.65K |  28 |  285.0 |  9.0 |  89.0 | 606.0 |   0:01:53 |
| 7_merge_unitigs_bcalm         |  99.03% |     45260 | 4.54M | 191 |      6765 | 304.08K | 120 |  284.0 | 12.0 |  86.7 | 616.0 |   0:02:09 |
| 7_merge_unitigs_superreads    |  98.98% |     43554 | 4.53M | 191 |      1578 | 190.39K | 127 |  284.0 | 14.0 |  85.3 | 624.0 |   0:02:03 |
| 7_merge_unitigs_tadpole       |  99.09% |     47231 | 4.53M | 182 |     11558 | 253.25K | 104 |  284.0 | 13.0 |  86.0 | 620.0 |   0:02:16 |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.54% |     54900 | 2.79M | 100 |      1255 |  16.6K | 144 |  285.0 | 10.0 |  88.3 | 610.0 |   0:00:50 |
| 8_mr_spades  |  98.37% |    148489 | 4.54M |  76 |       620 | 25.52K | 148 |  371.0 | 10.0 | 117.0 | 782.0 |   0:00:53 |
| 8_megahit    |  98.17% |     47257 | 3.71M | 160 |      1267 | 19.14K | 220 |  285.0 | 10.0 |  88.3 | 610.0 |   0:00:59 |
| 8_mr_megahit |  98.77% |    125756 | 4.56M |  76 |       804 | 24.66K | 139 |  371.0 |  8.5 | 118.0 | 776.0 |   0:00:52 |
| 8_platanus   |  97.89% |     47791 | 2.12M |  83 |       463 |  8.74K | 117 |  285.0 |  8.5 |  89.3 | 604.0 |   0:00:48 |


Table: statFinal

| Name                     |     N50 |     Sum |    # |
|:-------------------------|--------:|--------:|-----:|
| Genome                   | 4641652 | 4641652 |    1 |
| Paralogs                 |    1937 |  187300 |  106 |
| Repetitives              |    1205 |  110138 |  206 |
| 7_merge_anchors.anchors  |   46108 | 4540428 |  182 |
| 7_merge_anchors.others   |   18216 |  631202 |  162 |
| glue_anchors             |   58834 | 4539197 |  138 |
| fill_anchors             |   95911 | 4542653 |   84 |
| spades.contig            |  132608 | 4573776 |  137 |
| spades.scaffold          |  133063 | 4573960 |  133 |
| spades.non-contained     |  132608 | 4555564 |   75 |
| mr_spades.contig         |  148607 | 4587780 |  149 |
| mr_spades.scaffold       |  148607 | 4587980 |  147 |
| mr_spades.non-contained  |  148607 | 4570445 |   76 |
| megahit.contig           |   82825 | 4568234 |  149 |
| megahit.non-contained    |   82825 | 4551278 |  106 |
| mr_megahit.contig        |  132896 | 4612226 |  134 |
| mr_megahit.non-contained |  132896 | 4586194 |   70 |
| platanus.contig          |   14890 | 4714097 | 1183 |
| platanus.scaffold        |  148483 | 4578070 |  137 |
| platanus.non-contained   |  178251 | 4561075 |   62 |


# *Escherichia coli* str. K-12 substr. DH5alpha

## dh5alpha: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/dh5alpha/1_genome
cd ~/data/anchr/dh5alpha/1_genome

cp ~/data/anchr/ref/dh5alpha/genome.fa .
cp ~/data/anchr/ref/dh5alpha/paralogs.fa .
cp ~/data/anchr/ref/dh5alpha/repetitives.fa .

```

## dh5alpha: download

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p ena
cd ena

cat << EOF > source.csv
SRP251726,dh5alpha,HiSeq 2500 PE125
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -x 9 -s 3 -c -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

```

| name     | srx        | platform        | layout | ilength | srr         | spot    | base  |
|:---------|:-----------|:----------------|:-------|:--------|:------------|:--------|:------|
| dh5alpha | SRX7856678 | ILLUMINA        | PAIRED |         | SRR11245239 | 5881654 | 1.37G |
| dh5alpha | SRX7856679 | OXFORD_NANOPORE | SINGLE |         | SRR11245238 | 346489  | 3.35G |


* Illumina

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p 2_illumina
cd 2_illumina

ln -s ../ena/SRR11245239_1.fastq.gz R1.fq.gz
ln -s ../ena/SRR11245239_2.fastq.gz R2.fq.gz

```

## dh5alpha: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4583637 \
    --parallel 24 \
    --xmx 80g \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --readl 125 \
    --uscale 2 \
    --lscale 3 \
    --redo \
    \
    --extend

```

## dh5alpha: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 

# BASE_NAME=dh5alpha bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
#bkill -J "${BASE_NAME}-*"

#bash 0_master.sh
#bash 0_cleanup.sh
#rm -fr 4_down_sampling 6_down_sampling

```


Table: statInsertSize

| Group             |  Mean | Median |  STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|-------:|-------------------------------:|
| R.genome.bbtools  | 432.7 |    346 | 1241.2 |                         99.98% |
| R.tadpole.bbtools | 389.1 |    341 |  206.1 |                         95.12% |
| R.genome.picard   | 394.8 |    346 |  208.3 |                             FR |
| R.tadpole.picard  | 389.1 |    341 |  205.7 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 260       | 4346013         | 0.0814       | 51.62   |
| R.31 | 233       | 4360998         | 0.0563       | 51.54   |
| R.41 | 209       | 4433088         | 0.0607       | 51.50   |
| R.51 | 182       | 4413726         | 0.0150       | 51.47   |
| R.61 | 157       | 4456634         | 0.0332       | 51.45   |
| R.71 | 132       | 4495949         | 0.0194       | 51.44   |
| R.81 | 107       | 4475448         | 0.0160       | 51.42   |


Table: statReads

| Name       |     N50 |     Sum |        # |
|:-----------|--------:|--------:|---------:|
| Genome     | 4583637 | 4583637 |        1 |
| Paralogs   |    1737 |  188158 |      111 |
| Illumina.R |     125 |   1.47G | 11763308 |
| trim.R     |     125 |   1.37G | 10962374 |
| Q25L60     |     125 |   1.25G | 10281036 |
| Q30L60     |     125 |   1.13G |  9405631 |


Table: statTrimReads

| Name     | N50 |     Sum |        # |
|:---------|----:|--------:|---------:|
| clumpify | 125 |   1.37G | 10970448 |
| highpass | 125 |   1.37G | 10966250 |
| trim     | 125 |   1.37G | 10962374 |
| filter   | 125 |   1.37G | 10962374 |
| R1       | 125 |    683M |  5481187 |
| R2       | 125 | 683.64M |  5481187 |
| Rs       |   0 |       0 |        0 |


```text
#R.trim
#Matched	5620	0.05125%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	35781051
#error_kmers	31303388
#genomic_kmers	4477663
#main_peak	223
#genome_size_in_peaks	4561456
#genome_size	4571195
#haploid_genome_size	4571195
#fold_coverage	223
#haploid_fold_coverage	223
#ploidy	1
#percent_repeat_in_peaks	1.841
#percent_repeat	1.987
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |     Sum |        # |
|:--------------|----:|--------:|---------:|
| clumped       | 125 |   1.37G | 10959556 |
| ecco          | 125 |   1.37G | 10959556 |
| eccc          | 125 |   1.37G | 10959556 |
| ecct          | 125 |   1.37G | 10952902 |
| extended      | 165 |    1.8G | 10952902 |
| merged.raw    | 343 |    1.1G |  3511427 |
| unmerged.raw  | 165 | 645.95M |  3930048 |
| unmerged.trim | 165 | 645.95M |  3930048 |
| M1            | 343 |   1.06G |  3404031 |
| U1            | 165 | 322.88M |  1965024 |
| U2            | 165 | 323.07M |  1965024 |
| Us            |   0 |       0 |        0 |
| M.cor         | 250 |   1.71G | 10738110 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 180.7 |    181 |  32.1 |         23.40% |
| M.ihist.merge.txt  | 312.5 |    310 |  87.7 |         64.12% |


Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 298.2 |  262.2 |   12.07% | "63" | 4.58M |  4.6M |     1.00 | 0:08'26'' |
| Q25L60.R | 273.7 |  253.6 |    7.35% | "63" | 4.58M | 4.53M |     0.99 | 0:06'56'' |
| Q30L60.R | 246.5 |  232.4 |    5.71% | "63" | 4.58M | 4.52M |     0.99 | 0:06'23'' |


Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.16% |     25651 | 4.45M | 310 |       978 | 44.06K |  719 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:47 |
| Q0L0X40P001   |   40.0 |  97.34% |     29858 | 4.44M | 274 |      1054 |  40.5K |  662 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:47 |
| Q0L0X40P002   |   40.0 |  97.15% |     29672 | 4.44M | 262 |      1145 | 35.12K |  612 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:37 |
| Q0L0X80P000   |   80.0 |  96.07% |     12163 | 4.43M | 548 |      1022 |  22.3K | 1170 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:37 |
| Q0L0X80P001   |   80.0 |  95.80% |     12484 | 4.41M | 531 |        93 | 16.55K | 1112 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:37 |
| Q0L0X80P002   |   80.0 |  95.90% |     12511 | 4.44M | 557 |      1043 | 25.05K | 1160 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:37 |
| Q25L60X40P000 |   40.0 |  97.70% |     40766 | 4.45M | 211 |      1022 | 26.79K |  525 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:38 |
| Q25L60X40P001 |   40.0 |  97.69% |     41153 | 4.45M | 207 |       994 | 34.14K |  526 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:38 |
| Q25L60X40P002 |   40.0 |  97.71% |     43188 | 4.45M | 179 |      1221 | 34.53K |  482 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:36 |
| Q25L60X80P000 |   80.0 |  96.96% |     24985 | 4.38M | 320 |      1145 | 20.43K |  723 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:37 |
| Q25L60X80P001 |   80.0 |  97.08% |     25781 | 4.45M | 305 |       527 | 11.66K |  669 |   79.0 | 3.0 |  24.3 | 170.0 | "31,41,51,61,71,81" |   0:01:05 |   0:00:39 |
| Q25L60X80P002 |   80.0 |  97.02% |     25657 | 4.46M | 313 |      1182 | 15.74K |  690 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:37 |
| Q30L60X40P000 |   40.0 |  97.73% |     39169 | 4.45M | 192 |      1182 | 33.19K |  479 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:37 |
| Q30L60X40P001 |   40.0 |  97.84% |     42114 | 4.45M | 189 |      1022 |  22.1K |  476 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:42 |
| Q30L60X40P002 |   40.0 |  97.76% |     41396 | 4.45M | 196 |      1161 | 41.08K |  508 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:38 |
| Q30L60X80P000 |   80.0 |  97.14% |     26361 | 4.46M | 292 |      1345 | 19.78K |  655 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:05 |   0:00:36 |
| Q30L60X80P001 |   80.0 |  97.19% |     26750 | 4.37M | 286 |      1152 |  15.8K |  644 |   79.0 | 3.0 |  24.3 | 170.0 | "31,41,51,61,71,81" |   0:01:05 |   0:00:40 |


Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.60% |     63536 | 4.45M | 125 |       102 | 28.27K | 336 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:40 |
| MRX40P001 |   40.0 |  97.55% |     60809 | 4.27M | 123 |       187 |  34.2K | 329 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:40 |
| MRX40P002 |   40.0 |  97.62% |     59643 | 4.44M | 128 |       134 | 35.93K | 338 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:40 |
| MRX80P000 |   80.0 |  97.55% |     63568 | 4.45M | 125 |        88 | 23.41K | 329 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:42 |
| MRX80P001 |   80.0 |  97.51% |     60725 | 4.45M | 126 |        90 |  25.9K | 330 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:40 |
| MRX80P002 |   80.0 |  97.55% |     63708 | 4.28M | 120 |       100 | 28.83K | 327 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:41 |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.23% |     52820 | 4.45M | 162 |       886 |  47.4K | 754 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:48 |
| Q0L0X40P001   |   40.0 |  98.21% |     53766 | 4.44M | 150 |       988 | 46.62K | 660 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:45 |
| Q0L0X40P002   |   40.0 |  98.33% |     54860 | 4.45M | 148 |      1049 | 43.59K | 657 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:45 |
| Q0L0X80P000   |   80.0 |  97.09% |     36696 | 4.45M | 212 |      1022 |  19.7K | 443 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:36 |
| Q0L0X80P001   |   80.0 |  97.02% |     42702 | 4.43M | 190 |       809 | 15.62K | 396 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:36 |
| Q0L0X80P002   |   80.0 |  97.05% |     37879 | 4.46M | 226 |      1092 | 20.57K | 465 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:32 |   0:00:36 |
| Q25L60X40P000 |   40.0 |  98.41% |     63623 | 4.45M | 146 |       991 | 41.75K | 708 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:48 |
| Q25L60X40P001 |   40.0 |  98.48% |     56588 | 4.45M | 148 |       598 | 34.19K | 688 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:48 |
| Q25L60X40P002 |   40.0 |  98.31% |     57357 | 4.45M | 137 |      1221 | 43.71K | 675 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:44 |
| Q25L60X80P000 |   80.0 |  97.27% |     44618 | 4.45M | 175 |      1135 | 19.35K | 366 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:25 |   0:00:35 |
| Q25L60X80P001 |   80.0 |  97.32% |     43188 | 4.46M | 170 |      1293 | 18.71K | 360 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:38 |
| Q25L60X80P002 |   80.0 |  97.28% |     54271 | 4.46M | 162 |      1285 | 18.93K | 337 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:28 |   0:00:36 |
| Q30L60X40P000 |   40.0 |  98.42% |     63621 | 4.45M | 142 |      1006 | 48.05K | 739 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:49 |
| Q30L60X40P001 |   40.0 |  98.39% |     56791 | 4.45M | 140 |       781 | 29.61K | 711 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:45 |
| Q30L60X40P002 |   40.0 |  98.39% |     57447 | 4.45M | 145 |       994 | 50.44K | 713 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:15 |   0:00:46 |
| Q30L60X80P000 |   80.0 |  97.35% |     52933 | 4.46M | 165 |       818 |  11.8K | 350 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:25 |   0:00:37 |
| Q30L60X80P001 |   80.0 |  97.37% |     53009 | 4.46M | 161 |       926 | 17.44K | 341 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:25 |   0:00:35 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.11% |     63536 | 4.45M | 125 |       243 | 27.84K | 265 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:18 |   0:00:37 |
| MRX40P001 |   40.0 |  96.85% |     63605 | 4.44M | 125 |       453 | 30.02K | 253 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:15 |   0:00:34 |
| MRX40P002 |   40.0 |  96.93% |     60742 | 4.44M | 125 |       774 | 32.05K | 249 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:33 |
| MRX80P000 |   80.0 |  96.83% |     63568 | 4.28M | 117 |        90 |  17.1K | 230 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:35 |
| MRX80P001 |   80.0 |  96.82% |     63274 | 4.45M | 123 |        90 | 17.66K | 236 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:36 |
| MRX80P002 |   80.0 |  96.83% |     63707 | 4.28M | 119 |       105 | 20.73K | 237 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:34 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.28% |     59681 | 4.45M | 138 |       978 |  47.7K | 466 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:46 |
| Q0L0X40P001   |   40.0 |  98.25% |     59688 | 4.24M | 130 |      5577 | 92.81K | 436 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:43 |
| Q0L0X40P002   |   40.0 |  98.19% |     60891 | 4.45M | 131 |      1137 | 41.14K | 443 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:44 |
| Q0L0X80P000   |   80.0 |  97.18% |     60912 | 4.45M | 135 |      1054 | 18.57K | 275 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:34 |
| Q0L0X80P001   |   80.0 |  97.19% |     58821 | 4.46M | 131 |      1176 | 12.95K | 265 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:47 |   0:00:36 |
| Q0L0X80P002   |   80.0 |  97.24% |     56019 | 4.45M | 146 |      1094 | 17.11K | 291 |   80.0 | 3.0 |  24.7 | 172.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:36 |
| Q25L60X40P000 |   40.0 |  98.37% |     63705 | 4.45M | 133 |      1022 | 36.88K | 460 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:45 |
| Q25L60X40P001 |   40.0 |  98.31% |     60835 | 4.46M | 137 |       787 | 31.51K | 473 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:44 |
| Q25L60X40P002 |   40.0 |  98.15% |     61983 | 4.45M | 128 |      1222 |  39.7K | 436 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:40 |
| Q25L60X80P000 |   80.0 |  97.34% |     63631 | 4.45M | 129 |      1143 | 16.81K | 263 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:36 |
| Q25L60X80P001 |   80.0 |  97.43% |     63615 | 4.46M | 134 |      1019 | 14.77K | 263 |   80.0 | 3.5 |  24.3 | 174.0 | "31,41,51,61,71,81" |   0:00:47 |   0:00:38 |
| Q25L60X80P002 |   80.0 |  97.34% |     57961 | 4.46M | 135 |      1182 | 14.38K | 271 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:36 |
| Q30L60X40P000 |   40.0 |  98.31% |     63621 | 4.45M | 135 |      1045 | 42.92K | 469 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:47 |
| Q30L60X40P001 |   40.0 |  98.30% |     60394 | 4.45M | 132 |      1013 | 26.81K | 463 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:48 |
| Q30L60X40P002 |   40.0 |  98.23% |     58818 | 4.45M | 138 |      1345 | 58.15K | 467 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:43 |
| Q30L60X80P000 |   80.0 |  97.34% |     63629 | 4.46M | 133 |      1037 |  9.59K | 266 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:35 |
| Q30L60X80P001 |   80.0 |  97.54% |     63706 | 4.45M | 133 |      1302 | 19.09K | 269 |   80.0 | 3.0 |  24.7 | 172.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:37 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.06% |     63536 | 4.45M | 125 |       256 | 26.35K | 251 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:39 |   0:00:34 |
| MRX40P001 |   40.0 |  96.88% |     63605 | 4.44M | 124 |       709 | 28.77K | 241 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:34 |
| MRX40P002 |   40.0 |  96.87% |     60742 | 4.45M | 124 |       774 | 32.97K | 243 |   40.0 | 2.0 |  12.0 |  88.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:35 |
| MRX80P000 |   80.0 |  96.84% |     64143 | 4.45M | 121 |        90 | 17.07K | 231 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:49 |   0:00:35 |
| MRX80P001 |   80.0 |  96.82% |     60725 | 4.28M | 117 |        90 | 17.38K | 228 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:47 |   0:00:36 |
| MRX80P002 |   80.0 |  96.85% |     63707 | 4.28M | 116 |       105 | 20.57K | 232 |   80.0 | 4.0 |  24.0 | 176.0 | "31,41,51,61,71,81" |   0:00:47 |   0:00:35 |


Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |  # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|---:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  97.45% |     67379 | 4.46M | 118 |      4167 | 143.81K | 56 |  263.0 | 13.0 |  79.0 | 578.0 |   0:00:51 |
| 7_merge_mr_unitigs_bcalm      |  98.80% |     67317 | 4.46M | 117 |      1945 |  39.99K | 25 |  262.0 | 15.0 |  77.3 | 584.0 |   0:01:34 |
| 7_merge_mr_unitigs_superreads |  98.76% |     67317 | 4.46M | 117 |      1917 |   39.2K | 24 |  262.0 | 15.0 |  77.3 | 584.0 |   0:01:35 |
| 7_merge_mr_unitigs_tadpole    |  98.82% |     67317 | 4.46M | 117 |      1369 |   41.6K | 29 |  262.0 | 14.0 |  78.0 | 580.0 |   0:01:37 |
| 7_merge_unitigs_bcalm         |  98.58% |     67379 | 4.46M | 119 |      1916 |  72.56K | 45 |  262.0 | 14.0 |  78.0 | 580.0 |   0:01:21 |
| 7_merge_unitigs_superreads    |  98.96% |     67379 | 4.46M | 119 |      2988 |   88.3K | 40 |  263.0 | 14.0 |  78.3 | 582.0 |   0:01:46 |
| 7_merge_unitigs_tadpole       |  98.44% |     67378 | 4.46M | 118 |      4377 | 122.34K | 41 |  262.0 | 13.5 |  78.3 | 578.0 |   0:01:18 |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.45% |    112386 |  4.3M |  82 |      1241 | 16.17K | 131 |  263.0 | 12.0 |  79.7 | 574.0 |   0:00:47 |
| 8_mr_spades  |  98.94% |    132374 | 4.22M |  71 |       720 | 19.86K | 121 |  376.0 | 18.0 | 113.3 | 824.0 |   0:00:52 |
| 8_megahit    |  98.44% |     64146 | 4.29M | 118 |      1232 | 20.54K | 198 |  263.0 | 14.0 |  78.3 | 582.0 |   0:00:49 |
| 8_mr_megahit |  99.23% |    132630 |  4.5M |  79 |      1191 | 24.83K | 125 |  376.0 | 18.0 | 113.3 | 824.0 |   0:00:52 |
| 8_platanus   |  97.80% |    114666 | 4.48M |  78 |       444 | 12.02K | 121 |  263.0 | 12.0 |  79.7 | 574.0 |   0:00:48 |


Table: statFinal

| Name                     |     N50 |     Sum |   # |
|:-------------------------|--------:|--------:|----:|
| Genome                   | 4583637 | 4583637 |   1 |
| Paralogs                 |    1737 |  188158 | 111 |
| Repetitives              |    1200 |  115795 | 213 |
| 7_merge_anchors.anchors  |   67379 | 4461187 | 118 |
| 7_merge_anchors.others   |    4167 |  143808 |  56 |
| glue_anchors             |   86558 | 4460549 | 106 |
| fill_anchors             |  102091 | 4463524 |  89 |
| spades.contig            |  125538 | 4514902 | 165 |
| spades.scaffold          |  143522 | 4515652 | 155 |
| spades.non-contained     |  132337 | 4493720 |  79 |
| mr_spades.contig         |  155808 | 4523183 |  89 |
| mr_spades.scaffold       |  203812 | 4523493 |  85 |
| mr_spades.non-contained  |  155808 | 4513386 |  61 |
| megahit.contig           |   82955 | 4508133 | 168 |
| megahit.non-contained    |   82955 | 4487915 | 116 |
| mr_megahit.contig        |  133730 | 4557848 | 158 |
| mr_megahit.non-contained |  133730 | 4522134 |  70 |
| platanus.contig          |  108571 | 4558622 | 408 |
| platanus.scaffold        |  125790 | 4514164 | 168 |
| platanus.non-contained   |  132296 | 4492200 |  65 |

