# Assemble genomes of model organisms by Anchr

[TOC levels=1-3]: # ""

- [Assemble genomes of model organisms by Anchr](#assemble-genomes-of-model-organisms-by-anchr)
- [More tools on downloading and preprocessing data](#more-tools-on-downloading-and-preprocessing-data)
  - [Extra external executables](#extra-external-executables)
  - [Other leading assemblers](#other-leading-assemblers)
- [*Escherichia* virus Lambda](#escherichia-virus-lambda)
  - [lambda: reference](#lambda-reference)
  - [lambda: download](#lambda-download)
  - [lambda: template](#lambda-template)
  - [lambda: run](#lambda-run)
- [*Escherichia coli* str. K-12 substr. MG1655](#escherichia-coli-str-k-12-substr-mg1655)
  - [mg1655: reference](#mg1655-reference)
  - [mg1655: download](#mg1655-download)
  - [mg1655: template](#mg1655-template)
  - [mg1655: run](#mg1655-run)
- [*Escherichia coli* str. K-12 substr. DH5alpha](#escherichia-coli-str-k-12-substr-dh5alpha)
  - [dh5alpha: reference](#dh5alpha-reference)
  - [dh5alpha: download](#dh5alpha-download)
  - [dh5alpha: template](#dh5alpha-template)
  - [dh5alpha: run](#dh5alpha-run)


# More tools on downloading and preprocessing data

## Extra external executables

```shell script
brew install aria2 curl         # downloading tools
brew install miller

brew tap brewsci/bio
brew tap brewsci/science

brew install --HEAD quast       # assembly quality assessment. https://github.com/ablab/quast/issues/140
quast --test                    # may recompile the bundled nucmer

brew install --ignore-dependencies picard-tools

#kat comp -t 4 -n -o R 2_illumina/R1.fq.gz 2_illumina/R2.fq.gz -m 51
#kat plot spectra-mx -i -o R-spectra R-main.mx

```

## Other leading assemblers

```shell script
brew install spades
brew install megahit
brew install wang-q/tap/platanus

```

# *Escherichia* virus Lambda

Runs under macOS and Linux.

## lambda: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/lambda/1_genome
cd ~/data/anchr/lambda/1_genome

cp ~/data/anchr/ref/lambda/genome.fa .

touch paralogs.fa

```

## lambda: download

```shell script
mkdir -p ~/data/anchr/lambda/ena
cd ~/data/anchr/lambda/ena

cat << EOF > source.csv
SRX2365802,Lambda,HiSeq 2500
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -x 9 -s 3 -c -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

# sampling reads as test materials
seqtk sample -s 23 SRR5042715_1.fastq.gz 100000 | pigz > R1.fq.gz
seqtk sample -s 23 SRR5042715_2.fastq.gz 100000 | pigz > R2.fq.gz

```

| name   | srx        | platform | layout | ilength | srr        | spot     | base  |
|:-------|:-----------|:---------|:-------|:--------|:-----------|:---------|:------|
| Lambda | SRX2365802 | ILLUMINA | PAIRED |         | SRR5042715 | 16540237 | 3.33G |

* Illumina

```shell script
cd ~/data/anchr/lambda

mkdir -p 2_illumina
cd 2_illumina

seqtk sample -s 23 ../ena/SRR5042715_1.fastq.gz 100000 | pigz > R1.fq.gz
seqtk sample -s 23 ../ena/SRR5042715_2.fastq.gz 100000 | pigz > R2.fq.gz

```

## lambda: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=lambda

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 48502 \
    --parallel 4 \
    --xmx 6g \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --cutoff 30 --cutk 31" \
    --qual "20 25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "bcalm tadpole" \
    --statp 2 \
    --readl 108 \
    --uscale 2 \
    --lscale 3

```

## lambda: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=lambda

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 

bash 0_master.sh
#bash 0_cleanup.sh

```

Table: statInsertSize

| Group             |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|------:|-------------------------------:|
| R.genome.bbtools  | 430.6 |    413 | 367.3 |                         96.29% |
| R.tadpole.bbtools | 418.3 |    407 | 101.3 |                         81.92% |
| R.genome.picard   | 424.6 |  412.5 | 103.1 |                             FR |
| R.tadpole.picard  | 418.3 |    407 | 101.2 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 334       | 46251           | 0.0000       | 51.29   |
| R.31 | 297       | 46451           | 0.0000       | 50.64   |
| R.41 | 260       | 46286           | 0.0000       | 50.53   |
| R.51 | 217       | 46045           | 0.0000       | 50.65   |
| R.61 | 180       | 46218           | 0.0000       | 50.77   |
| R.71 | 143       | 46402           | 0.0000       | 50.77   |
| R.81 | 104       | 46385           | 0.0000       | 50.80   |


Table: statReads

| Name       |   N50 |    Sum |      # |
|:-----------|------:|-------:|-------:|
| Genome     | 48502 |  48502 |      1 |
| Paralogs   |     0 |      0 |      0 |
| Illumina.R |   108 |  21.6M | 200000 |
| trim.R     |   105 | 18.82M | 181790 |
| Q20L60     |   105 |  18.6M | 179679 |
| Q25L60     |   105 | 17.94M | 175041 |
| Q30L60     |   105 | 15.99M | 161943 |


Table: statTrimReads

| Name     | N50 |    Sum |      # |
|:---------|----:|-------:|-------:|
| clumpify | 108 | 21.56M | 199606 |
| highpass | 108 |  21.5M | 199066 |
| trim     | 105 | 18.82M | 181790 |
| filter   | 105 | 18.82M | 181790 |
| R1       | 105 |  9.48M |  90895 |
| R2       | 105 |  9.34M |  90895 |
| Rs       |   0 |      0 |      0 |


```text
#R.trim
#Matched	3805	1.91143%
#Name	Reads	ReadsPct
Reverse_adapter	1878	0.94341%
TruSeq_Universal_Adapter	1792	0.90020%
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	235058
#error_kmers	185792
#genomic_kmers	49266
#main_peak	269
#genome_size_in_peaks	2577144
#genome_size	2634418
#haploid_genome_size	48785
#fold_coverage	5
#haploid_fold_coverage	269
#ploidy	54
#het_rate	0.00080
#percent_repeat_in_peaks	0.000
#percent_repeat	98.128
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |    Sum |      # |
|:--------------|----:|-------:|-------:|
| clumped       | 105 | 18.82M | 181786 |
| ecco          | 105 | 18.82M | 181786 |
| eccc          | 105 | 18.82M | 181786 |
| ecct          | 105 | 18.77M | 181266 |
| extended      | 145 | 25.97M | 181266 |
| merged.raw    | 389 | 15.01M |  40051 |
| unmerged.raw  | 145 | 14.42M | 101164 |
| unmerged.trim | 145 | 14.42M | 101156 |
| M1            | 389 | 15.01M |  40029 |
| U1            | 145 |  7.27M |  50578 |
| U2            | 145 |  7.14M |  50578 |
| Us            |   0 |      0 |      0 |
| M.cor         | 267 | 29.46M | 181214 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 163.4 |    175 |  31.9 |          0.44% |
| M.ihist.merge.txt  | 374.9 |    383 |  47.4 |         44.19% |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.22% |     46140 | 47.14K |  2 |        77 |   640 | 12 |   39.0 | 2.0 |  11.7 |  86.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q0L0X40P001   |   40.0 |  97.33% |     35666 | 48.08K |  2 |        85 |   373 |  9 |   39.0 | 4.0 |  10.3 |  94.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q0L0X40P002   |   40.0 |  97.22% |     35581 | 48.03K |  2 |        71 |   447 |  8 |   38.0 | 2.0 |  11.3 |  84.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q0L0X80P000   |   80.0 |  94.67% |     35676 | 46.23K |  2 |      1144 | 1.36K |  7 |   78.0 | 4.0 |  23.3 | 172.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:12 |
| Q0L0X80P001   |   80.0 |  96.11% |     16516 | 47.94K |  3 |        92 |   496 | 12 |   76.0 | 4.5 |  22.3 | 170.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:12 |
| Q0L0X80P002   |   80.0 |  97.77% |     35621 | 47.09K |  2 |      1109 | 1.68K | 11 |   79.0 | 6.0 |  22.3 | 182.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q20L60X40P000 |   40.0 |  97.46% |     35651 | 48.09K |  2 |        83 |   295 |  6 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q20L60X40P001 |   40.0 |  97.55% |     35666 | 48.41K |  3 |        54 |   289 | 10 |   39.0 | 5.0 |   9.7 |  98.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q20L60X40P002 |   40.0 |  95.32% |     35611 | 48.01K |  3 |        83 |   278 |  5 |   38.0 | 2.0 |  11.3 |  84.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q20L60X80P000 |   80.0 |  98.23% |     35649 | 48.15K |  3 |        69 |   510 | 11 |   79.0 | 4.0 |  23.7 | 174.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q20L60X80P001 |   80.0 |  95.63% |     30610 | 47.08K |  4 |       210 |   405 |  9 |   78.0 | 8.0 |  20.7 | 188.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:12 |
| Q20L60X80P002 |   80.0 |  94.21% |     16727 | 47.45K |  5 |        68 |   264 | 11 |   76.0 | 8.5 |  19.7 | 186.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:12 |
| Q25L60X40P000 |   40.0 |  96.53% |     48039 | 48.04K |  1 |       256 |   328 |  2 |   38.0 | 0.0 |  12.7 |  76.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:11 |
| Q25L60X40P001 |   40.0 |  98.02% |     30602 | 48.15K |  4 |        69 |   575 | 19 |   39.0 | 3.0 |  11.0 |  90.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q25L60X40P002 |   40.0 |  96.48% |     35672 | 48.17K |  2 |       170 |   324 |  6 |   39.0 | 5.0 |   9.7 |  98.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q25L60X80P000 |   80.0 |  97.52% |     35676 | 48.17K |  3 |       101 |   343 | 10 |   78.0 | 7.0 |  21.3 | 184.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q25L60X80P001 |   80.0 |  98.01% |     35650 | 45.37K |  3 |       876 | 2.61K | 10 |   80.0 | 0.0 |  26.7 | 160.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q25L60X80P002 |   80.0 |  96.53% |     48004 |    48K |  1 |       250 |   387 |  2 |   78.0 | 1.0 |  25.3 | 160.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q30L60X40P000 |   40.0 |  97.33% |     26354 | 47.96K |  4 |         0 |     0 |  0 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q30L60X40P001 |   40.0 |  97.14% |     35395 | 47.91K |  3 |       102 |   425 |  4 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q30L60X40P002 |   40.0 |  97.14% |     30570 |  47.7K |  3 |       286 |   519 |  4 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:04 |   0:00:10 |
| Q30L60X80P000 |   80.0 |  98.63% |     26331 |  47.9K |  4 |       142 |   265 |  4 |   78.0 | 0.5 |  25.7 | 158.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q30L60X80P001 |   80.0 |  98.83% |     35642 | 48.07K |  3 |       101 |   589 |  9 |   80.0 | 2.0 |  25.3 | 168.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q30L60X80P002 |   80.0 |  98.12% |     35606 | 47.98K |  3 |       244 |   666 |  7 |   80.0 | 5.0 |  23.3 | 180.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  99.81% |     48187 | 48.19K |  1 |       185 | 327 |  2 |   40.0 | 1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| MRX40P001 |   40.0 |  99.59% |     35435 | 48.02K |  2 |       158 | 368 |  3 |   40.0 | 1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| MRX40P002 |   40.0 |  99.61% |     47973 | 47.97K |  1 |       248 | 415 |  2 |   40.0 | 1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| MRX80P000 |   80.0 |  99.66% |     35817 | 48.25K |  2 |       236 | 342 |  3 |   81.0 | 6.0 |  23.0 | 186.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:10 |
| MRX80P001 |   80.0 |  99.22% |     14224 | 48.03K |  4 |       205 | 724 |  7 |   80.0 | 6.0 |  22.7 | 184.0 | "31,41,51,61,71,81" |   0:00:08 |   0:00:10 |
| MRX80P002 |   80.0 |  99.61% |     35808 | 48.29K |  2 |       184 | 304 |  4 |   79.0 | 5.5 |  22.7 | 180.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:10 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  96.45% |     46156 | 47.16K |  2 |       125 |   574 |  8 |   38.0 | 1.0 |  12.0 |  80.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q0L0X40P001   |   40.0 |  95.59% |     35675 | 48.09K |  2 |        91 |   336 |  7 |   38.0 | 3.5 |  10.3 |  90.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q0L0X40P002   |   40.0 |  95.03% |     45317 | 48.01K |  2 |       267 |   373 |  3 |   38.0 | 0.0 |  12.7 |  76.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q0L0X80P000   |   80.0 |  97.97% |     35670 | 46.22K |  2 |      1236 | 1.97K | 10 |   78.0 | 0.0 |  26.0 | 156.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q0L0X80P001   |   80.0 |  95.65% |     35668 | 48.12K |  2 |       201 |   556 |  5 |   76.0 | 7.0 |  20.7 | 180.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q0L0X80P002   |   80.0 |  95.92% |     47095 |  47.1K |  1 |      1114 | 1.37K |  5 |   75.0 | 7.0 |  20.3 | 178.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q20L60X40P000 |   40.0 |  97.56% |     35651 |  48.1K |  2 |        88 |   455 |  6 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q20L60X40P002 |   40.0 |  95.48% |     35611 | 48.01K |  3 |       225 |   399 |  4 |   38.0 | 2.0 |  11.3 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q20L60X80P000 |   80.0 |  98.27% |     35632 | 48.13K |  3 |       107 |   745 | 10 |   79.0 | 0.0 |  26.3 | 158.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q20L60X80P001 |   80.0 |  97.14% |     35404 | 47.82K |  2 |       126 |   549 |  6 |   78.0 | 2.0 |  24.7 | 164.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q20L60X80P002 |   80.0 |  95.70% |     16728 | 48.26K |  4 |       174 |   561 | 10 |   76.0 | 8.0 |  20.0 | 184.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X40P000 |   40.0 |  96.54% |     48050 | 48.05K |  1 |       256 |   334 |  2 |   38.0 | 0.0 |  12.7 |  76.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X40P001 |   40.0 |  96.40% |     30608 | 48.18K |  4 |       114 |   213 |  9 |   38.0 | 4.0 |  10.0 |  92.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q25L60X40P002 |   40.0 |  96.39% |     37524 | 48.08K |  2 |       174 |   307 |  3 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| Q25L60X80P000 |   80.0 |  97.97% |     35674 | 48.17K |  3 |       210 |   602 |  8 |   78.0 | 5.5 |  22.3 | 178.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X80P001 |   80.0 |  98.59% |     35678 | 49.05K |  3 |      1114 | 1.65K | 10 |   79.0 | 5.5 |  22.7 | 180.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q25L60X80P002 |   80.0 |  96.63% |     48004 |    48K |  1 |       260 |   510 |  2 |   77.0 | 0.0 |  25.7 | 154.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| Q30L60X40P000 |   40.0 |  97.47% |     26354 | 47.96K |  4 |         0 |     0 |  0 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q30L60X40P001 |   40.0 |  97.16% |     35395 | 47.91K |  3 |       188 |   519 |  4 |   39.0 | 1.0 |  12.3 |  82.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| Q30L60X40P002 |   40.0 |  97.16% |     30570 |  47.7K |  3 |       157 |   583 |  4 |   39.0 | 0.0 |  13.0 |  78.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:10 |
| Q30L60X80P000 |   80.0 |  97.41% |     26331 |  47.9K |  4 |         0 |     0 |  0 |   78.0 | 1.0 |  25.3 | 160.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:10 |
| Q30L60X80P001 |   80.0 |  97.32% |     37389 | 48.05K |  2 |       230 |   345 |  3 |   78.0 | 1.0 |  25.3 | 160.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:11 |
| Q30L60X80P002 |   80.0 |  98.00% |     48079 | 48.08K |  1 |       262 |   440 |  4 |   75.0 | 6.0 |  21.0 | 174.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  99.81% |     48187 | 48.19K |  1 |       185 | 327 |  2 |   40.0 |  1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:11 |
| MRX40P001 |   40.0 |  99.60% |     35435 | 48.02K |  2 |       158 | 386 |  3 |   40.0 |  1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| MRX40P002 |   40.0 |  99.72% |     47989 | 47.99K |  1 |       277 | 525 |  2 |   40.0 |  1.0 |  12.7 |  84.0 | "31,41,51,61,71,81" |   0:00:05 |   0:00:09 |
| MRX80P000 |   80.0 |  99.66% |     35817 | 48.25K |  2 |       236 | 342 |  3 |   81.0 |  6.0 |  23.0 | 186.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| MRX80P001 |   80.0 |  99.53% |     47276 | 48.32K |  2 |       182 | 270 |  4 |   81.0 | 15.0 |  17.0 | 222.0 | "31,41,51,61,71,81" |   0:00:06 |   0:00:11 |
| MRX80P002 |   80.0 |  99.61% |     35809 |  48.3K |  2 |       184 | 298 |  3 |   79.0 |  5.5 |  22.7 | 180.0 | "31,41,51,61,71,81" |   0:00:07 |   0:00:11 |


Table: statMergeAnchors.md

| Name                       | Mapped% | N50Anchor |    Sum |  # | N50Others |   Sum |  # | median | MAD | lower | upper | RunTimeAN |
|:---------------------------|--------:|----------:|-------:|---:|----------:|------:|---:|-------:|----:|------:|------:|----------:|
| 7_merge_anchors            |   0.00% |     48372 | 48.37K |  1 |      1236 | 1.24K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_mr_unitigs_bcalm   |   0.00% |     48364 | 48.36K |  1 |      1236 | 1.24K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_mr_unitigs_tadpole |   0.00% |     48372 | 48.37K |  1 |         0 |     0 |  0 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_unitigs_bcalm      |   0.00% |     48180 | 48.18K |  1 |      1144 | 1.14K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |
| 7_merge_unitigs_tadpole    |   0.00% |     48196 |  48.2K |  1 |      1236 | 1.24K |  1 |    0.0 | 0.0 |   0.0 |   0.0 |           |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |    Sum |  # | N50Others | Sum |  # | median | MAD | lower |  upper | RunTimeAN |
|:-------------|--------:|----------:|-------:|---:|----------:|----:|---:|-------:|----:|------:|-------:|----------:|
| 8_mr_spades  |  99.57% |     26405 | 48.13K |  2 |       429 | 494 |  3 |  603.0 | 1.0 | 200.3 | 1210.0 |   0:00:11 |
| 8_megahit    |  95.15% |     48024 | 48.02K |  1 |       262 | 490 |  2 |  370.0 | 1.0 | 122.7 |  744.0 |   0:00:12 |
| 8_mr_megahit |  99.74% |     48072 | 48.07K |  1 |       242 | 442 |  2 |  605.0 | 0.0 | 201.7 | 1210.0 |   0:00:11 |


Table: statFinal

| Name                     |   N50 |   Sum |  # |
|:-------------------------|------:|------:|---:|
| Genome                   | 48502 | 48502 |  1 |
| Paralogs                 |     0 |     0 |  0 |
| 7_merge_anchors.anchors  | 48372 | 48372 |  1 |
| 7_merge_anchors.others   |  1236 |  1236 |  1 |
| spades.non-contained     |     0 |     0 |  0 |
| mr_spades.contig         | 48627 | 48753 |  2 |
| mr_spades.scaffold       | 48627 | 48753 |  2 |
| mr_spades.non-contained  | 48627 | 48627 |  1 |
| megahit.contig           | 48514 | 48514 |  1 |
| megahit.non-contained    | 48514 | 48514 |  1 |
| mr_megahit.contig        | 48514 | 48514 |  1 |
| mr_megahit.non-contained | 48514 | 48514 |  1 |
| platanus.non-contained   |     0 |     0 |  0 |


# *Escherichia coli* str. K-12 substr. MG1655

## mg1655: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/mg1655/1_genome
cd ~/data/anchr/mg1655/1_genome

cp ~/data/anchr/ref/mg1655/genome.fa .
cp ~/data/anchr/ref/mg1655/paralogs.fa .
cp ~/data/anchr/ref/mg1655/repetitives.fa .

```

## mg1655: download

* Illumina

```shell script
cd ~/data/anchr/mg1655

mkdir -p 2_illumina
cd 2_illumina

aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz
aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz

ln -s MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz R1.fq.gz
ln -s MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz R2.fq.gz

```

## mg1655: template

* Rsync to hpcc

```shell script
rsync -avP \
    ~/data/anchr/mg1655/ \
    wangq@202.119.37.251:data/anchr/mg1655

# rsync -avP wangq@202.119.37.251:data/anchr/mg1655/ ~/data/anchr/mg1655

```

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4641652 \
    --parallel 24 \
    --xmx 80g \
    --queue mpi \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --tile --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --readl 151 \
    --uscale 2 \
    --lscale 3 \
    --redo \
    \
    --extend

```

## mg1655: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 
# rm -fr 4_down_sampling 6_down_sampling

# BASE_NAME=mg1655 bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
# bkill -J "${BASE_NAME}-*"

# bash 0_master.sh
# bash 0_cleanup.sh

```

Table: statInsertSize

| Group             |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|------:|-------------------------------:|
| R.genome.bbtools  | 321.9 |    298 | 967.3 |                         96.08% |
| R.tadpole.bbtools | 294.7 |    296 |  22.0 |                         81.33% |
| R.genome.picard   | 298.2 |    298 |  18.0 |                             FR |
| R.tadpole.picard  | 294.9 |    296 |  21.6 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 292       | 4390759         | 0.1630       | 54.20   |
| R.31 | 265       | 4388871         | 0.1382       | 54.04   |
| R.41 | 239       | 4375332         | 0.1108       | 53.90   |
| R.51 | 216       | 4407280         | 0.1088       | 53.78   |
| R.61 | 191       | 4393867         | 0.1081       | 53.68   |
| R.71 | 168       | 4390152         | 0.1042       | 53.58   |
| R.81 | 144       | 4387393         | 0.0989       | 53.50   |


Table: statReads

| Name        |     N50 |     Sum |        # |
|:------------|--------:|--------:|---------:|
| Genome      | 4641652 | 4641652 |        1 |
| Paralogs    |    1937 |  187300 |      106 |
| Repetitives |    1205 |  110138 |      206 |
| Illumina.R  |     151 |   1.73G | 11458940 |
| trim.R      |     149 |   1.43G | 10368378 |
| Q20L60      |     149 |   1.41G | 10229452 |
| Q25L60      |     148 |   1.32G |  9943905 |
| Q30L60      |     128 |   1.11G |  9315096 |


Table: statTrimReads

| Name           | N50 |     Sum |        # |
|:---------------|----:|--------:|---------:|
| clumpify       | 151 |   1.73G | 11439000 |
| filteredbytile | 151 |   1.67G | 11066320 |
| highpass       | 151 |   1.66G | 10993372 |
| trim           | 149 |   1.43G | 10368378 |
| filter         | 149 |   1.43G | 10368378 |
| R1             | 150 |  736.3M |  5184189 |
| R2             | 144 | 690.68M |  5184189 |
| Rs             |   0 |       0 |        0 |


```text
#R.trim
#Matched	17496	0.15915%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	20721642
#error_kmers	16193867
#genomic_kmers	4527775
#main_peak	240
#genome_size_in_peaks	4602227
#genome_size	4642460
#haploid_genome_size	4642460
#fold_coverage	240
#haploid_fold_coverage	240
#ploidy	1
#percent_repeat_in_peaks	1.620
#percent_repeat	1.824
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |    Sum |        # |
|:--------------|----:|-------:|---------:|
| clumped       | 149 |  1.43G | 10367200 |
| ecco          | 149 |  1.43G | 10367200 |
| eccc          | 149 |  1.43G | 10367200 |
| ecct          | 149 |  1.42G | 10318674 |
| extended      | 189 |  1.83G | 10318674 |
| merged.raw    | 339 |  1.72G |  5091675 |
| unmerged.raw  | 175 | 19.83M |   135324 |
| unmerged.trim | 175 | 19.83M |   135264 |
| M1            | 339 |  1.71G |  5064129 |
| U1            | 181 | 10.43M |    67632 |
| U2            | 168 |   9.4M |    67632 |
| Us            |   0 |      0 |        0 |
| M.cor         | 338 |  1.73G | 10263522 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 271.6 |    277 |  23.6 |         10.85% |
| M.ihist.merge.txt  | 337.7 |    338 |  19.3 |         98.69% |


Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 307.4 |  285.7 |    7.05% | "67" | 4.64M | 4.66M |     1.00 | 0:02'55'' |
| Q25L60.R | 284.1 |  273.2 |    3.84% | "63" | 4.64M | 4.57M |     0.98 | 0:02'41'' |
| Q30L60.R | 238.4 |  233.6 |    2.03% | "51" | 4.64M | 4.55M |     0.98 | 0:02'12'' |


Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |    # | N50Others |    Sum |    # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|-----:|----------:|-------:|-----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  96.02% |      9716 |  4.5M |  702 |        33 | 40.48K | 1507 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:52 |
| Q0L0X40P001   |   40.0 |  96.09% |     10416 |  4.5M |  643 |        34 | 39.49K | 1479 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:52 |
| Q0L0X40P002   |   40.0 |  96.04% |     10729 | 4.49M |  651 |        37 | 42.75K | 1466 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:32 |
| Q0L0X80P000   |   80.0 |  92.32% |      4815 | 4.34M | 1158 |        41 | 81.05K | 2320 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:38 |
| Q0L0X80P001   |   80.0 |  92.00% |      4956 | 4.32M | 1170 |        42 | 83.05K | 2348 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:36 |
| Q0L0X80P002   |   80.0 |  91.78% |      4799 | 4.31M | 1173 |        48 | 89.12K | 2335 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:40 |
| Q25L60X40P000 |   40.0 |  97.81% |     16546 | 4.53M |  427 |        34 | 32.79K | 1218 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:37 |
| Q25L60X40P001 |   40.0 |  97.76% |     16858 | 4.53M |  405 |        32 | 28.45K | 1150 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:46 |
| Q25L60X40P002 |   40.0 |  97.88% |     18246 | 4.52M |  409 |        36 | 32.97K | 1213 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:38 |
| Q25L60X80P000 |   80.0 |  96.53% |     10031 |  4.5M |  677 |        40 | 44.87K | 1434 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:38 |
| Q25L60X80P001 |   80.0 |  96.64% |     10465 | 4.49M |  639 |        44 | 47.62K | 1402 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:37 |
| Q25L60X80P002 |   80.0 |  96.64% |     10803 | 4.49M |  639 |        45 | 46.78K | 1398 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:38 |
| Q30L60X40P000 |   40.0 |  98.53% |     29210 | 4.54M |  260 |        36 | 38.46K | 1294 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:46 |
| Q30L60X40P001 |   40.0 |  98.59% |     31262 | 4.54M |  234 |        32 | 32.79K | 1282 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:43 |
| Q30L60X40P002 |   40.0 |  98.51% |     31227 | 4.53M |  243 |        33 | 35.56K | 1264 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:40 |
| Q30L60X80P000 |   80.0 |  98.50% |     30562 | 4.53M |  252 |        34 | 30.93K | 1114 |   79.0 | 14.0 |  12.3 | 242.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:43 |
| Q30L60X80P001 |   80.0 |  98.52% |     31598 | 4.53M |  236 |        32 | 29.59K | 1097 |   79.0 | 13.0 |  13.3 | 236.0 | "31,41,51,61,71,81" |   0:01:09 |   0:00:47 |


Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.95% |     41431 | 4.51M | 190 |        94 | 29.32K | 452 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:58 |
| MRX40P001 |   40.0 |  96.69% |     41001 | 4.51M | 200 |        94 | 31.63K | 459 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:57 |   0:00:56 |
| MRX40P002 |   40.0 |  96.97% |     40700 | 4.52M | 190 |        89 | 28.42K | 464 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:58 |   0:00:31 |
| MRX80P000 |   80.0 |  96.18% |     28048 |  4.5M | 283 |       110 | 45.78K | 626 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:36 |   0:00:30 |
| MRX80P001 |   80.0 |  96.21% |     28782 |  4.5M | 261 |       107 | 40.75K | 577 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:39 |   0:00:31 |
| MRX80P002 |   80.0 |  96.29% |     30464 |  4.5M | 280 |       109 |  46.6K | 624 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:34 |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.99% |     20953 | 4.53M | 362 |        36 | 49.17K | 1661 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:03:04 |   0:01:05 |
| Q0L0X40P001   |   40.0 |  98.00% |     20650 | 4.53M | 352 |        33 | 44.92K | 1622 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:01:19 |   0:01:04 |
| Q0L0X40P002   |   40.0 |  98.07% |     20853 | 4.53M | 345 |        32 | 44.61K | 1625 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:01:36 |   0:00:41 |
| Q0L0X80P000   |   80.0 |  95.85% |      9427 | 4.48M | 702 |        42 | 56.01K | 1617 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:02:22 |   0:00:38 |
| Q0L0X80P001   |   80.0 |  95.99% |      9567 | 4.48M | 699 |        43 | 58.61K | 1632 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:02:34 |   0:00:40 |
| Q0L0X80P002   |   80.0 |  95.83% |      8866 | 4.47M | 739 |        41 | 59.42K | 1730 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:02:06 |   0:00:44 |
| Q25L60X40P000 |   40.0 |  98.51% |     25536 | 4.54M | 310 |        58 | 76.51K | 1734 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:44 |
| Q25L60X40P001 |   40.0 |  98.38% |     24620 | 4.53M | 310 |        37 | 50.56K | 1645 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:02:12 |   0:00:46 |
| Q25L60X40P002 |   40.0 |  98.32% |     23968 | 4.53M | 316 |        37 | 53.82K | 1680 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:01:57 |   0:00:42 |
| Q25L60X80P000 |   80.0 |  97.64% |     15013 | 4.52M | 461 |        38 | 41.55K | 1371 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:39 |
| Q25L60X80P001 |   80.0 |  97.76% |     15584 | 4.52M | 465 |        43 | 50.38K | 1456 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:57 |   0:00:41 |
| Q25L60X80P002 |   80.0 |  97.78% |     17036 | 4.52M | 445 |        43 | 47.93K | 1418 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:34 |   0:00:43 |
| Q30L60X40P000 |   40.0 |  98.40% |     24458 | 4.53M | 329 |       587 | 62.23K | 1802 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:02:03 |   0:00:44 |
| Q30L60X40P001 |   40.0 |  98.46% |     24191 | 4.54M | 324 |        37 | 55.49K | 1859 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:02:06 |   0:00:45 |
| Q30L60X40P002 |   40.0 |  98.38% |     22308 | 4.54M | 339 |        33 |  51.3K | 1786 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:03:03 |   0:00:45 |
| Q30L60X80P000 |   80.0 |  98.57% |     30332 | 4.54M | 256 |        30 | 36.86K | 1417 |   80.0 | 14.0 |  12.7 | 244.0 | "31,41,51,61,71,81" |   0:01:46 |   0:00:47 |
| Q30L60X80P001 |   80.0 |  98.57% |     30006 | 4.53M | 244 |        35 |  41.7K | 1405 |   80.0 | 14.0 |  12.7 | 244.0 | "31,41,51,61,71,81" |   0:02:31 |   0:00:44 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.67% |     55842 | 4.52M | 140 |        88 | 18.67K | 301 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:57 |   0:00:46 |
| MRX40P001 |   40.0 |  96.65% |     58649 | 4.52M | 150 |        89 | 19.73K | 315 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:02:45 |   0:00:47 |
| MRX40P002 |   40.0 |  96.70% |     53682 | 4.52M | 142 |        87 | 17.71K | 308 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:31 |
| MRX80P000 |   80.0 |  96.34% |     39030 | 4.51M | 206 |       108 | 29.93K | 433 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:46 |   0:00:33 |
| MRX80P001 |   80.0 |  96.42% |     43094 | 4.51M | 192 |       104 |  26.9K | 405 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:52 |   0:00:37 |
| MRX80P002 |   80.0 |  96.45% |     42631 | 4.51M | 193 |       100 | 26.05K | 403 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:41 |   0:00:36 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.23% |     30976 | 4.53M | 256 |        42 | 42.18K | 1193 |   40.0 |  6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:02:01 |   0:00:45 |
| Q0L0X40P001   |   40.0 |  98.20% |     30946 | 4.53M | 250 |        37 | 36.75K | 1156 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:46 |
| Q0L0X40P002   |   40.0 |  98.20% |     33920 | 4.52M | 230 |        40 | 38.58K | 1109 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:36 |
| Q0L0X80P000   |   80.0 |  97.14% |     16205 | 4.51M | 431 |        42 | 43.97K | 1165 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:33 |
| Q0L0X80P001   |   80.0 |  97.17% |     17349 | 4.52M | 424 |        42 | 44.24K | 1145 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:00:48 |   0:00:34 |
| Q0L0X80P002   |   80.0 |  97.02% |     15814 | 4.51M | 435 |        39 | 42.82K | 1222 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:35 |
| Q25L60X40P000 |   40.0 |  98.51% |     34354 | 4.53M | 228 |        37 | 39.61K | 1260 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:40 |
| Q25L60X40P001 |   40.0 |  98.52% |     35170 | 4.53M | 222 |        34 | 35.75K | 1287 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:41 |
| Q25L60X40P002 |   40.0 |  98.42% |     31576 | 4.52M | 237 |        38 | 41.34K | 1286 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:41 |
| Q25L60X80P000 |   80.0 |  98.08% |     22799 | 4.53M | 328 |        37 | 35.94K | 1148 |   80.0 | 12.0 |  14.7 | 232.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:40 |
| Q25L60X80P001 |   80.0 |  98.05% |     22738 | 4.52M | 332 |        40 | 40.67K | 1165 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:39 |
| Q25L60X80P002 |   80.0 |  98.06% |     22924 | 4.52M | 324 |        43 | 41.64K | 1138 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:00:47 |   0:00:38 |
| Q30L60X40P000 |   40.0 |  98.49% |     29246 | 4.53M | 271 |        36 | 45.22K | 1465 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:39 |
| Q30L60X40P001 |   40.0 |  98.53% |     28683 | 4.53M | 259 |        30 | 39.87K | 1524 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:44 |
| Q30L60X40P002 |   40.0 |  98.48% |     25288 | 4.54M | 269 |        87 | 60.08K | 1504 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:44 |
| Q30L60X80P000 |   80.0 |  98.55% |     32565 | 4.53M | 228 |        32 | 33.49K | 1213 |   80.0 | 14.0 |  12.7 | 244.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:45 |
| Q30L60X80P001 |   80.0 |  98.55% |     34034 | 4.53M | 219 |        31 | 31.83K | 1209 |   80.0 | 13.0 |  13.7 | 238.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:47 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.76% |     60374 | 4.52M | 121 |        86 | 16.72K | 263 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:18 |   0:00:41 |
| MRX40P001 |   40.0 |  96.74% |     60741 | 4.52M | 129 |        84 |  17.3K | 277 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:42 |
| MRX40P002 |   40.0 |  96.74% |     62433 | 4.52M | 124 |        82 | 16.02K | 268 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:30 |
| MRX80P000 |   80.0 |  96.49% |     51961 | 4.52M | 161 |        99 | 21.19K | 332 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:00:50 |   0:00:32 |
| MRX80P001 |   80.0 |  96.50% |     51277 | 4.52M | 159 |       105 | 22.95K | 329 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:33 |
| MRX80P002 |   80.0 |  96.63% |     57756 | 4.52M | 150 |        99 | 22.03K | 316 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:00:54 |   0:00:33 |


Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |  # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|---:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  98.20% |     63664 | 4.53M | 129 |     19518 |  311.4K | 73 |  285.0 | 35.0 |  60.0 | 780.0 |   0:01:06 |
| 7_merge_mr_unitigs_bcalm      |  98.91% |     63552 | 4.53M | 132 |      1054 |    8.7K |  9 |  285.0 | 35.0 |  60.0 | 780.0 |   0:01:56 |
| 7_merge_mr_unitigs_superreads |  98.97% |     59668 | 4.52M | 136 |     39160 |  78.22K | 12 |  285.0 | 35.0 |  60.0 | 780.0 |   0:01:53 |
| 7_merge_mr_unitigs_tadpole    |  98.96% |     63608 | 4.53M | 129 |     84935 |  97.02K | 10 |  284.0 | 36.0 |  58.7 | 784.0 |   0:01:45 |
| 7_merge_unitigs_bcalm         |  98.95% |     63412 | 4.53M | 139 |      2553 | 100.49K | 52 |  284.0 | 35.0 |  59.7 | 778.0 |   0:01:48 |
| 7_merge_unitigs_superreads    |  98.83% |     59677 | 4.53M | 137 |      2273 |  52.73K | 29 |  284.0 | 35.0 |  59.7 | 778.0 |   0:01:29 |
| 7_merge_unitigs_tadpole       |  98.92% |     63656 | 4.53M | 134 |      9538 |  92.58K | 36 |  285.0 | 35.0 |  60.0 | 780.0 |   0:01:50 |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.54% |     95433 | 4.27M |  87 |      1124 | 16.29K | 160 |  285.0 | 35.0 |  60.0 | 780.0 |   0:00:51 |
| 8_mr_spades  |  98.40% |    148466 | 4.55M |  71 |      1029 | 23.43K | 147 |  371.0 | 36.0 |  87.7 | 958.0 |   0:00:54 |
| 8_megahit    |  98.18% |     65379 | 4.53M | 125 |       960 | 18.76K | 232 |  285.0 | 35.0 |  60.0 | 780.0 |   0:00:53 |
| 8_mr_megahit |  98.81% |    132783 | 4.57M |  70 |       778 | 18.46K | 136 |  372.0 | 36.0 |  88.0 | 960.0 |   0:00:56 |
| 8_platanus   |  97.87% |     96791 | 4.06M |  81 |       310 |  9.55K | 138 |  285.0 | 35.0 |  60.0 | 780.0 |   0:00:46 |


Table: statFinal

| Name                     |     N50 |     Sum |    # |
|:-------------------------|--------:|--------:|-----:|
| Genome                   | 4641652 | 4641652 |    1 |
| Paralogs                 |    1937 |  187300 |  106 |
| Repetitives              |    1205 |  110138 |  206 |
| 7_merge_anchors.anchors  |   63664 | 4530052 |  129 |
| 7_merge_anchors.others   |   19518 |  311402 |   73 |
| glue_anchors             |   73237 | 4529725 |  120 |
| fill_anchors             |  112448 | 4532047 |   83 |
| spades.contig            |  132608 | 4573776 |  137 |
| spades.scaffold          |  133063 | 4573960 |  133 |
| spades.non-contained     |  132608 | 4555564 |   75 |
| mr_spades.contig         |  148607 | 4588030 |  151 |
| mr_spades.scaffold       |  148607 | 4588230 |  149 |
| mr_spades.non-contained  |  148607 | 4569189 |   76 |
| megahit.contig           |   82825 | 4568235 |  150 |
| megahit.non-contained    |   82825 | 4551812 |  107 |
| mr_megahit.contig        |  132896 | 4607281 |  122 |
| mr_megahit.non-contained |  132896 | 4584522 |   66 |
| platanus.contig          |   14887 | 4712287 | 1172 |
| platanus.scaffold        |  148483 | 4577850 |  142 |
| platanus.non-contained   |  176491 | 4560066 |   64 |


# *Escherichia coli* str. K-12 substr. DH5alpha

## dh5alpha: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/dh5alpha/1_genome
cd ~/data/anchr/dh5alpha/1_genome

cp ~/data/anchr/ref/dh5alpha/genome.fa .
cp ~/data/anchr/ref/dh5alpha/paralogs.fa .
cp ~/data/anchr/ref/dh5alpha/repetitives.fa .

```

## dh5alpha: download

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p ena
cd ena

cat << EOF > source.csv
SRP251726,dh5alpha,HiSeq 2500 PE125
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -x 9 -s 3 -c -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

```

| name     | srx        | platform        | layout | ilength | srr         | spot    | base  |
|:---------|:-----------|:----------------|:-------|:--------|:------------|:--------|:------|
| dh5alpha | SRX7856678 | ILLUMINA        | PAIRED |         | SRR11245239 | 5881654 | 1.37G |
| dh5alpha | SRX7856679 | OXFORD_NANOPORE | SINGLE |         | SRR11245238 | 346489  | 3.35G |


* Illumina

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p 2_illumina
cd 2_illumina

ln -s ../ena/SRR11245239_1.fastq.gz R1.fq.gz
ln -s ../ena/SRR11245239_2.fastq.gz R2.fq.gz

```

## dh5alpha: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4583637 \
    --parallel 24 \
    --xmx 80g \
    --queue mpi \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --readl 125 \
    --uscale 2 \
    --lscale 3 \
    --redo \
    \
    --extend

```

## dh5alpha: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md 
# rm -fr 4_down_sampling 6_down_sampling

# BASE_NAME=dh5alpha bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
# bkill -J "${BASE_NAME}-*"

# bash 0_master.sh
# bash 0_cleanup.sh

```


Table: statInsertSize

| Group             |  Mean | Median |  STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|-------:|-------------------------------:|
| R.genome.bbtools  | 432.7 |    346 | 1241.2 |                         99.98% |
| R.tadpole.bbtools | 389.1 |    341 |  206.1 |                         95.12% |
| R.genome.picard   | 394.8 |    346 |  208.3 |                             FR |
| R.tadpole.picard  | 389.1 |    341 |  205.7 |                             FR |


Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 260       | 4346013         | 0.0814       | 51.62   |
| R.31 | 233       | 4360998         | 0.0563       | 51.54   |
| R.41 | 209       | 4433088         | 0.0607       | 51.50   |
| R.51 | 182       | 4413726         | 0.0150       | 51.47   |
| R.61 | 157       | 4456634         | 0.0332       | 51.45   |
| R.71 | 132       | 4495949         | 0.0194       | 51.44   |
| R.81 | 107       | 4475448         | 0.0160       | 51.42   |


Table: statReads

| Name        |     N50 |     Sum |        # |
|:------------|--------:|--------:|---------:|
| Genome      | 4583637 | 4583637 |        1 |
| Paralogs    |    1737 |  188158 |      111 |
| Repetitives |    1200 |  115795 |      213 |
| Illumina.R  |     125 |   1.47G | 11763308 |
| trim.R      |     125 |   1.37G | 10962262 |
| Q25L60      |     125 |   1.25G | 10280938 |
| Q30L60      |     125 |   1.13G |  9405535 |


Table: statTrimReads

| Name     | N50 |     Sum |        # |
|:---------|----:|--------:|---------:|
| clumpify | 125 |   1.37G | 10970530 |
| highpass | 125 |   1.37G | 10966136 |
| trim     | 125 |   1.37G | 10962262 |
| filter   | 125 |   1.37G | 10962262 |
| R1       | 125 |    683M |  5481131 |
| R2       | 125 | 683.63M |  5481131 |
| Rs       |   0 |       0 |        0 |


```text
#R.trim
#Matched	5620	0.05125%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	35779038
#error_kmers	31288377
#genomic_kmers	4490661
#main_peak	221
#genome_size_in_peaks	9162708
#genome_size	10213708
#haploid_genome_size	5106854
#fold_coverage	96
#haploid_fold_coverage	221
#ploidy	2
#het_rate	0.00012
#percent_repeat_in_peaks	2.524
#percent_repeat	55.129
#start	center	stop	max	volume
```


Table: statMergeReads

| Name          | N50 |     Sum |        # |
|:--------------|----:|--------:|---------:|
| clumped       | 125 |   1.37G | 10959444 |
| ecco          | 125 |   1.37G | 10959444 |
| eccc          | 125 |   1.37G | 10959444 |
| ecct          | 125 |   1.37G | 10952912 |
| extended      | 165 |    1.8G | 10952912 |
| merged.raw    | 343 |    1.1G |  3511631 |
| unmerged.raw  | 165 | 645.89M |  3929650 |
| unmerged.trim | 165 | 645.89M |  3929650 |
| M1            | 343 |   1.06G |  3404347 |
| U1            | 165 | 322.85M |  1964825 |
| U2            | 165 | 323.04M |  1964825 |
| Us            |   0 |       0 |        0 |
| M.cor         | 250 |   1.71G | 10738344 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 180.2 |    180 |  32.0 |         24.10% |
| M.ihist.merge.txt  | 312.5 |    310 |  87.7 |         64.12% |


Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 298.2 |  262.2 |   12.07% | "87" | 4.58M |  4.6M |     1.00 | 0:03'09'' |
| Q25L60.R | 273.7 |  253.6 |    7.35% | "87" | 4.58M | 4.53M |     0.99 | 0:02'40'' |
| Q30L60.R | 246.5 |  232.4 |    5.71% | "87" | 4.58M | 4.52M |     0.99 | 0:02'26'' |


Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.18% |     29825 | 4.46M | 272 |        38 | 23.42K |  635 |   39.0 | 6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:50 |   0:00:37 |
| Q0L0X40P001   |   40.0 |  97.23% |     31028 | 4.46M | 263 |        35 | 22.45K |  623 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:39 |
| Q0L0X40P002   |   40.0 |  97.15% |     30806 | 4.45M | 266 |        34 | 21.43K |  654 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:31 |
| Q0L0X80P000   |   80.0 |  96.17% |     12488 | 4.43M | 545 |        34 | 44.59K | 1159 |   79.0 | 9.0 |  17.3 | 212.0 | "31,41,51,61,71,81" |   0:01:08 |   0:00:33 |
| Q0L0X80P001   |   80.0 |  95.82% |     12438 | 4.42M | 538 |        32 | 40.02K | 1153 |   79.0 | 9.0 |  17.3 | 212.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:34 |
| Q0L0X80P002   |   80.0 |  95.87% |     13319 | 4.39M | 512 |        34 | 43.08K | 1108 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:37 |
| Q25L60X40P000 |   40.0 |  97.59% |     43856 | 4.46M | 195 |        42 | 22.24K |  504 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:35 |
| Q25L60X40P001 |   40.0 |  97.71% |     47476 | 4.46M | 175 |        35 |  17.8K |  493 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:35 |
| Q25L60X40P002 |   40.0 |  97.62% |     42681 | 4.46M | 184 |        39 | 19.31K |  501 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:33 |
| Q25L60X80P000 |   80.0 |  97.04% |     22455 | 4.45M | 321 |        31 | 24.68K |  736 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:08 |   0:00:42 |
| Q25L60X80P001 |   80.0 |  97.01% |     25537 | 4.45M | 312 |        33 | 25.91K |  707 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:08 |   0:00:48 |
| Q25L60X80P002 |   80.0 |  97.10% |     23832 | 4.45M | 314 |        35 | 28.81K |  719 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:09 |   0:00:45 |
| Q30L60X40P000 |   40.0 |  97.83% |     42116 | 4.46M | 190 |        36 |  19.6K |  531 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:43 |
| Q30L60X40P001 |   40.0 |  97.79% |     45905 | 4.46M | 183 |        35 | 18.06K |  512 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:42 |
| Q30L60X40P002 |   40.0 |  97.85% |     48604 | 4.45M | 167 |        36 | 17.92K |  498 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:40 |
| Q30L60X80P000 |   80.0 |  97.17% |     26743 | 4.45M | 292 |        31 | 22.21K |  673 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:44 |
| Q30L60X80P001 |   80.0 |  97.09% |     24640 | 4.42M | 281 |        31 | 21.57K |  647 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:39 |


Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.58% |     64161 | 4.46M | 119 |        71 | 18.17K | 325 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:56 |   0:00:38 |
| MRX40P001 |   40.0 |  97.60% |     67268 | 4.45M | 115 |        86 | 22.24K | 322 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:39 |
| MRX40P002 |   40.0 |  97.58% |     67296 | 4.29M | 111 |        69 | 15.95K | 323 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:36 |
| MRX80P000 |   80.0 |  97.56% |     63654 | 4.45M | 119 |        83 | 21.71K | 329 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:38 |
| MRX80P001 |   80.0 |  97.56% |     64140 | 4.46M | 122 |        72 | 15.96K | 332 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:36 |
| MRX80P002 |   80.0 |  97.59% |     73656 | 4.46M | 116 |        80 | 17.25K | 329 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:28 |   0:00:37 |


Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.26% |     57929 | 4.46M | 140 |        59 | 36.05K | 675 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:40 |   0:01:06 |
| Q0L0X40P001   |   40.0 |  98.43% |     63584 | 4.46M | 138 |        49 | 32.65K | 670 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:13 |   0:01:08 |
| Q0L0X40P002   |   40.0 |  98.13% |     55658 | 4.46M | 148 |        42 | 27.08K | 649 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:35 |
| Q0L0X80P000   |   80.0 |  97.08% |     43442 | 4.46M | 185 |        45 | 19.83K | 387 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:56 |   0:00:30 |
| Q0L0X80P001   |   80.0 |  97.01% |     40119 | 4.46M | 200 |        36 | 16.66K | 408 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:47 |   0:00:30 |
| Q0L0X80P002   |   80.0 |  97.02% |     40328 | 4.44M | 205 |        43 | 21.42K | 430 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:31 |   0:00:32 |
| Q25L60X40P000 |   40.0 |  98.52% |     63598 | 4.46M | 130 |        43 | 29.19K | 677 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:42 |
| Q25L60X40P001 |   40.0 |  98.43% |     63099 | 4.46M | 129 |        36 | 24.04K | 660 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:20 |   0:00:40 |
| Q25L60X40P002 |   40.0 |  98.47% |     56929 | 4.47M | 134 |        44 | 29.23K | 708 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:14 |   0:00:39 |
| Q25L60X80P000 |   80.0 |  97.32% |     60890 | 4.46M | 147 |        39 |  14.1K | 326 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:31 |
| Q25L60X80P001 |   80.0 |  97.29% |     47016 | 4.46M | 164 |        49 | 17.78K | 355 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:35 |   0:00:31 |
| Q25L60X80P002 |   80.0 |  97.26% |     49208 | 4.46M | 161 |        79 | 20.13K | 359 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:35 |   0:00:31 |
| Q30L60X40P000 |   40.0 |  98.47% |     59661 | 4.46M | 130 |        42 | 28.51K | 716 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:02:14 |   0:00:39 |
| Q30L60X40P001 |   40.0 |  98.46% |     63674 | 4.46M | 130 |        35 | 25.78K | 742 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:38 |
| Q30L60X40P002 |   40.0 |  98.36% |     63600 | 4.46M | 132 |       775 | 41.95K | 720 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:19 |   0:00:37 |
| Q30L60X80P000 |   80.0 |  97.37% |     43865 | 4.46M | 161 |        60 | 17.63K | 356 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:36 |
| Q30L60X80P001 |   80.0 |  97.49% |     61181 | 4.46M | 142 |        93 | 16.84K | 312 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:31 |


Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.14% |     67338 | 4.46M | 119 |        77 | 15.25K | 250 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:49 |   0:00:50 |
| MRX40P001 |   40.0 |  97.20% |     64124 | 4.45M | 116 |       128 | 19.83K | 249 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:21 |   0:00:50 |
| MRX40P002 |   40.0 |  96.98% |     67296 | 4.46M | 115 |        72 | 12.59K | 230 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:22 |   0:00:31 |
| MRX80P000 |   80.0 |  96.87% |     64144 | 4.45M | 117 |       275 | 17.75K | 244 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:50 |   0:00:32 |
| MRX80P001 |   80.0 |  96.82% |     64140 | 4.29M | 113 |        72 |  10.3K | 228 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:42 |   0:00:32 |
| MRX80P002 |   80.0 |  96.83% |     73618 | 4.46M | 114 |        81 | 12.07K | 230 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:33 |


Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.30% |     63647 | 4.46M | 123 |       336 | 29.54K | 426 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:59 |   0:00:50 |
| Q0L0X40P001   |   40.0 |  98.11% |     64137 | 4.46M | 123 |        87 | 25.05K | 431 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:48 |
| Q0L0X40P002   |   40.0 |  98.12% |     60764 | 4.46M | 124 |        43 | 17.96K | 430 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:35 |
| Q0L0X80P000   |   80.0 |  97.17% |     63652 | 4.46M | 129 |       134 | 15.65K | 273 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:30 |
| Q0L0X80P001   |   80.0 |  97.09% |     57836 | 4.46M | 134 |        38 | 11.84K | 268 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:30 |
| Q0L0X80P002   |   80.0 |  97.20% |     60857 | 4.44M | 135 |        42 | 13.43K | 272 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:29 |
| Q25L60X40P000 |   40.0 |  98.41% |     63646 | 4.46M | 124 |        51 | 21.92K | 463 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:39 |
| Q25L60X40P001 |   40.0 |  98.16% |     67339 | 4.46M | 117 |        41 |  17.1K | 404 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:38 |
| Q25L60X40P002 |   40.0 |  98.44% |     63592 | 4.46M | 124 |        45 | 20.82K | 490 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:38 |
| Q25L60X80P000 |   80.0 |  97.52% |     67327 | 4.46M | 119 |        43 | 12.83K | 262 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:48 |   0:00:33 |
| Q25L60X80P001 |   80.0 |  97.25% |     64134 | 4.46M | 121 |        62 | 12.31K | 244 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:48 |   0:00:31 |
| Q25L60X80P002 |   80.0 |  97.27% |     63669 | 4.46M | 122 |        56 | 12.71K | 257 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:31 |
| Q30L60X40P000 |   40.0 |  98.41% |     64143 | 4.46M | 120 |        50 | 20.97K | 474 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:38 |
| Q30L60X40P001 |   40.0 |  98.26% |     64145 | 4.46M | 123 |        34 | 15.56K | 472 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:36 |
| Q30L60X40P002 |   40.0 |  98.26% |     63655 | 4.46M | 122 |        69 | 23.25K | 475 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:36 |
| Q30L60X80P000 |   80.0 |  97.51% |     59661 | 4.46M | 123 |       106 | 14.04K | 269 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:38 |
| Q30L60X80P001 |   80.0 |  97.58% |     67342 | 4.46M | 119 |       240 |    15K | 260 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:34 |


Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.90% |     64161 | 4.28M | 113 |        74 | 12.88K | 229 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:40 |
| MRX40P001 |   40.0 |  97.21% |     64143 | 4.45M | 114 |       422 | 21.98K | 248 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:42 |
| MRX40P002 |   40.0 |  96.84% |     73688 | 4.29M | 109 |        67 | 10.72K | 223 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:31 |
| MRX80P000 |   80.0 |  96.88% |     67247 | 4.46M | 115 |        89 | 15.05K | 230 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:00:50 |   0:00:34 |
| MRX80P001 |   80.0 |  96.84% |     67304 | 4.29M | 112 |        72 |  10.1K | 226 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:33 |
| MRX80P002 |   80.0 |  96.84% |     73675 | 4.46M | 114 |        85 | 13.06K | 233 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:00:51 |   0:00:37 |


Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |  # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|---:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  97.92% |     67361 | 4.46M | 112 |     85543 | 358.39K | 55 |  263.0 | 22.0 |  65.7 | 658.0 |   0:01:00 |
| 7_merge_mr_unitigs_bcalm      |  98.80% |     67350 | 4.46M | 112 |    102087 | 185.72K | 20 |  263.0 | 22.0 |  65.7 | 658.0 |   0:01:42 |
| 7_merge_mr_unitigs_superreads |  98.80% |     67363 | 4.46M | 112 |      1197 |  14.57K | 14 |  262.0 | 22.0 |  65.3 | 656.0 |   0:01:26 |
| 7_merge_mr_unitigs_tadpole    |  98.87% |     73714 | 4.46M | 111 |     85543 |  100.6K | 16 |  261.0 | 22.0 |  65.0 | 654.0 |   0:01:29 |
| 7_merge_unitigs_bcalm         |  98.88% |     67348 | 4.46M | 113 |     11897 | 118.27K | 50 |  263.0 | 23.0 |  64.7 | 664.0 |   0:01:28 |
| 7_merge_unitigs_superreads    |  98.94% |     67359 | 4.46M | 111 |      1032 |  25.28K | 23 |  262.0 | 22.0 |  65.3 | 656.0 |   0:01:35 |
| 7_merge_unitigs_tadpole       |  98.80% |     67354 | 4.46M | 112 |      1051 |  29.75K | 29 |  264.0 | 22.0 |  66.0 | 660.0 |   0:01:26 |


Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.45% |    114635 | 4.48M |  78 |      1048 | 18.49K | 157 |  263.0 | 22.0 |  65.7 | 658.0 |   0:00:51 |
| 8_mr_spades  |  98.94% |    143464 | 4.17M |  62 |       682 | 18.57K | 119 |  376.0 | 29.0 |  96.3 | 926.0 |   0:00:50 |
| 8_megahit    |  98.45% |     73665 | 4.46M | 116 |      1024 | 24.57K | 230 |  263.0 | 22.0 |  65.7 | 658.0 |   0:01:08 |
| 8_mr_megahit |  99.22% |    132660 |  4.5M |  75 |       673 |  22.4K | 146 |  376.0 | 28.0 |  97.3 | 920.0 |   0:01:00 |
| 8_platanus   |  97.80% |    114608 | 4.48M |  72 |       374 | 10.89K | 137 |  263.0 | 22.0 |  65.7 | 658.0 |   0:00:47 |


Table: statFinal

| Name                     |     N50 |     Sum |   # |
|:-------------------------|--------:|--------:|----:|
| Genome                   | 4583637 | 4583637 |   1 |
| Paralogs                 |    1737 |  188158 | 111 |
| Repetitives              |    1200 |  115795 | 213 |
| 7_merge_anchors.anchors  |   67361 | 4462976 | 112 |
| 7_merge_anchors.others   |   85543 |  358388 |  55 |
| glue_anchors             |   80527 | 4462648 | 104 |
| fill_anchors             |  108438 | 4464314 |  85 |
| spades.contig            |  125538 | 4514902 | 165 |
| spades.scaffold          |  143522 | 4515652 | 155 |
| spades.non-contained     |  132337 | 4493720 |  79 |
| mr_spades.contig         |  178373 | 4522934 |  83 |
| mr_spades.scaffold       |  203812 | 4523044 |  81 |
| mr_spades.non-contained  |  178373 | 4513608 |  58 |
| megahit.contig           |   82955 | 4507738 | 162 |
| megahit.non-contained    |   82955 | 4488116 | 114 |
| mr_megahit.contig        |  133730 | 4560575 | 164 |
| mr_megahit.non-contained |  133730 | 4521987 |  71 |
| platanus.contig          |  108571 | 4558622 | 408 |
| platanus.scaffold        |  125790 | 4514164 | 168 |
| platanus.non-contained   |  132296 | 4492200 |  65 |

