# Assemble genomes of model organisms by Anchr

- [Assemble genomes of model organisms by Anchr](#assemble-genomes-of-model-organisms-by-anchr)
    * [*Mycoplasma genitalium* G37](#mycoplasma-genitalium-g37)
        + [g37: download](#g37-download)
        + [g37: template](#g37-template)
        + [g37: run](#g37-run)
    * [*E. coli* str. K-12 substr. MG1655](#e-coli-str-k-12-substr-mg1655)
        + [mg1655: reference](#mg1655-reference)
        + [mg1655: download](#mg1655-download)
        + [mg1655: template](#mg1655-template)
        + [mg1655: run](#mg1655-run)
    * [*E. coli* str. K-12 substr. DH5alpha](#e-coli-str-k-12-substr-dh5alpha)
        + [dh5alpha: reference](#dh5alpha-reference)
        + [dh5alpha: download](#dh5alpha-download)
        + [dh5alpha: template](#dh5alpha-template)
        + [dh5alpha: run](#dh5alpha-run)

## *Mycoplasma genitalium* G37

### g37: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/g37/1_genome
cd ~/data/anchr/g37/1_genome

cp ~/data/anchr/ref/g37/genome.fa .

```

### g37: download

```shell script
mkdir -p ~/data/anchr/g37/ena
cd ~/data/anchr/g37/ena

cat << EOF > source.csv
ERX452667,G37,MiSeq
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -j 4 -x 4 -s 2 -c --file-allocation=none -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

```

| name | srx       | platform | layout | ilength | srr       | spots  | bases  |
|------|-----------|----------|--------|---------|-----------|--------|--------|
| G37  | ERX452667 | ILLUMINA | PAIRED | 447     | ERR486835 | 680644 | 97.37M |

* Illumina

```shell script
cd ~/data/anchr/g37

mkdir -p 2_illumina
cd 2_illumina

ln -s ../ena/ERR486835_1.fastq.gz R1.fq.gz
ln -s ../ena/ERR486835_2.fastq.gz R2.fq.gz

```

### g37: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=g37

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 580076 \
    --parallel 8 \
    --xmx 4g \
    \
    --fastqc \
    --insertsize \
    --fastk \
    \
    --trim "--dedupe --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "bcalm bifrost superreads tadpole" \
    --statp 2 \
    --readl 125 \
    --uscale 2 \
    --lscale 3 \
    --redo \
    \
    --extend

```

### g37: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=g37

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md
# rm -fr 4_down_sampling 6_down_sampling

bash 0_master.sh
# bash 0_cleanup.sh

```

## *E. coli* str. K-12 substr. MG1655

### mg1655: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/mg1655/1_genome
cd ~/data/anchr/mg1655/1_genome

cp ~/data/anchr/ref/mg1655/genome.fa .
cp ~/data/anchr/ref/mg1655/paralogs.fa .
cp ~/data/anchr/ref/mg1655/repetitives.fa .

```

### mg1655: download

* Illumina

```shell script
cd ~/data/anchr/mg1655

mkdir -p 2_illumina
cd 2_illumina

aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz
aria2c -x 9 -s 3 -c ftp://webdata:webdata@ussd-ftp.illumina.com/Data/SequencingRuns/MG1655/MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz

ln -s MiSeq_Ecoli_MG1655_110721_PF_R1.fastq.gz R1.fq.gz
ln -s MiSeq_Ecoli_MG1655_110721_PF_R2.fastq.gz R2.fq.gz

```

### mg1655: template

* Rsync to hpcc

```shell script
rsync -avP \
    ~/data/anchr/mg1655/ \
    wangq@202.119.37.251:data/anchr/mg1655

# rsync -avP wangq@202.119.37.251:data/anchr/mg1655/ ~/data/anchr/mg1655

```

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4641652 \
    --parallel 24 \
    --xmx 80g \
    --queue mpi \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --tile --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --bwa "Q25L60" \
    --gatk \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --readl 151 \
    --uscale 2 \
    --lscale 3 \
    --redo \
    \
    --extend \
    \
    --busco

```

### mg1655: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=mg1655

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md
# rm -fr 4_down_sampling 6_down_sampling

# BASE_NAME=mg1655 bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
# bkill -J "${BASE_NAME}-*"

# bash 0_master.sh
# bash 0_cleanup.sh

```

Table: statInsertSize

| Group             |  Mean | Median | STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|------:|-------------------------------:|
| R.genome.bbtools  | 321.9 |    298 | 967.3 |                         96.08% |
| R.tadpole.bbtools | 294.7 |    296 |  22.0 |                         81.33% |
| R.genome.picard   | 298.2 |    298 |  18.0 |                             FR |
| R.tadpole.picard  | 294.9 |    296 |  21.6 |                             FR |

Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 292       | 4390759         | 0.1630       | 54.20   |
| R.31 | 265       | 4388871         | 0.1382       | 54.04   |
| R.41 | 239       | 4375332         | 0.1108       | 53.90   |
| R.51 | 216       | 4407280         | 0.1088       | 53.78   |
| R.61 | 191       | 4393867         | 0.1081       | 53.68   |
| R.71 | 168       | 4390152         | 0.1042       | 53.58   |
| R.81 | 144       | 4387393         | 0.0989       | 53.50   |

Table: statReads

| Name        |     N50 |     Sum |        # |
|:------------|--------:|--------:|---------:|
| Genome      | 4641652 | 4641652 |        1 |
| Paralogs    |    1937 |  187300 |      106 |
| Repetitives |    1205 |  110138 |      206 |
| Illumina.R  |     151 |   1.73G | 11458940 |
| trim.R      |     149 |   1.43G | 10370832 |
| Q25L60      |     148 |   1.32G |  9945743 |
| Q30L60      |     128 |   1.11G |  9316333 |

Table: statTrimReads

| Name           | N50 |     Sum |        # |
|:---------------|----:|--------:|---------:|
| clumpify       | 151 |   1.73G | 11439000 |
| filteredbytile | 151 |   1.67G | 11069366 |
| highpass       | 151 |   1.66G | 10996268 |
| trim           | 149 |   1.43G | 10370832 |
| filter         | 149 |   1.43G | 10370832 |
| R1             | 150 | 736.47M |  5185416 |
| R2             | 144 | 690.82M |  5185416 |
| Rs             |   0 |       0 |        0 |

```text
#R.trim
#Matched	17496	0.15911%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	20743057
#error_kmers	16215186
#genomic_kmers	4527871
#main_peak	239
#genome_size_in_peaks	4599179
#genome_size	4644433
#haploid_genome_size	4644433
#fold_coverage	239
#haploid_fold_coverage	239
#ploidy	1
#percent_repeat_in_peaks	1.553
#percent_repeat	1.837
#start	center	stop	max	volume
```

Table: statMergeReads

| Name          | N50 |    Sum |        # |
|:--------------|----:|-------:|---------:|
| clumped       | 149 |  1.43G | 10369656 |
| ecco          | 149 |  1.43G | 10369656 |
| eccc          | 149 |  1.43G | 10369656 |
| ecct          | 149 |  1.42G | 10321128 |
| extended      | 189 |  1.83G | 10321128 |
| merged.raw    | 339 |  1.72G |  5092769 |
| unmerged.raw  | 175 | 19.87M |   135590 |
| unmerged.trim | 175 | 19.86M |   135530 |
| M1            | 339 |  1.71G |  5065247 |
| U1            | 181 | 10.45M |    67765 |
| U2            | 168 |  9.41M |    67765 |
| Us            |   0 |      0 |        0 |
| M.cor         | 338 |  1.74G | 10266024 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 271.6 |    277 |  23.6 |         10.85% |
| M.ihist.merge.txt  | 337.7 |    338 |  19.3 |         98.69% |

Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 307.5 |  285.8 |    7.06% | "93" | 4.64M | 4.66M |     1.00 | 0:02'49'' |
| Q25L60.R | 284.2 |  273.2 |    3.85% | "61" | 4.64M | 4.57M |     0.98 | 0:02'39'' |
| Q30L60.R | 238.4 |  233.6 |    2.03% | "71" | 4.64M | 4.55M |     0.98 | 0:02'23'' |

Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |    # | N50Others |    Sum |    # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|-----:|----------:|-------:|-----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  96.01% |      9976 |  4.5M |  668 |        33 | 39.13K | 1495 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:01:21 |   0:00:49 |
| Q0L0X40P001   |   40.0 |  96.07% |      9328 | 4.49M |  701 |        34 | 40.11K | 1532 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:48 |
| Q0L0X40P002   |   40.0 |  96.22% |      9040 |  4.5M |  697 |        35 | 44.42K | 1585 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:30 |
| Q0L0X80P000   |   80.0 |  92.29% |      4830 | 4.34M | 1165 |        40 | 81.39K | 2343 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:32 |
| Q0L0X80P001   |   80.0 |  92.35% |      4935 | 4.33M | 1162 |        49 | 91.58K | 2349 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:33 |
| Q0L0X80P002   |   80.0 |  91.90% |      5142 | 4.31M | 1144 |        45 | 86.25K | 2315 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:32 |
| Q25L60X40P000 |   40.0 |  97.97% |     17345 | 4.53M |  398 |        37 | 33.97K | 1251 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:34 |
| Q25L60X40P001 |   40.0 |  97.89% |     17793 | 4.54M |  395 |        38 | 34.57K | 1176 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:34 |
| Q25L60X40P002 |   40.0 |  97.82% |     19591 | 4.52M |  388 |        33 | 29.66K | 1175 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:34 |
| Q25L60X80P000 |   80.0 |  96.77% |     10970 | 4.49M |  618 |        45 | 48.15K | 1375 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:34 |
| Q25L60X80P001 |   80.0 |  96.67% |     10660 | 4.49M |  651 |        41 | 45.07K | 1404 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:10 |   0:00:35 |
| Q25L60X80P002 |   80.0 |  96.55% |      9737 | 4.49M |  663 |        44 | 48.83K | 1438 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:35 |
| Q30L60X40P000 |   40.0 |  98.57% |     31278 | 4.53M |  237 |        29 | 30.82K | 1310 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:42 |
| Q30L60X40P001 |   40.0 |  98.55% |     30996 | 4.53M |  236 |        28 | 28.84K | 1288 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:40 |
| Q30L60X40P002 |   40.0 |  98.59% |     28714 | 4.52M |  254 |        80 | 54.06K | 1332 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:41 |
| Q30L60X80P000 |   80.0 |  98.49% |     30851 | 4.54M |  242 |        33 | 29.58K | 1097 |   79.0 | 14.0 |  12.3 | 242.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:44 |
| Q30L60X80P001 |   80.0 |  98.50% |     30332 | 4.55M |  240 |        37 | 33.81K | 1093 |   79.0 | 13.0 |  13.3 | 236.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:43 |

Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.96% |     42614 | 4.52M | 188 |        89 | 28.75K | 458 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:44 |
| MRX40P001 |   40.0 |  96.95% |     46678 | 4.52M | 188 |        91 | 27.16K | 451 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:54 |   0:00:43 |
| MRX40P002 |   40.0 |  96.89% |     42127 | 4.51M | 195 |        84 | 27.56K | 460 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:54 |   0:00:31 |
| MRX80P000 |   80.0 |  96.31% |     28330 |  4.5M | 277 |       108 | 45.02K | 631 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:32 |   0:00:35 |
| MRX80P001 |   80.0 |  96.20% |     26887 |  4.5M | 279 |       108 | 41.96K | 620 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:31 |   0:00:33 |
| MRX80P002 |   80.0 |  96.23% |     27814 |  4.5M | 268 |       115 | 46.23K | 601 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:32 |   0:00:34 |

Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.92% |     21703 | 4.54M | 382 |        37 | 50.33K | 1697 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:52 |   0:01:06 |
| Q0L0X40P001   |   40.0 |  97.97% |     17907 | 4.53M | 374 |        37 | 49.41K | 1666 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:01:18 |   0:01:07 |
| Q0L0X40P002   |   40.0 |  97.93% |     19949 | 4.54M | 378 |        30 | 40.48K | 1672 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:36 |
| Q0L0X80P000   |   80.0 |  95.69% |      8783 | 4.48M | 731 |        45 | 63.39K | 1712 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:01:31 |   0:00:36 |
| Q0L0X80P001   |   80.0 |  95.89% |      8584 | 4.48M | 723 |        40 | 54.43K | 1677 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:31 |   0:00:34 |
| Q0L0X80P002   |   80.0 |  95.87% |      9482 | 4.47M | 698 |        43 | 57.94K | 1639 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:32 |   0:00:35 |
| Q25L60X40P000 |   40.0 |  98.44% |     26937 | 4.53M | 307 |        46 | 60.27K | 1627 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:01:22 |   0:00:40 |
| Q25L60X40P001 |   40.0 |  98.38% |     24895 | 4.54M | 308 |        36 | 48.07K | 1630 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:01:18 |   0:00:40 |
| Q25L60X40P002 |   40.0 |  98.34% |     25739 | 4.52M | 293 |        37 | 50.39K | 1591 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:01:17 |   0:00:39 |
| Q25L60X80P000 |   80.0 |  97.84% |     16199 | 4.53M | 427 |        42 | 46.19K | 1371 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:41 |
| Q25L60X80P001 |   80.0 |  97.69% |     15315 | 4.53M | 452 |        40 | 44.13K | 1383 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:38 |   0:00:39 |
| Q25L60X80P002 |   80.0 |  97.81% |     15274 | 4.52M | 471 |        44 | 52.65K | 1457 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:39 |
| Q30L60X40P000 |   40.0 |  98.42% |     21933 | 4.53M | 340 |        25 | 39.37K | 1838 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:41 |
| Q30L60X40P001 |   40.0 |  98.44% |     24077 | 4.53M | 312 |        31 | 47.55K | 1804 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:01:16 |   0:00:39 |
| Q30L60X40P002 |   40.0 |  98.40% |     23136 | 4.53M | 340 |        62 |    71K | 1797 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:01:20 |   0:00:39 |
| Q30L60X80P000 |   80.0 |  98.60% |     29369 | 4.54M | 255 |        32 |  40.2K | 1416 |   80.0 | 14.0 |  12.7 | 244.0 | "31,41,51,61,71,81" |   0:01:40 |   0:00:48 |
| Q30L60X80P001 |   80.0 |  98.53% |     28067 | 4.53M | 247 |        31 | 37.48K | 1374 |   80.0 | 14.0 |  12.7 | 244.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:45 |

Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.67% |     54055 | 4.52M | 142 |        89 | 19.23K | 307 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:57 |   0:00:50 |
| MRX40P001 |   40.0 |  96.75% |     57786 | 4.52M | 144 |        83 |    18K | 307 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:19 |   0:00:50 |
| MRX40P002 |   40.0 |  96.75% |     55838 | 4.52M | 144 |        82 |  18.4K | 315 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:20 |   0:00:32 |
| MRX80P000 |   80.0 |  96.38% |     47694 | 4.51M | 195 |       108 | 28.22K | 409 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:33 |
| MRX80P001 |   80.0 |  96.34% |     42717 | 4.51M | 194 |       103 |  26.1K | 398 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:37 |   0:00:34 |
| MRX80P002 |   80.0 |  96.39% |     36596 | 4.51M | 201 |       109 |  31.9K | 422 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:01:36 |   0:00:32 |

Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.14% |     32152 | 4.53M | 254 |        38 |  38.1K | 1202 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:01:24 |   0:00:44 |
| Q0L0X40P001   |   40.0 |  98.21% |     31145 | 4.53M | 257 |        44 | 43.98K | 1180 |   40.0 |  6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:45 |
| Q0L0X40P002   |   40.0 |  98.14% |     31615 | 4.54M | 242 |        32 | 31.08K | 1169 |   39.0 |  7.0 |   6.0 | 120.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:36 |
| Q0L0X80P000   |   80.0 |  97.17% |     16918 | 4.51M | 428 |        42 | 44.31K | 1183 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:35 |
| Q0L0X80P001   |   80.0 |  97.11% |     16189 | 4.51M | 440 |        39 | 45.42K | 1219 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:00:48 |   0:00:35 |
| Q0L0X80P002   |   80.0 |  97.13% |     16547 | 4.51M | 428 |        43 | 45.86K | 1186 |   79.0 | 11.0 |  15.3 | 224.0 | "31,41,51,61,71,81" |   0:00:47 |   0:00:34 |
| Q25L60X40P000 |   40.0 |  98.53% |     34606 | 4.53M | 226 |        34 | 35.95K | 1265 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:40 |
| Q25L60X40P001 |   40.0 |  98.52% |     34384 | 4.54M | 225 |        40 | 41.52K | 1238 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:39 |   0:00:41 |
| Q25L60X40P002 |   40.0 |  98.41% |     34327 | 4.52M | 222 |        36 | 38.92K | 1200 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:38 |
| Q25L60X80P000 |   80.0 |  98.08% |     26215 | 4.52M | 303 |        40 | 38.05K | 1089 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:44 |
| Q25L60X80P001 |   80.0 |  98.11% |     24249 | 4.53M | 314 |        40 | 39.12K | 1138 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:00:45 |   0:00:44 |
| Q25L60X80P002 |   80.0 |  98.05% |     20936 | 4.55M | 340 |        39 | 40.48K | 1225 |   79.0 | 12.0 |  14.3 | 230.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:40 |
| Q30L60X40P000 |   40.0 |  98.49% |     31674 | 4.53M | 261 |        27 | 34.15K | 1495 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:42 |
| Q30L60X40P001 |   40.0 |  98.50% |     29964 | 4.54M | 253 |        26 | 32.65K | 1496 |   40.0 |  8.0 |   5.3 | 128.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:41 |
| Q30L60X40P002 |   40.0 |  98.55% |     27852 | 4.54M | 275 |        36 | 47.12K | 1540 |   40.0 |  7.0 |   6.3 | 122.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:40 |
| Q30L60X80P000 |   80.0 |  98.58% |     34032 | 4.53M | 219 |        29 |  28.8K | 1209 |   80.0 | 14.0 |  12.7 | 244.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:46 |
| Q30L60X80P001 |   80.0 |  98.58% |     31599 | 4.53M | 221 |        40 | 40.82K | 1206 |   80.0 | 13.0 |  13.7 | 238.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:44 |

Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  96.77% |     73623 | 4.52M | 123 |        85 | 17.31K | 271 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:46 |   0:00:32 |
| MRX40P001 |   40.0 |  96.73% |     63113 | 4.52M | 125 |        92 |  18.2K | 266 |   40.0 |  6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:33 |
| MRX40P002 |   40.0 |  96.74% |     62484 | 4.52M | 123 |        80 | 16.45K | 274 |   39.0 |  6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:40 |   0:00:31 |
| MRX80P000 |   80.0 |  96.55% |     53721 | 4.52M | 151 |        94 | 21.47K | 322 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:33 |
| MRX80P001 |   80.0 |  96.51% |     52134 | 4.52M | 154 |        98 | 20.63K | 324 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:00:51 |   0:00:33 |
| MRX80P002 |   80.0 |  96.59% |     51879 | 4.51M | 159 |       105 |  24.9K | 337 |   79.0 | 10.0 |  16.3 | 218.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:35 |

Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |  # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|---:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  98.10% |     65406 | 4.54M | 129 |     36465 |  337.8K | 77 |  285.0 | 35.0 |  60.0 | 780.0 |   0:00:59 |
| 7_merge_mr_unitigs_bcalm      |  98.98% |     63214 | 4.53M | 133 |     60806 | 190.19K | 11 |  284.0 | 36.0 |  58.7 | 784.0 |   0:01:37 |
| 7_merge_mr_unitigs_superreads |  99.00% |     59608 | 4.52M | 137 |     12585 |  21.76K | 10 |  284.0 | 36.0 |  58.7 | 784.0 |   0:01:40 |
| 7_merge_mr_unitigs_tadpole    |  98.94% |     63616 | 4.53M | 130 |      1054 |   8.01K |  8 |  285.0 | 35.0 |  60.0 | 780.0 |   0:01:30 |
| 7_merge_unitigs_bcalm         |  99.01% |     58780 | 4.54M | 142 |      3101 | 105.89K | 53 |  284.0 | 35.0 |  59.7 | 778.0 |   0:01:57 |
| 7_merge_unitigs_superreads    |  98.80% |     58785 | 4.53M | 141 |      9743 |  98.18K | 40 |  284.0 | 36.0 |  58.7 | 784.0 |   0:01:45 |
| 7_merge_unitigs_tadpole       |  98.99% |     63481 | 4.53M | 133 |      1091 |  33.87K | 29 |  284.0 | 35.0 |  59.7 | 778.0 |   0:01:42 |

Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.54% |     95432 | 4.27M |  87 |      1124 | 16.23K | 160 |  286.0 | 35.0 |  60.3 | 782.0 |   0:00:43 |
| 8_mr_spades  |  98.39% |    148466 | 4.55M |  71 |      1029 | 23.15K | 146 |  372.0 | 36.0 |  88.0 | 960.0 |   0:00:49 |
| 8_megahit    |  98.18% |     65379 | 4.53M | 125 |       960 | 18.07K | 230 |  285.0 | 35.0 |  60.0 | 780.0 |   0:00:47 |
| 8_mr_megahit |  98.81% |    132772 | 4.57M |  70 |       781 | 19.29K | 138 |  372.0 | 36.0 |  88.0 | 960.0 |   0:01:03 |
| 8_platanus   |  97.87% |     96792 | 4.06M |  81 |       254 |  9.36K | 138 |  285.0 | 35.0 |  60.0 | 780.0 |   0:00:43 |

Table: statFinal

| Name                     |     N50 |     Sum |    # |
|:-------------------------|--------:|--------:|-----:|
| Genome                   | 4641652 | 4641652 |    1 |
| Paralogs                 |    1937 |  187300 |  106 |
| Repetitives              |    1205 |  110138 |  206 |
| 7_merge_anchors.anchors  |   65406 | 4539016 |  129 |
| 7_merge_anchors.others   |   36465 |  337798 |   77 |
| glue_anchors             |   78838 | 4538271 |  121 |
| fill_anchors             |   97542 | 4538788 |   89 |
| spades.contig            |  132608 | 4573776 |  137 |
| spades.scaffold          |  133063 | 4573960 |  133 |
| spades.non-contained     |  132608 | 4555564 |   75 |
| mr_spades.contig         |  148607 | 4587905 |  150 |
| mr_spades.scaffold       |  148607 | 4588105 |  148 |
| mr_spades.non-contained  |  148607 | 4569064 |   75 |
| megahit.contig           |   82825 | 4568046 |  147 |
| megahit.non-contained    |   82825 | 4551522 |  105 |
| mr_megahit.contig        |  132896 | 4609378 |  128 |
| mr_megahit.non-contained |  132896 | 4584972 |   68 |
| platanus.contig          |   14890 | 4712258 | 1171 |
| platanus.scaffold        |  148483 | 4577638 |  142 |
| platanus.non-contained   |  176491 | 4559854 |   64 |

Table: statBusco run_bacteria_odb10

| NAME                |   C |   S |  D |  F |  M | Total |
|:--------------------|----:|----:|---:|---:|---:|------:|
| Genome              | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_superreads    | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_bcalm         | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_tadpole       | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_mr_superreads | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_mr_bcalm      | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_mr_tadpole    | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_anchors       | 124 | 124 |  0 |  0 |  0 |   124 |
| glue_anchors        | 124 | 124 |  0 |  0 |  0 |   124 |
| fill_anchors        | 124 | 124 |  0 |  0 |  0 |   124 |
| spades              | 124 | 124 |  0 |  0 |  0 |   124 |
| mr_spades           | 124 | 124 |  0 |  0 |  0 |   124 |
| megahit             | 124 | 124 |  0 |  0 |  0 |   124 |
| mr_megahit          | 124 | 124 |  0 |  0 |  0 |   124 |
| platanus            | 124 | 124 |  0 |  0 |  0 |   124 |

Table: statBusco run_enterobacterales_odb10

| NAME                |   C |   S |  D |  F |  M | Total |
|:--------------------|----:|----:|---:|---:|---:|------:|
| Genome              | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_superreads    | 437 | 435 |  2 |  3 |  0 |   440 |
| merge_bcalm         | 437 | 435 |  2 |  3 |  0 |   440 |
| merge_tadpole       | 437 | 435 |  2 |  3 |  0 |   440 |
| merge_mr_superreads | 438 | 436 |  2 |  2 |  0 |   440 |
| merge_mr_bcalm      | 438 | 436 |  2 |  2 |  0 |   440 |
| merge_mr_tadpole    | 438 | 436 |  2 |  2 |  0 |   440 |
| merge_anchors       | 438 | 436 |  2 |  2 |  0 |   440 |
| glue_anchors        | 439 | 437 |  2 |  1 |  0 |   440 |
| fill_anchors        | 439 | 437 |  2 |  1 |  0 |   440 |
| spades              | 440 | 438 |  2 |  0 |  0 |   440 |
| mr_spades           | 440 | 438 |  2 |  0 |  0 |   440 |
| megahit             | 440 | 438 |  2 |  0 |  0 |   440 |
| mr_megahit          | 440 | 438 |  2 |  0 |  0 |   440 |
| platanus            | 440 | 438 |  2 |  0 |  0 |   440 |

## *E. coli* str. K-12 substr. DH5alpha

### dh5alpha: reference

* Reference genome

```shell script
mkdir -p ~/data/anchr/dh5alpha/1_genome
cd ~/data/anchr/dh5alpha/1_genome

cp ~/data/anchr/ref/dh5alpha/genome.fa .
cp ~/data/anchr/ref/dh5alpha/paralogs.fa .
cp ~/data/anchr/ref/dh5alpha/repetitives.fa .

```

### dh5alpha: download

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p ena
cd ena

cat << EOF > source.csv
SRP251726,dh5alpha,HiSeq 2500 PE125
EOF

anchr ena info | perl - -v source.csv > ena_info.yml
anchr ena prep | perl - ena_info.yml

mlr --icsv --omd cat ena_info.csv

aria2c -x 9 -s 3 -c -i ena_info.ftp.txt

md5sum --check ena_info.md5.txt

```

| name     | srx        | platform        | layout | ilength | srr         | spot    | base  |
|:---------|:-----------|:----------------|:-------|:--------|:------------|:--------|:------|
| dh5alpha | SRX7856678 | ILLUMINA        | PAIRED |         | SRR11245239 | 5881654 | 1.37G |
| dh5alpha | SRX7856679 | OXFORD_NANOPORE | SINGLE |         | SRR11245238 | 346489  | 3.35G |

* Illumina

```shell script
cd ~/data/anchr/dh5alpha

mkdir -p 2_illumina
cd 2_illumina

ln -s ../ena/SRR11245239_1.fastq.gz R1.fq.gz
ln -s ../ena/SRR11245239_2.fastq.gz R2.fq.gz

```

### dh5alpha: template

* template

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}

rm *.sh
anchr template \
    --genome 4583637 \
    --parallel 24 \
    --xmx 80g \
    --queue mpi \
    \
    --fastqc \
    --insertsize \
    --kat \
    \
    --trim "--dedupe --cutoff 30 --cutk 31" \
    --qual "25 30" \
    --len "60" \
    --filter "adapter artifact" \
    \
    --quorum \
    --merge \
    --ecphase "1 2 3" \
    \
    --cov "40 80" \
    --unitigger "superreads bcalm tadpole" \
    --statp 2 \
    --readl 125 \
    --uscale 2 \
    --lscale 3 \
    --redo \
    \
    --extend \
    \
    --busco

```

### dh5alpha: run

```shell script
WORKING_DIR=${HOME}/data/anchr
BASE_NAME=dh5alpha

cd ${WORKING_DIR}/${BASE_NAME}
# rm -fr 4_*/ 6_*/ 7_*/ 8_*/
# rm -fr 2_illumina/trim 2_illumina/merge statReads.md
# rm -fr 4_down_sampling 6_down_sampling

# BASE_NAME=dh5alpha bash 0_bsub.sh
bsub -q mpi -n 24 -J "${BASE_NAME}-0_master" "bash 0_master.sh"
# bkill -J "${BASE_NAME}-*"

# bash 0_master.sh
# bash 0_cleanup.sh

```

Table: statInsertSize

| Group             |  Mean | Median |  STDev | PercentOfPairs/PairOrientation |
|:------------------|------:|-------:|-------:|-------------------------------:|
| R.genome.bbtools  | 432.7 |    346 | 1241.2 |                         99.98% |
| R.tadpole.bbtools | 389.1 |    341 |  206.1 |                         95.12% |
| R.genome.picard   | 394.8 |    346 |  208.3 |                             FR |
| R.tadpole.picard  | 389.1 |    341 |  205.7 |                             FR |

Table: statKAT

| k    | mean_freq | est_genome_size | est_het_rate | mean_gc |
|:-----|:----------|:----------------|:-------------|:--------|
| R.21 | 260       | 4346013         | 0.0814       | 51.62   |
| R.31 | 233       | 4360998         | 0.0563       | 51.54   |
| R.41 | 209       | 4433088         | 0.0607       | 51.50   |
| R.51 | 182       | 4413726         | 0.0150       | 51.47   |
| R.61 | 157       | 4456634         | 0.0332       | 51.45   |
| R.71 | 132       | 4495949         | 0.0194       | 51.44   |
| R.81 | 107       | 4475448         | 0.0160       | 51.42   |

Table: statReads

| Name        |     N50 |     Sum |        # |
|:------------|--------:|--------:|---------:|
| Genome      | 4583637 | 4583637 |        1 |
| Paralogs    |    1737 |  188158 |      111 |
| Repetitives |    1200 |  115795 |      213 |
| Illumina.R  |     125 |   1.47G | 11763308 |
| trim.R      |     125 |   1.37G | 10962262 |
| Q25L60      |     125 |   1.25G | 10280938 |
| Q30L60      |     125 |   1.13G |  9405535 |

Table: statTrimReads

| Name     | N50 |     Sum |        # |
|:---------|----:|--------:|---------:|
| clumpify | 125 |   1.37G | 10970530 |
| highpass | 125 |   1.37G | 10966136 |
| trim     | 125 |   1.37G | 10962262 |
| filter   | 125 |   1.37G | 10962262 |
| R1       | 125 |    683M |  5481131 |
| R2       | 125 | 683.63M |  5481131 |
| Rs       |   0 |       0 |        0 |

```text
#R.trim
#Matched	5620	0.05125%
#Name	Reads	ReadsPct
```

```text
#R.filter
#Matched	0	0.00000%
#Name	Reads	ReadsPct
```

```text
#R.peaks
#k	31
#unique_kmers	35779038
#error_kmers	31288377
#genomic_kmers	4490661
#main_peak	221
#genome_size_in_peaks	9162708
#genome_size	10213708
#haploid_genome_size	5106854
#fold_coverage	96
#haploid_fold_coverage	221
#ploidy	2
#het_rate	0.00012
#percent_repeat_in_peaks	2.524
#percent_repeat	55.129
#start	center	stop	max	volume
```

Table: statMergeReads

| Name          | N50 |     Sum |        # |
|:--------------|----:|--------:|---------:|
| clumped       | 125 |   1.37G | 10959444 |
| ecco          | 125 |   1.37G | 10959444 |
| eccc          | 125 |   1.37G | 10959444 |
| ecct          | 125 |   1.37G | 10952912 |
| extended      | 165 |    1.8G | 10952912 |
| merged.raw    | 343 |    1.1G |  3511622 |
| unmerged.raw  | 165 | 645.89M |  3929668 |
| unmerged.trim | 165 | 645.89M |  3929668 |
| M1            | 343 |   1.06G |  3404325 |
| U1            | 165 | 322.85M |  1964834 |
| U2            | 165 | 323.04M |  1964834 |
| Us            |   0 |       0 |        0 |
| M.cor         | 250 |   1.71G | 10738318 |

| Group              |  Mean | Median | STDev | PercentOfPairs |
|:-------------------|------:|-------:|------:|---------------:|
| M.ihist.merge1.txt | 180.2 |    180 |  32.0 |         24.10% |
| M.ihist.merge.txt  | 312.5 |    310 |  87.7 |         64.12% |

Table: statQuorum

| Name     | CovIn | CovOut | Discard% | Kmer | RealG |  EstG | Est/Real |   RunTime |
|:---------|------:|-------:|---------:|-----:|------:|------:|---------:|----------:|
| Q0L0.R   | 298.2 |  262.2 |   12.07% | "87" | 4.58M |  4.6M |     1.00 | 0:02'44'' |
| Q25L60.R | 273.7 |  253.6 |    7.35% | "87" | 4.58M | 4.53M |     0.99 | 0:02'37'' |
| Q30L60.R | 246.5 |  232.4 |    5.71% | "87" | 4.58M | 4.52M |     0.99 | 0:02'14'' |

Table: statUnitigsSuperreads.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |    # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|-----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  97.18% |     31096 | 4.45M | 262 |        48 | 29.67K |  643 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:37 |
| Q0L0X40P001   |   40.0 |  97.13% |     27144 | 4.45M | 294 |        38 | 27.09K |  695 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:38 |
| Q0L0X40P002   |   40.0 |  97.15% |     26026 | 4.46M | 264 |        33 |  20.4K |  635 |   39.0 | 6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:29 |
| Q0L0X80P000   |   80.0 |  95.95% |     12946 | 4.42M | 527 |        35 | 44.34K | 1119 |   79.0 | 9.0 |  17.3 | 212.0 | "31,41,51,61,71,81" |   0:01:08 |   0:00:33 |
| Q0L0X80P001   |   80.0 |  95.85% |     12796 | 4.42M | 546 |        33 |  41.4K | 1165 |   79.0 | 9.0 |  17.3 | 212.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:32 |
| Q0L0X80P002   |   80.0 |  95.91% |     13279 | 4.42M | 519 |        33 | 40.39K | 1109 |   79.0 | 9.0 |  17.3 | 212.0 | "31,41,51,61,71,81" |   0:01:08 |   0:00:33 |
| Q25L60X40P000 |   40.0 |  97.85% |     43277 | 4.46M | 175 |        39 | 18.54K |  490 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:33 |
| Q25L60X40P001 |   40.0 |  97.67% |     42111 | 4.46M | 192 |        38 | 19.84K |  496 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:31 |
| Q25L60X40P002 |   40.0 |  97.70% |     40217 | 4.46M | 195 |        35 | 18.86K |  540 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:32 |
| Q25L60X80P000 |   80.0 |  97.17% |     27143 | 4.45M | 296 |        32 | 22.85K |  674 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:32 |
| Q25L60X80P001 |   80.0 |  97.03% |     21632 | 4.45M | 329 |        35 | 28.96K |  740 |   79.0 | 9.0 |  17.3 | 212.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:33 |
| Q25L60X80P002 |   80.0 |  96.92% |     26330 | 4.45M | 307 |        32 | 24.85K |  683 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:07 |   0:00:31 |
| Q30L60X40P000 |   40.0 |  97.78% |     46178 | 4.45M | 163 |        36 | 16.43K |  458 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:31 |
| Q30L60X40P001 |   40.0 |  97.79% |     47349 | 4.47M | 180 |        49 | 21.01K |  504 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:33 |
| Q30L60X40P002 |   40.0 |  97.83% |     44914 | 4.46M | 171 |     24692 | 41.58K |  506 |   39.0 | 6.0 |   7.0 | 114.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:31 |
| Q30L60X80P000 |   80.0 |  97.29% |     27104 | 4.45M | 280 |        32 | 23.52K |  640 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:32 |
| Q30L60X80P001 |   80.0 |  97.29% |     27205 | 4.45M | 275 |        34 | 25.26K |  642 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:06 |   0:00:32 |

Table: statMRUnitigsSuperreads.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.63% |     67275 | 4.45M | 122 |        81 | 20.95K | 331 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:34 |
| MRX40P001 |   40.0 |  97.59% |     67328 | 4.45M | 117 |        91 | 23.48K | 323 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:53 |   0:00:35 |
| MRX40P002 |   40.0 |  97.58% |     67196 | 4.46M | 118 |        76 | 18.09K | 332 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:36 |
| MRX80P000 |   80.0 |  97.48% |     64119 | 4.46M | 121 |        72 | 16.97K | 324 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:38 |
| MRX80P001 |   80.0 |  97.59% |     67250 | 4.46M | 113 |        77 | 17.27K | 325 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:28 |   0:00:38 |
| MRX80P002 |   80.0 |  97.56% |     64176 | 4.46M | 121 |        84 | 19.81K | 335 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:27 |   0:00:40 |

Table: statUnitigsBcalm.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.29% |     57013 | 4.46M | 141 |        48 | 32.68K | 697 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:21 |   0:00:39 |
| Q0L0X40P001   |   40.0 |  98.25% |     54641 | 4.46M | 149 |        46 | 34.29K | 710 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:40 |
| Q0L0X40P002   |   40.0 |  98.24% |     58771 | 4.46M | 131 |        91 | 37.62K | 683 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:17 |   0:00:46 |
| Q0L0X80P000   |   80.0 |  97.00% |     35763 | 4.46M | 220 |        43 | 22.87K | 471 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:34 |
| Q0L0X80P001   |   80.0 |  96.98% |     41931 | 4.46M | 199 |        39 | 18.45K | 414 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:33 |   0:00:33 |
| Q0L0X80P002   |   80.0 |  97.05% |     40711 | 4.45M | 195 |        59 |  22.8K | 427 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:28 |   0:00:32 |
| Q25L60X40P000 |   40.0 |  98.40% |     58790 | 4.46M | 132 |        42 | 28.14K | 692 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:24 |   0:00:42 |
| Q25L60X40P001 |   40.0 |  98.45% |     63674 | 4.46M | 129 |     39382 | 69.62K | 711 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:12 |   0:00:40 |
| Q25L60X40P002 |   40.0 |  98.50% |     63682 | 4.46M | 134 |        35 | 24.75K | 735 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:16 |   0:00:41 |
| Q25L60X80P000 |   80.0 |  97.40% |     51384 | 4.46M | 157 |        40 | 15.07K | 338 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:26 |   0:00:33 |
| Q25L60X80P001 |   80.0 |  97.25% |     43280 | 4.46M | 171 |        50 | 18.74K | 379 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:30 |   0:00:33 |
| Q25L60X80P002 |   80.0 |  97.24% |     52376 | 4.36M | 154 |        48 | 16.56K | 338 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:25 |   0:00:33 |
| Q30L60X40P000 |   40.0 |  98.43% |     60844 | 4.46M | 132 |        43 | 30.05K | 713 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:40 |
| Q30L60X40P001 |   40.0 |  98.44% |     60874 | 4.47M | 136 |        47 | 34.16K | 775 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:13 |   0:00:43 |
| Q30L60X40P002 |   40.0 |  98.47% |     60871 | 4.46M | 132 |        39 |  28.7K | 773 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:01:11 |   0:00:40 |
| Q30L60X80P000 |   80.0 |  97.34% |     54822 | 4.46M | 149 |        66 | 16.58K | 327 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:29 |   0:00:32 |
| Q30L60X80P001 |   80.0 |  97.44% |     53170 | 4.46M | 157 |        43 | 16.11K | 358 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:01:32 |   0:00:33 |

Table: statMRUnitigsBcalm.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.16% |     73681 | 4.45M | 118 |       601 | 23.46K | 259 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:21 |   0:00:30 |
| MRX40P001 |   40.0 |  97.16% |     67315 | 4.45M | 116 |       198 | 21.29K | 254 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:18 |   0:00:32 |
| MRX40P002 |   40.0 |  97.09% |     67196 | 4.46M | 118 |        87 | 15.22K | 247 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:01:17 |   0:00:31 |
| MRX80P000 |   80.0 |  96.85% |     67364 | 4.28M | 112 |        74 | 11.51K | 228 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:41 |   0:00:36 |
| MRX80P001 |   80.0 |  96.84% |     73702 | 4.46M | 111 |        81 | 12.61K | 226 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:36 |   0:00:33 |
| MRX80P002 |   80.0 |  96.84% |     66225 | 4.46M | 118 |        80 | 13.06K | 236 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:01:36 |   0:00:34 |

Table: statUnitigsTadpole.md

| Name          | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:--------------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| Q0L0X40P000   |   40.0 |  98.12% |     64138 | 4.46M | 120 |      1017 | 26.83K | 410 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:37 |   0:00:34 |
| Q0L0X40P001   |   40.0 |  98.09% |     60860 | 4.45M | 123 |       358 |  25.5K | 404 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:36 |
| Q0L0X40P002   |   40.0 |  98.03% |     67329 | 4.46M | 115 |        46 | 17.04K | 384 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:35 |
| Q0L0X80P000   |   80.0 |  97.16% |     60844 | 4.46M | 133 |        61 | 15.94K | 287 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:33 |
| Q0L0X80P001   |   80.0 |  97.18% |     57835 | 4.46M | 133 |        42 | 13.36K | 282 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:32 |
| Q0L0X80P002   |   80.0 |  97.25% |     64115 | 4.46M | 125 |       384 | 16.32K | 268 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:31 |
| Q25L60X40P000 |   40.0 |  98.32% |     63680 | 4.46M | 121 |        64 | 22.67K | 468 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:35 |   0:00:38 |
| Q25L60X40P001 |   40.0 |  98.31% |     66049 | 4.46M | 118 |        93 | 25.49K | 443 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:33 |   0:00:37 |
| Q25L60X40P002 |   40.0 |  98.26% |     63682 | 4.46M | 122 |        41 | 19.75K | 473 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:36 |   0:00:37 |
| Q25L60X80P000 |   80.0 |  97.34% |     63593 | 4.46M | 123 |        40 | 10.78K | 245 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:32 |
| Q25L60X80P001 |   80.0 |  97.30% |     58770 | 4.46M | 130 |        57 | 14.49K | 284 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:34 |
| Q25L60X80P002 |   80.0 |  97.36% |     63641 | 4.46M | 119 |       541 | 16.04K | 255 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:43 |   0:00:34 |
| Q30L60X40P000 |   40.0 |  98.25% |     63654 | 4.46M | 123 |        45 | 19.77K | 481 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:38 |
| Q30L60X40P001 |   40.0 |  98.31% |     64116 | 4.46M | 119 |      1343 | 41.65K | 496 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:34 |   0:00:39 |
| Q30L60X40P002 |   40.0 |  98.36% |     63678 | 4.46M | 122 |        46 | 21.18K | 498 |   40.0 | 6.0 |   7.3 | 116.0 | "31,41,51,61,71,81" |   0:00:33 |   0:00:37 |
| Q30L60X80P000 |   80.0 |  97.36% |     67334 | 4.46M | 118 |        50 | 11.82K | 247 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:33 |
| Q30L60X80P001 |   80.0 |  97.46% |     63673 | 4.46M | 122 |        51 | 12.96K | 271 |   80.0 | 9.0 |  17.7 | 214.0 | "31,41,51,61,71,81" |   0:00:42 |   0:00:33 |

Table: statMRUnitigsTadpole.md

| Name      | CovCor | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median | MAD | lower | upper |                Kmer | RunTimeUT | RunTimeAN |
|:----------|-------:|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|----:|------:|------:|--------------------:|----------:|----------:|
| MRX40P000 |   40.0 |  97.00% |     73666 | 4.45M | 117 |        92 | 15.38K | 230 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:38 |   0:00:30 |
| MRX40P001 |   40.0 |  96.98% |     67328 | 4.45M | 115 |       446 | 22.53K | 240 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:41 |   0:00:30 |
| MRX40P002 |   40.0 |  97.04% |     67196 | 4.46M | 117 |       103 | 16.34K | 236 |   40.0 | 5.0 |   8.3 | 110.0 | "31,41,51,61,71,81" |   0:00:44 |   0:00:30 |
| MRX80P000 |   80.0 |  96.87% |     73678 | 4.46M | 115 |        86 | 14.15K | 234 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:00:49 |   0:00:31 |
| MRX80P001 |   80.0 |  96.86% |     73689 | 4.46M | 111 |       103 | 13.52K | 229 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:00:52 |   0:00:33 |
| MRX80P002 |   80.0 |  96.85% |     66221 | 4.46M | 116 |        81 | 12.82K | 234 |   80.0 | 8.0 |  18.7 | 208.0 | "31,41,51,61,71,81" |   0:00:49 |   0:00:32 |

Table: statMergeAnchors.md

| Name                          | Mapped% | N50Anchor |   Sum |   # | N50Others |     Sum |  # | median |  MAD | lower | upper | RunTimeAN |
|:------------------------------|--------:|----------:|------:|----:|----------:|--------:|---:|-------:|-----:|------:|------:|----------:|
| 7_merge_anchors               |  97.99% |     67358 | 4.46M | 112 |     24692 | 203.78K | 42 |  262.0 | 22.0 |  65.3 | 656.0 |   0:00:57 |
| 7_merge_mr_unitigs_bcalm      |  98.74% |     67362 | 4.46M | 112 |      1345 |  17.74K | 12 |  263.0 | 22.0 |  65.7 | 658.0 |   0:01:17 |
| 7_merge_mr_unitigs_superreads |  98.75% |     67361 | 4.46M | 111 |      3330 |   14.7K |  8 |  263.0 | 23.0 |  64.7 | 664.0 |   0:01:23 |
| 7_merge_mr_unitigs_tadpole    |  98.85% |     73698 | 4.46M | 110 |      1377 |  16.47K | 10 |  260.0 | 22.0 |  64.7 | 652.0 |   0:01:26 |
| 7_merge_unitigs_bcalm         |  98.93% |     67344 | 4.46M | 111 |     39382 | 154.36K | 40 |  263.0 | 22.0 |  65.7 | 658.0 |   0:01:35 |
| 7_merge_unitigs_superreads    |  98.98% |     67353 | 4.46M | 113 |     23775 | 126.43K | 32 |  262.0 | 22.0 |  65.3 | 656.0 |   0:01:43 |
| 7_merge_unitigs_tadpole       |  98.88% |     67350 | 4.46M | 111 |     19926 |  77.76K | 26 |  262.0 | 22.0 |  65.3 | 656.0 |   0:01:36 |

Table: statOtherAnchors.md

| Name         | Mapped% | N50Anchor |   Sum |   # | N50Others |    Sum |   # | median |  MAD | lower | upper | RunTimeAN |
|:-------------|--------:|----------:|------:|----:|----------:|-------:|----:|-------:|-----:|------:|------:|----------:|
| 8_spades     |  98.45% |    114635 | 4.48M |  78 |      1048 | 18.49K | 157 |  263.0 | 22.0 |  65.7 | 658.0 |   0:00:41 |
| 8_mr_spades  |  98.93% |    132590 | 4.22M |  64 |       682 |  18.9K | 123 |  376.0 | 28.0 |  97.3 | 920.0 |   0:00:46 |
| 8_megahit    |  98.47% |     73665 | 4.46M | 116 |      1031 | 25.17K | 231 |  263.0 | 22.0 |  65.7 | 658.0 |   0:00:41 |
| 8_mr_megahit |  99.22% |    132754 |  4.5M |  75 |       673 | 22.14K | 147 |  376.0 | 28.0 |  97.3 | 920.0 |   0:00:46 |
| 8_platanus   |  97.80% |    114608 | 4.48M |  72 |       374 | 10.89K | 137 |  263.0 | 22.0 |  65.7 | 658.0 |   0:00:42 |

Table: statFinal

| Name                     |     N50 |     Sum |   # |
|:-------------------------|--------:|--------:|----:|
| Genome                   | 4583637 | 4583637 |   1 |
| Paralogs                 |    1737 |  188158 | 111 |
| Repetitives              |    1200 |  115795 | 213 |
| 7_merge_anchors.anchors  |   67358 | 4463149 | 112 |
| 7_merge_anchors.others   |   24692 |  203783 |  42 |
| glue_anchors             |   85589 | 4462785 | 104 |
| fill_anchors             |  102355 | 4462986 |  91 |
| spades.contig            |  125538 | 4514902 | 165 |
| spades.scaffold          |  143522 | 4515652 | 155 |
| spades.non-contained     |  132337 | 4493720 |  79 |
| mr_spades.contig         |  155808 | 4522785 |  87 |
| mr_spades.scaffold       |  203812 | 4522995 |  84 |
| mr_spades.non-contained  |  155808 | 4512983 |  60 |
| megahit.contig           |   82955 | 4508030 | 164 |
| megahit.non-contained    |   82955 | 4488852 | 115 |
| mr_megahit.contig        |  133730 | 4557477 | 156 |
| mr_megahit.non-contained |  133730 | 4522636 |  72 |
| platanus.contig          |  108571 | 4558622 | 408 |
| platanus.scaffold        |  125790 | 4514164 | 168 |
| platanus.non-contained   |  132296 | 4492200 |  65 |

Table: statBusco run_bacteria_odb10

| NAME                |   C |   S |  D |  F |  M | Total |
|:--------------------|----:|----:|---:|---:|---:|------:|
| Genome              | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_superreads    | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_bcalm         | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_tadpole       | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_mr_superreads | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_mr_bcalm      | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_mr_tadpole    | 124 | 124 |  0 |  0 |  0 |   124 |
| merge_anchors       | 124 | 124 |  0 |  0 |  0 |   124 |
| glue_anchors        | 124 | 124 |  0 |  0 |  0 |   124 |
| fill_anchors        | 124 | 124 |  0 |  0 |  0 |   124 |
| spades              | 124 | 124 |  0 |  0 |  0 |   124 |
| mr_spades           | 124 | 124 |  0 |  0 |  0 |   124 |
| megahit             | 124 | 124 |  0 |  0 |  0 |   124 |
| mr_megahit          | 124 | 124 |  0 |  0 |  0 |   124 |
| platanus            | 124 | 124 |  0 |  0 |  0 |   124 |

Table: statBusco run_enterobacterales_odb10

| NAME                |   C |   S |  D |  F |  M | Total |
|:--------------------|----:|----:|---:|---:|---:|------:|
| Genome              | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_superreads    | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_bcalm         | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_tadpole       | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_mr_superreads | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_mr_bcalm      | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_mr_tadpole    | 440 | 438 |  2 |  0 |  0 |   440 |
| merge_anchors       | 440 | 438 |  2 |  0 |  0 |   440 |
| glue_anchors        | 440 | 438 |  2 |  0 |  0 |   440 |
| fill_anchors        | 440 | 438 |  2 |  0 |  0 |   440 |
| spades              | 440 | 438 |  2 |  0 |  0 |   440 |
| mr_spades           | 440 | 438 |  2 |  0 |  0 |   440 |
| megahit             | 440 | 438 |  2 |  0 |  0 |   440 |
| mr_megahit          | 440 | 438 |  2 |  0 |  0 |   440 |
| platanus            | 440 | 438 |  2 |  0 |  0 |   440 |

